---
title: 'WP3: Data preparation in ALSPAC'
author: "Gemma Sharp and Kayleigh Easey"
date: '2019-07-01'
output: 
  rmdformats::material
---
# Introduction {.tabset}

## Description {.tabset}

## Connect to ALSPAC server {.tabset}
Before you begin, you must be connected to the ALSPAC server. Command + K for "Connect to server" then smb://central-gpfs.isys.bris.ac.uk/ALSPAC-Data

## Tidyverse {.tabset}
We are using tidyverse, which is a set of packages that work in harmony. The tidyverse package is designed to make it easy to install and load core packages from the tidyverse in a single command.
```{r eval=FALSE, include=TRUE}
library("tidyverse")
```

# Linking ALSPAC trios {.tabset}

## Linking
* ALN is a pregnancy ID
* QLET is a child ID for a specific pregnancy (i.e. in a twin pair, twin 1 will have qlet=A and twin 2 will have qlet=B)
* Sometimes the same mother (or father) had more than one pregnancy in the 18 month recruitment period. Those individuals have two ALNs.
  +  There were 657 pregnancies to women who had more than one pregnancy within the core ALSPAC sample. It should be noted that many of these pregnancies involved miscarriages and/or terminations and do not necessarily relate to more than one live born outcome of pregnancy (this info is from the MZ documentation file).
* For EPoCH, it would be useful to create individual IDs for mothers, fathers and children, and a family ID to link them. This is because we are interested in performing sibling comparisons.

* We can do this as follows:
  + the ALSPAC sample is described in the cohort profile. This file includes information about whether the pregnancy was included in the core ALSPAC sample or later phases, as well as child's sex and whether they survived at one year:
```{r eval=FALSE, include=TRUE}
library(haven) #load package to read dta files

cp<-read_dta("/Volumes/sscm alspac/Data/Current/Other/Cohort Profile/cp_2b.dta")

cp$covs_sex <- NA
cp$covs_sex[cp$kz021==2]<-"female" #Female
cp$covs_sex[cp$kz021==1]<-"male" #Male

cp$survive_one_year <-NA
cp$survive_one_year[cp$kz011b==1] <-1
cp$survive_one_year[cp$kz011b==2] <-0

```
  
  + the mz file is at the "pregnancy level"
```{r eval=FALSE, include=TRUE}
mz<-read_dta("/Volumes/sscm alspac/Data/Current/Other/Sample Definition/mz_5a.dta") #child sample definition
mz <- mz[mz$mz001==1,] #select core sample pregnancies; n=14,541
mz <- mz[-which(mz$mz010<1),] #remove 69 pregnancies with no known outcome; n=14,472
```
  
  + the kz file is at the "child level"
```{r eval=FALSE, include=TRUE}
kz<-read_dta("/Volumes/sscm alspac/Data/Current/Other/Sample Definition/kz_5c.dta") #child sample definition
kz <- kz[kz$kz001==1,] #select core sample children; n=14,676
```
  
  + in the Useful Information folder there is a folder called Multiple Pregnancies that can be used to link siblings to mothers when the mother has more than one child in the core sample:
```{r eval=FALSE, include=TRUE}
mult_pregs_core<-read_dta("/Volumes/sscm alspac/Data/Useful_data/Multiple pregs/mult_pregs in alspac_CORE.dta") #definition of multiple pregnancies within cohort
mult_pregs_core<-mult_pregs_core[-which(mult_pregs_core$aln %in% c(36891,45649)),] #remove 36891 from multiple pregnancy file as it doesn't appear in mz or kz but is linked as a sib to 45649 (who does appear in mz)
```
  
  + the multiple pregnancies and mz files can be used to indicate the order in which siblings were born (pregnancy_order_no):
```{r eval=FALSE, include=TRUE}
mult_pregs_core<-left_join(mult_pregs_core,mz,by="aln")
mult_pregs_core<-mult_pregs_core[order(mult_pregs_core$mz024b,mult_pregs_core$mz024a),]#order by year and month of delivery
mult_pregs_core$pregnancy_order_no <-1 #first child
mult_pregs_core$pregnancy_order_no[duplicated(mult_pregs_core$mult_no)]<-2 #second child
mult_pregs_core[-which(is.na(mult_pregs_core$aln3)),][3,"pregnancy_order_no"]<-3 #third child
mult_pregs_core<-mult_pregs_core[,c("aln","pregnancy_order_no","mult_no","aln2","aln3")]
mz<-full_join(mz,mult_pregs_core,by="aln")
mz$pregnancy_order_no[is.na(mz$pregnancy_order_no)]<-0 #0 for women with only one pregnancy in core sample
```
  + *note: there is no equivalent of the mz and kz files for partners. They can be matched to pregnancies by aln, but I don't think we are able to identify dads who father more than one study child (or dads that are partners to more than one study mum). It looks like this is possible at the Focus On Fathers clinic, but that was quite a long time after the start of the study and there was a fair bit of loss-to-follow-up.*
  
  + joining everything together to make a file at the child level:
```{r eval=FALSE, include=TRUE}
dat<-full_join(kz,mz,by="aln")
dat<-left_join(dat,cp[,-which(names(cp)%in%names(dat)[-which(names(dat) =="aln_qlet")])],by="aln_qlet")
```
  
  + generate helpful variables:
```{r eval=FALSE, include=TRUE}
#generate unique identifier for children that incorporates mother ID, pregnancy ID (aln) and twin ID (qlet)
dat$mother_aln_qlet <- paste0(dat$pregnancy_order_no,"_",dat$aln_qlet)
#rename mult_no to siblings_mother_id can be used to link siblings
names(dat)[names(dat)=="mult_no"]<-"siblings_mother_id"

dat$multiple_pregnancy <- dat$mz010a

```

## Examples
*Examples of working with linked family members to select specific samples (do not run as standard!)*

*If you wanted to select only singletons (i.e. not twins or triplets), you would:*
```{r eval=FALSE, include=TRUE}
singletons <- dat[dat$multiple_pregnancy==1,]
```

*To select only children from independent mothers:*
```{r eval=FALSE, include=TRUE}
independent_mothers <- dat[dat$pregnancy_order_no==0|dat$pregnancy_order_no==1,]
```

*To select and match siblings:*
```{r eval=FALSE, include=TRUE}
Sib1 <- dat[dat$pregnancy_order_no==1,] #oldest
colnames(Sib1)<-paste0(colnames(Sib1),"_sib1")
Sib2 <- dat[dat$pregnancy_order_no==2,] #middle child
colnames(Sib2)<-paste0(colnames(Sib2),"_sib2")
Sib3 <- dat[dat$pregnancy_order_no==3,] #youngest
colnames(Sib3)<-paste0(colnames(Sib3),"_sib3")
All_sibs_wide_format<- left_join(Sib1,Sib2,by=c("siblings_mother_id_sib1"="siblings_mother_id_sib2"))
All_sibs_wide_format<- left_join(All_sibs_wide_format,Sib3,by=c("siblings_mother_id_sib1"="siblings_mother_id_sib3"))
```

# Useful information from the sample files (mz and kz) {.tabset}

## Derive clean variables
There are some other useful variables in the mz and kz files that we can extract.

* Live births, miscarriages, terminations, death due to congenital malformations:
```{r eval=FALSE, include=TRUE}
#live births (could be useful outcome, or way to select sample)
dat$live_birth <-NA
dat$live_birth[dat$kz011==1]<-1
dat$live_birth[dat$kz011==2]<-0
#miscarriage or termination before 20 weeks vs live birth (could be useful outcome, or way to select sample)
dat$miscarriage_termination_b420wks <- NA
dat$miscarriage_termination_b420wks[dat$kz011==0]<-1
dat$miscarriage_termination_b420wks[dat$live_birth==1]<-0
#fetal death/stillbirth after 20 weeks vs live birth (could be useful outcome, or way to select sample)
dat$fetal_death_after_20wks <- NA
dat$fetal_death_after_20wks[dat$kz011==1]<-1
dat$fetal_death_after_20wks[dat$live_birth==1]<-0
#termination for congenital malformation (could be useful outcome, or way to select sample)
dat$congenital_malf_termination <-NA
dat$congenital_malf_termination[dat$mz011b==1]<-1
dat$congenital_malf_termination[dat$mz011b==3]<-0
dat$congenital_malf_termination[dat$mz011b==7]<-0
#death with a congenital defect up to one year (vs deaths before 1 with no congenital malformation)
dat$congenital_malf_death_before_one_year <-NA
dat$congenital_malf_death_before_one_year[dat$kz017==1] <-1
dat$congenital_malf_death_before_one_year[dat$kz017==2] <-0
```

* Anthropometry at birth:
```{r eval=FALSE, include=TRUE}
#birthweight
dat$anthro_birthweight <-dat$kz030
dat$anthro_birthweight[dat$anthro_birthweight<0]<-NA #removing missing data
dat$anthro_birthweight_zscore <-scale(dat$anthro_birthweight)
#neonatal measurements age in days
dat$covs_age_birthmeasures_days <- dat$kz028
dat$covs_age_birthmeasures_days[dat$covs_age_birthmeasures_days<0]<-NA
#head circumference in cm
dat$anthro_head_circ_birth <- dat$kz031
dat$anthro_head_circ_birth[dat$anthro_head_circ_birth<0]<-NA
dat$anthro_head_circ_birth_zscore <-scale(dat$anthro_head_circ_birth)
#crown-heel in cm
dat$anthro_crown_heel_birth <- dat$kz032
dat$anthro_crown_heel_birth[dat$anthro_crown_heel_birth<0]<-NA
dat$anthro_crown_heel_birth_zscore <-scale(dat$anthro_crown_heel_birth)
```

* Maternal age:
```{r eval=FALSE, include=TRUE}
#mother's age at conception (note that to preserve confidentiality these data have been effectively winsorised)
dat$covs_age_mother_conception <- dat$mz028a
dat$covs_age_mother_conception[dat$covs_age_mother_conception<0]<-NA
#mother's age at delivery (note that to preserve confidentiality these data have been effectively winsorised)
dat$covs_age_mother_delivery <- dat$mz028b
dat$covs_age_mother_delivery[dat$covs_age_mother_delivery<0]<-NA
```

## Remove original variables
Remove original variables we don't need anymore:
```{r eval=FALSE, include=TRUE}
dat<-dat[,-grep(names(dat),pattern="kz|mz")]
```

# Relevant ALSPAC questionnaires {.tabset}

## Mothers:

All following information is taken from the ALSPAC documentation on the R drive.

* Approximately 7 days after the brochure had been sent out, and provided we had not heard from the mother that she did not want to take part in the study, the first questionnaire was posted to her. The nature of the questionnaire depended on her gestation at enrolment.
* There were 6 questionnaires administered to the mother at the start of the study and asking questions about pregnancy:
  + **HAB** "Having a Baby"" at 18 weeks gestation
  + **YP** "Your Pregnancy" at 32 weeks gestation
  + **Env** 'Your Environment' was sent immediately after enrolment provided the mother enrolled before 14 weeks gestation. This questionnaire was designed in particular to identify those features of the early environment that might be responsible for effects on the fetus. 
  + **AY** About Yourself" was sent during pregnancy or up to 4 months after. It was mainly concerned with features that referred to the mother's past medical, social and environmental history, and consequently the timing of administration was relatively unimportant.
  + (**YHL** 'Your Home and Lifestyle') - for mothers enrolling after 23 weeks, the "Having a Baby" questionnaire was not likely to be valid. For these women, therefore, the appropriate information, otherwise obtained from the questionnaires "Your Environment"" and "Having a Baby", was combined into a single questionnaire "Your Home & Lifestyle".
  + (**FTG** 'Filling the gaps) - For a variety of reasons some mothers did not receive the questionnaire "Your Pregnancy" which includes questions on ethnic origin, education, social, and occupation levels. It also includes the questions on early sexual experience. All such questions which were not specific to the third trimester of pregnancy were therefore included in a short questionnaire entitled "Filling the Gaps", which was administered when the child was 12 months of age.
  + **MMB** 'Me and My Baby' - Sent to the mother 8 weeks after the child was born. It was accompanied by a questionnaire for the partner entitled "Being a Father". It contains questions about the health and lifestyle of mothers in the last months of pregnancy, and about the labour and delivery.
* Information from the questionnaires has been combined into the following data files:
  + A: Env + YHL
  + B: HAB + YHL
  + C: YP + FTG
  + D: AY
  + E: MMB

* The next file (F) contains information from 'Looking After the Baby', which is a questionnaire administered 8 months after the child was born and is not focused on prenatal exposures. The last maternal data file to contain relevant information for EPoCH is R, which was sent to mothers when the child was 11 years.

## Partners/Fathers:
All following information is taken from the ALSPAC documentation on the R drive.

* Questionnaires were also designed for the partner. We did not, however, have the name of the mother's partner and indeed we did not want to involve the partner without the mother's permission.
* We therefore invoked the following procedure. Questionnaires for partners were timed to coincide with a questionnaire for the mother. The pair was sent together to the mother, accompanied by a letter which explained that we were enclosing a questionnaire for her partner: if she would like him to be involved, then we would like her to hand it to him, but if she did not want him to be involved that was fine. The decision was hers. 
* We did not stipulate a particular definition of partner (i.e. male or female), and left it to the mother to make such a decision. 
* Nevertheless, information obtained from the partner identifies whether he is (or thinks he is) the biological father of the child and indeed whether he is living with the mother of the child. 
* A letter was enclosed with each partner's questionnaire and also a pre-paid envelope, so that his responses could remain confidential if he so wished. 
* If a woman wrote and informed us that she had no partner, or that he was not interested, then we did not include questionnaires for him from then on.
* There were two questionnaires administered to the partner at the start of the study and asking questions about pregnancy and the prenatal period:
  + **YYE** 'You and Your Environment', which accompanied the' maternal questionaire 'About Yourself' and was sent during pregnancy or postnatally
  + **PQ** 'Partner's questionnaire', which accompanied the maternal questionnaire 'Having a Baby' for enrolments before 24 weeks, and 'Your Home & Lifestyle' for enrolments at 24 weeks or later.
* Information from these questionnaires is stored in the following data files:

  + PA: YYE
  + PB: PQ

* The next questionnaire 'Being a father' (file PC) accompanied the maternal questionnaire 'Me and My Baby' but did not ask questions about prenatal exposures. The final paternal data file relevant to EPoCH is PP, administered when the child was 11.

## Offspring outcomes

Offspring outcomes will be in the following files:

* Child-based (parent report): ka|kb|kc|kd|ke|kf|kg|kh|ki|kj|kk|kl|km|kn|ko|kp|kq|kr|ks|kt|ku|kv|kw
* Child completed: ccs|ccb|ccc|ccd|cce|ccf|ccg|cch|cci|ccj|cck
* Child clinic: f07|f08|f09|f10|f11|cif
* Mother: a|b|c|d|e|f|g|h|i|j|k|l|m|n|o|p|q|r 
* Partner: pa|pb|pc|pd|pe|pf|pg|ph|pi|pj|pk|pl|pm|pn|po|pp
* Obstetric: OA|OB|OC

# The ALSPAC R package {.tabset}
This R package was curated by Gib Hemani and the documentation can be found here: https://github.com/explodecomputer/alspac

The benefit of using this package is that the withdrawals of consent are done automatically. 

It is regularly updated, so may need to be re-installed periodically.
```{r eval=FALSE, include=TRUE}
#install.packages("devtools")
#library(devtools)
#install_github("explodecomputer/alspac")
```

Loading the package and variables:
```{r eval=FALSE, include=TRUE}
library(alspac)

setDataDir("/Volumes/sscm alspac/Data/") #telling the package where the data is stored (make sure you are connected to the ALSPAC server before running this)

current <- createDictionary("Current", name="current")
useful <- createDictionary("Useful_data", name="useful")

data(current)
data(useful)
```

# Time of questionnaire completion {.tabset}

## Relevant variables
Some questions refer to time relative to when the questionnaire was completed (e.g. "have you smoked in the past two weeks?"). Therefore, when defining the timing of an exposure, it's useful to know when the questionnaire was completed relative to pregnancy/birth (e.g. what trimester). For mothers and fathers, the prenatal/birth questionnaires contain information on the gestational age of the child at questionnaire completion and/or the questionnaire completion time in weeks after the birth of the child.

* a902, b924, c991, d990, pa900, pb900 = gestational age of child at completion
* e699, pa901, pb901 = weeks since delivery at completion

## Set up meta-data
Start by creating a dataframe of "meta data" describing each variable we might be interested in:
```{r eval=FALSE, include=TRUE}
time_vars <- c("a902","b924","c991","d990","e699","pa900","pa901","pb900","pb901")
time_meta_data <- findVars(time_vars) #create a table with all the necessary meta data for each variable
time_meta_data <- subset(time_meta_data,subset=tolower(name) %in% time_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
Then extract the variables and merge with the main dataframe:
```{r eval=FALSE, include=TRUE}
time_data <- extractVars(time_meta_data) #extracting individual level data and performing withdrawal of consent
time_data<-time_data[,c("aln","qlet",time_vars)]
dat <- left_join(dat,time_data,by=c("aln","qlet")) #merge with main dataframe (any people who have withdrawn consent will not have time_data, so they will be given NA in dat for all variables extracted using the ALSPAC package, and therefore not included in our analysis)
```

## Derive clean variables
Then code the data in an intuitive/memorable way while setting missing values to NA:
```{r eval=FALSE, include=TRUE}
dat$time_qA_mother_gestation<-dat$a902
dat$time_qA_mother_gestation[dat$time_qA_mother_gestation<0] <- NA

dat$time_qB_mother_gestation<-dat$b924
dat$time_qB_mother_gestation[dat$time_qB_mother_gestation<0] <- NA

dat$time_qC_mother_gestation<-dat$c991
dat$time_qC_mother_gestation[dat$time_qC_mother_gestation<0] <- NA

dat$time_qD_mother_gestation<-dat$d990
dat$time_qD_mother_gestation[dat$time_qD_mother_gestation<0] <- NA

dat$time_qPA_partner_gestation<-dat$pa900
dat$time_qPA_partner_gestation[dat$time_qPA_partner_gestation<0] <- NA

dat$time_qPB_partner_gestation<-dat$pb900
dat$time_qPB_partner_gestation[dat$time_qPB_partner_gestation<0] <- NA

dat$time_qE_mother_sincebirth<-dat$e699
dat$time_qE_mother_sincebirth[dat$time_qE_mother_sincebirth<0] <- NA

dat$time_qPA_partner_sincebirth<-dat$pa901
dat$time_qPA_partner_sincebirth[dat$time_qPA_partner_sincebirth<0] <- NA

dat$time_qPB_partner_sincebirth<-dat$pa901
dat$time_qPB_partner_sincebirth[dat$time_qPB_partner_sincebirth<0] <- NA
```

## Remove original variables
Finally, remove the original variables:
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(names(dat) %in% time_vars)]
```

# Partner's status {.tabset}

## Relevant variables

### Biological father
There are 59 partners that the mothers say are NOT the biological fathers, and 15 where the mother is unsure. For 364 pregnancies there is "no partner". There are 13090 biological fathers. Given these numbers, and the complexities introduced by comparing biological/non-biological partners, let's just study biological fathers, which make up the vast majority of cases.

* a521 = partner is father of unborn child (1=Yes, 2=No, 3=unsure, -2 = no partner)

### Father living with mother prenatally
Whether or not the father is living with the mother prenatally will be important when assessing the father's influence on the mother, and also on the child

* a522 = partner lives with mum (1=Yes,2=No)

### Parents living with child postnatally
Whether or not the mother/father is living with the child postnatally will be important for assessing the parents' influence on the child postnatally (although this shouldn't be a true confounder). I have found the following variables that look relevant: j374, n8050, n8060, q3050, q3060, pl8050, pl8060. I'm not sure what to do about these as they're all quite late with low-ish numbers, so have left out of the dataset for now.

### Dad returned either pregnancy questionnaire (paternal participation)

```{r eval=FALSE, include=TRUE}
require(readstata13)
A<-read.dta13("/Volumes/sscm alspac/Data/Current/Quest/Partner/pa_4a.dta")
B<-read.dta13("/Volumes/sscm alspac/Data/Current/Quest/Partner/pb_4b.dta")
C<-read.dta13("/Volumes/sscm alspac/Data/Current/Quest/Partner/pc_3a.dta")

A[A=="-1"]<-NA #changing -1 to missing
B[B=="-1"]<-NA #changing -1 to missing
C[C=="-1"]<-NA #changing -1 to missing

dat$paternal_participation <- ifelse(dat$aln %in% unique(c(A$aln,B$aln,C$aln)),1,0)
rm(A,B,C)

```

## Set up meta-data
```{r eval=FALSE, include=TRUE}
parent_status_vars <- c("a521","a522")

parent_status_meta_data <- findVars(parent_status_vars) #create a table with all the necessary meta data for each variable
parent_status_meta_data <- subset(parent_status_meta_data,subset=tolower(name) %in% parent_status_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
```{r eval=FALSE, include=TRUE}
parent_status_data <- extractVars(parent_status_meta_data) #extracting individual level data and performing withdrawal of consent
parent_status_data<-parent_status_data[,c("aln","qlet",parent_status_vars)]

dat <- left_join(dat,parent_status_data,by=c("aln","qlet")) #merge with main dataframe
```

## Derive clean variables
```{r eval=FALSE, include=TRUE}
dat$covs_biological_father<-dat$a521
dat$covs_biological_father[dat$covs_biological_father==1]<-"bio_dad"
dat$covs_biological_father[dat$covs_biological_father==(-2)]<-"no_partner"
dat$covs_biological_father[dat$covs_biological_father==(-1)]<-"missing"
dat$covs_biological_father[dat$covs_biological_father==(2)]<-"not_bio_dad"
dat$covs_biological_father[dat$covs_biological_father==(3)]<-"unsure"

dat$covs_partner_lives_with_mother_prenatal<-NA
dat$covs_partner_lives_with_mother_prenatal[dat$a522==1]<-1
dat$covs_partner_lives_with_mother_prenatal[dat$a522==2]<-0
```

## Remove original variables
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(names(dat) %in% parent_status_vars)]
```

# Parents' smoking {.tabset}
## Ideal variables
Ideal harmonised smoking exposures for both parents are:

*	Any time up to birth: ever vs never smoker: binary variable
*	Early-onset: Smoking before age 11 vs no smoking before age 11
*	Pre-conception: Any vs no smoking in the three months prior to pregnancy
*	Pre-conception: Number of cigarettes per day in the three months prior to pregnancy (ordered categorical)
*	Pregnancy: Any smoking vs no smoking at any time during pregnancy
*	Pregnancy: Any smoking vs no smoking in the first trimester
*	Pregnancy: Any smoking vs no smoking in the second trimester
*	Pregnancy: Any smoking vs no smoking in the third trimester
*	Pregnancy: Number of cigarettes per day in the first trimester (ordered categorical)
*	Pregnancy: Number of cigarettes per day in the second trimester (ordered categorical)
*	Pregnancy: Number of cigarettes per day in the third trimester (ordered categorical)
* Passive smoke during pregnancy: binary variable
* Postnatal: ordered categorical up until age 2 (in line with the definition of the first 1000 days)
* Postnatal: any vs no smoking up to age 2

Smoking is defined as smoking tobacco in cigarettes, pipes, cigars or unspecified. In ALSPAC some questions refer to smoking cannabis, but these have not been used to define the smoking variables due to ambiguity around whether/how much tobacco was included.

## Set up meta-data
To get a complete list of relevant variables, we can use the alspac package to search the variable catalogue. So we start with a long list of variables in the meta_data object and then reduce it down to the most relevant ones.

Select variables containing key words and completed by mothers or partners:
```{r eval=FALSE, include=TRUE}
#get a list of smoking questions asked to parents:
parent_smoking_meta_data <- findVars("smoking","smoke","cigarette","tobacco")
parent_smoking_meta_data_mother <-parent_smoking_meta_data[parent_smoking_meta_data$cat3=="Mother",]
parent_smoking_meta_data_father <-parent_smoking_meta_data[parent_smoking_meta_data$cat3=="Partner",]
```

Questions regarding smoking before or during pregnancy (exposures) are going to be in the early questionnaires described above: A, B, C, D and E (PA and PB for dads). Postnatal smoking will be up to questionnaire R for mothers and PP for partners.
```{r eval=FALSE, include=TRUE}
#select questions asked in questionnaires A-R or PA to PP:
parent_smoking_meta_data_father <-parent_smoking_meta_data_father[grep(lapply(parent_smoking_meta_data_father$obj,substring,1,2),pattern="pa|pb|pc|pd|pe|pf|pg|ph|pi|pj|pk|pl|pm|pn|po|pp"),]

parent_smoking_meta_data_mother <-parent_smoking_meta_data_mother[grep(lapply(parent_smoking_meta_data_mother$obj,substring,1,1),pattern="a|b|c|d|e|f|g|h|i|j|k|l|m|n|o|p|q|r"),]
```

At this point, it's necessary to go through the table (parent_smoking_meta_data) manually, referring to the ALSPAC documentation where necessary, to decide which variables to use and for what.

### Maternal
**Maternal smoking any time up to birth: ever vs never smoker (binary variable)**

* The main variable to define this is b650 (Ever smoked)
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c("b650")
```

**Maternal early-onset: Smoking before age 11 vs no smoking before age 11**

* b651	Age started REG smoking
* b652	Age started REG smoking
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("b651","b652"))
```

**Maternal pre-conception: Any vs no smoking in the three months prior to pregnancy**

* b660	Time in years given up smoking
* b663	Tobacco smoked REG PRE PREG
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("b660","b663"))
```

**Maternal pre-conception: Number of cigarettes per day in the three months prior to pregnancy (ordered categorical)**

* b669	NO smoked per day just PRE PREG
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("b669"))
```

**Maternal pregnancy: Any smoking vs no smoking in the first trimester**

* b665	Tobacco smoked in 1ST 3MTHS of PREG
* b667	Tobacco smoked in last 2WKS
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("b665","b667"))
```

**Maternal pregnancy: Any smoking vs no smoking in the second trimester**

* c482	CIGS smoked per day
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("c482"))
```

**Maternal pregnancy: Any smoking vs no smoking in the third trimester**

* e178	NO smoked daily in last 2MTHS of PREG
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("e178"))
```

**Maternal pregnancy: Any smoking vs no smoking at any time during pregnancy**

* can be derived from the trimester specific smoking variables

**Maternal pregnancy: Number of cigarettes per day in the first trimester (ordered categorical)**

* b670	NO smoked per day in 1ST 3MTHS of PREG
* b671	NO smoked per day in last 2WKS
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("b670","b671"))
```

**Maternal pregnancy: Number of cigarettes per day in the second trimester (ordered categorical)**

* c482 CIGS smoked per day (already in list)

**Maternal pregnancy: Number of cigarettes per day in the third trimester (ordered categorical)**

* e178	NO smoked daily in last 2MTHS of PREG (already in list)

**Maternal passive smoke during pregnancy: binary variable**

* c481a	Passive Smoke Exposure
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("b695","c481a"))
```

**Maternal postnatal**

* f620, g820
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("f620", "g820"))
```

### Paternal
**Paternal any time up to birth: ever vs never smoker: binary variable**

* b683	PTNR smokes
* b685	NO of times per day PTNR smokes
* b690	Type PTNR smokes
* pb071 Ever been a smoker
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("b683","b685","b690","pb071"))
```

**Paternal early-onset: Smoking before age 11 vs no smoking before age 11**

* pb072 Age started smoking REG
* b691	Age PTNR started smoking
* b692	Age PTNR started smoking
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("pb072","b691","b692"))
```

**Paternal pre-conception: Any vs no smoking in the three months prior to pregnancy**

* pb075 Years stopped smoking
* pb076 Months stopped smoking
* pb077 Smoked regularly in last 9 months
* pb078 Times smoked per day at start of pregnancy
* pb079 Times smoked per day in last two weeks
* b683	PTNR smokes (already in list)
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("pb077","pb078","pb079","pb075","pb076"))
```

**Paternal pre-conception: Number of cigarettes per day in the three months prior to pregnancy (ordered categorical)**

* b685	NO of times per day PTNR smokes (already in list)

**Paternal pregnancy: Any smoking vs no smoking in the first trimester**

* pb078 Times smoked per day at start of pregnancy (already added to list)

**Paternal pregnancy: Any smoking vs no smoking in the second trimester**

* pb078 Times smoked per day at start of pregnancy (already added to list)

**Paternal pregnancy: Any smoking vs no smoking in the third trimester**

* e185	NO PTNR smoked in last 2MTHS of PREG
* e186	NO PTNR smoked in past 2WKS
* pc250 CIGS smoked in last two months of pregnancy
* pc260 NO smoked in last two months of pregnancy
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("e185", "e186","pc250","pc260"))
```

**Paternal pregnancy: Any smoking vs no smoking at any time during pregnancy**

* can be derived from the trimester specific smoking variables

**Paternal pregnancy: Number of cigarettes per day in the first trimester (ordered categorical)**

* pb078 Times smoked per day at start of pregnancy (already added to list)

**Paternal pregnancy: Number of cigarettes per day in the second trimester (ordered categorical)**

* might be possible to derive

**Paternal pregnancy: Number of cigarettes per day in the third trimester (ordered categorical)**

* pc250, pc260, e185 and e186	already added to the list

**Paternal passive smoke during pregnancy: binary variable and ordered categorical**

* Not available

**Paternal postnatal**

* pc251, pd620, pe450, ph6180, pj5050, pj5051, pl5010, pp6020
* f560, g648, g649, h525, j630, l6070, l6071, n5040, n5042, p3070, p3071, r6040
```{r eval=FALSE, include=TRUE}
parent_smoking_vars <-c(parent_smoking_vars,c("pc251", "pd620", "pe450", "ph6180", "pj5050", "pj5051", "pl5010", "pp6020"))
```

### Produce final meta-data object
```{r eval=FALSE, include=TRUE}
parent_smoking_meta_data <- findVars(unique(parent_smoking_vars)) #create a table with all the necessary meta data for each variable

parent_smoking_meta_data <- subset(parent_smoking_meta_data,subset=tolower(name) %in% parent_smoking_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
```{r eval=FALSE, include=TRUE}
parent_smoking_data <- extractVars(parent_smoking_meta_data[parent_smoking_meta_data$name %in% unique(parent_smoking_vars),]) #extracting individual level data and performing withdrawal of consent

parent_smoking_data<-parent_smoking_data[,c("aln","qlet",parent_smoking_vars)]
dat <- left_join(dat,parent_smoking_data,by=c("aln","qlet")) 
```

## Derive clean variables

### Maternal
Any time up to birth: ever vs never smoker: binary variable
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_ever_life_binary <- NA
dat$smoking_mother_ever_life_binary[dat$b650==1] <- 1
dat$smoking_mother_ever_life_binary[dat$b650==2] <- 0
```

Early-onset: Smoking before or including age 11 vs no smoking before age 11
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_before11_binary <- NA
dat$smoking_mother_before11_binary[dat$b651<=11 & dat$b651>0] <- 1
dat$smoking_mother_before11_binary[dat$b651==(-2)|dat$b651>11] <- 0 #-2 is never smoked
```

Pre-conception: Any vs no smoking in the three months prior to pregnancy
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_preconception_binary <- NA
dat$smoking_mother_preconception_binary[dat$b663==1]<-0
dat$smoking_mother_preconception_binary[dat$b663%in%c(2,3,4,5)]<-1
```

Pre-conception: Number of cigarettes per day in the three months prior to pregnancy (ordered categorical)

* lifecyle have defined using none, <10 and >10. Looking at the variable catelogue, it looks like we can split into more categories for our cohorts (none, light, moderate, heavy)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_preconception_ordinal <- NA
dat$smoking_mother_preconception_ordinal[dat$b669==0]<-"None"
dat$smoking_mother_preconception_ordinal[dat$b669%in%c(1,5)]<-"Light" #i.e. <10 but not 0
dat$smoking_mother_preconception_ordinal[dat$b669==10 | dat$b669==15]<-"Moderate" #i.e. <20, >=10
dat$smoking_mother_preconception_ordinal[dat$b669>=20]<-"Heavy" #i.e.>=20
dat$smoking_mother_preconception_ordinal<-factor(dat$smoking_mother_preconception_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: N cigarettes in the first trimester (up to end of 12 weeks)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_firsttrim_ordinal <- NA
#b670 asks for number smoked in first trimester
dat$smoking_mother_firsttrim_ordinal[dat$b670==0]<-"None"
dat$smoking_mother_firsttrim_ordinal[dat$b670%in%c(1,5)]<-"Light" #i.e. <5 but not 0
dat$smoking_mother_firsttrim_ordinal[dat$b670==10 | dat$b670==15]<-"Moderate" #i.e. <20, >=10
dat$smoking_mother_firsttrim_ordinal[dat$b670>=20]<-"Heavy" #i.e.>=20
#b671 asks for number smoked in last two weeks, which could be first trimester depending on when she filled out the questionnaire
dat$smoking_mother_firsttrim_ordinal[which(dat$b671==1&dat$time_qB_mother_gestation<=12&is.na(dat$smoking_mother_firsttrim_ordinal))]<-"Light"
dat$smoking_mother_firsttrim_ordinal[which(dat$b671==5&dat$time_qB_mother_gestation<=12&is.na(dat$smoking_mother_firsttrim_ordinal))]<-"Light"
dat$smoking_mother_firsttrim_ordinal[which((dat$b671==10|dat$b671==15)&dat$time_qB_mother_gestation<=12&is.na(dat$smoking_mother_firsttrim_ordinal))]<-"Moderate"
dat$smoking_mother_firsttrim_ordinal[which(dat$b671>=20&dat$time_qB_mother_gestation<=12&is.na(dat$smoking_mother_firsttrim_ordinal))]<-"Heavy"

dat$smoking_mother_firsttrim_ordinal<-factor(dat$smoking_mother_firsttrim_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking in the first trimester (up to end of 12 weeks)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_firsttrim_binary <- NA
dat$smoking_mother_firsttrim_binary[dat$smoking_mother_firsttrim_ordinal=="None"]<-0
dat$smoking_mother_firsttrim_binary[dat$smoking_mother_firsttrim_ordinal%in% c("Light","Moderate","Heavy")]<-1
```

Pregnancy: N cigarettes in the second trimester (week 13 to end of 26)
* we can use 2 variables, b671 has most data, so we'll start with that and then fill in missing values with c482
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_secondtrim_ordinal <- NA
#b671 asks if mother smoked in last two weeks, which could be second trimester depending on when she answered the question
dat$smoking_mother_secondtrim_ordinal[which(dat$b671==0&dat$time_qB_mother_gestation%in%13:26)]<-"None"
dat$smoking_mother_secondtrim_ordinal[which(dat$b671==1&dat$time_qB_mother_gestation%in%13:26)]<-"Light"
dat$smoking_mother_secondtrim_ordinal[which(dat$b671==5&dat$time_qB_mother_gestation%in%13:26)]<-"Light"
dat$smoking_mother_secondtrim_ordinal[which((dat$b671==10|dat$b671==15)&dat$time_qB_mother_gestation%in%13:26)]<-"Moderate"
dat$smoking_mother_secondtrim_ordinal[which(dat$b671>=20&dat$time_qB_mother_gestation%in%13:26)]<-"Heavy"

#c482 asks for number smoked "at the moment", which could be second trimester
dat$smoking_mother_secondtrim_ordinal[which(dat$c482==0&dat$time_qC_mother_gestation%in%13:26&is.na(dat$smoking_mother_secondtrim_ordinal))]<-"None"
dat$smoking_mother_secondtrim_ordinal[which(dat$c482%in%1:4&dat$time_qC_mother_gestation%in%13:26&is.na(dat$smoking_mother_secondtrim_ordinal))]<-"Light"
dat$smoking_mother_secondtrim_ordinal[which(dat$c482%in%5:9&dat$time_qC_mother_gestation%in%13:26&is.na(dat$smoking_mother_secondtrim_ordinal))]<-"Light"
dat$smoking_mother_secondtrim_ordinal[which(dat$c482%in%10:19&dat$time_qC_mother_gestation%in%13:26&is.na(dat$smoking_mother_secondtrim_ordinal))]<-"Moderate"
dat$smoking_mother_secondtrim_ordinal[which(dat$c482>=20&dat$time_qC_mother_gestation%in%13:26&is.na(dat$smoking_mother_secondtrim_ordinal))]<-"Heavy"


dat$smoking_mother_secondtrim_ordinal<-factor(dat$smoking_mother_secondtrim_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking in the second trimester (week 13-end 26)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_secondtrim_binary <- NA
dat$smoking_mother_secondtrim_binary[dat$smoking_mother_secondtrim_ordinal=="None"]<-0
dat$smoking_mother_secondtrim_binary[dat$smoking_mother_secondtrim_ordinal%in% c("Light","Moderate","Heavy")]<-1
```

Pregnancy: N cigarettes per day in the third trimester (week 27 onwards)

* we can use three variables, e178 is the most straightforward, so we'll start with that and then fill in the missing values using the other variables where possible
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_thirdtrim_ordinal <- NA
#e178 asks how many times per day did they smoked in the last two months of pregnancy
dat$smoking_mother_thirdtrim_ordinal[dat$e178==0]<-"None"
dat$smoking_mother_thirdtrim_ordinal[dat$e178==1]<-"Light" #i.e. <5 but not 0
dat$smoking_mother_thirdtrim_ordinal[dat$e178==5]<-"Light" #i.e. <10, >=5
dat$smoking_mother_thirdtrim_ordinal[dat$e178==10 | dat$e178==15]<-"Moderate" #i.e. <20, >=10
dat$smoking_mother_thirdtrim_ordinal[dat$e178>=20]<-"Heavy" #i.e.>=20

#c482 asks for number smoked "at the moment", which could be third trimester
dat$smoking_mother_thirdtrim_ordinal[which(dat$c482==0&dat$time_qC_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"None"
dat$smoking_mother_thirdtrim_ordinal[which(dat$c482%in%1:4&dat$time_qC_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Light"
dat$smoking_mother_thirdtrim_ordinal[which(dat$c482%in%5:9&dat$time_qC_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Light"
dat$smoking_mother_thirdtrim_ordinal[which(dat$c482%in%10:19&dat$time_qC_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Moderate"
dat$smoking_mother_thirdtrim_ordinal[which(dat$c482>=20&dat$time_qC_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Heavy"

#b671 asks how much the mother smoked in last two weeks, which could be third trimester depending on when she filled out the questionnaire
dat$smoking_mother_thirdtrim_ordinal[which(dat$b671==1&dat$time_qB_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Light"
dat$smoking_mother_thirdtrim_ordinal[which(dat$b671==5&dat$time_qB_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Light"
dat$smoking_mother_thirdtrim_ordinal[which((dat$b671==10|dat$b671==15)&dat$time_qB_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Moderate"
dat$smoking_mother_thirdtrim_ordinal[which(dat$b671>=20&dat$time_qB_mother_gestation>=27&is.na(dat$smoking_mother_thirdtrim_ordinal))]<-"Heavy"

dat$smoking_mother_thirdtrim_ordinal<-factor(dat$smoking_mother_thirdtrim_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking in the third trimester (27 weeks onward)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_thirdtrim_binary <- NA
dat$smoking_mother_thirdtrim_binary[dat$smoking_mother_thirdtrim_ordinal=="None"]<-0
dat$smoking_mother_thirdtrim_binary[dat$smoking_mother_thirdtrim_ordinal%in% c("Light","Moderate","Heavy")]<-1
```

Pregnancy: Any smoking vs no smoking at any time during pregnancy
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_ever_pregnancy_binary<-NA
dat$smoking_mother_ever_pregnancy_binary[dat$smoking_mother_firsttrim_binary==0 & dat$smoking_mother_secondtrim_binary==0 &dat$smoking_mother_thirdtrim_binary==0] <- 0
dat$smoking_mother_ever_pregnancy_binary[dat$smoking_mother_firsttrim_binary==1 | dat$smoking_mother_secondtrim_binary==1 | dat$smoking_mother_thirdtrim_binary==1] <- 1
```

Passive smoke during pregnancy: binary variable
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_passive_binary<-NA
dat$smoking_mother_passive_binary[dat$c481a==1]<-0
dat$smoking_mother_passive_binary[dat$c481a>1]<-1
```

Postnatal: ordinal smoking after birth of child, up to 2 years (first 1000 days)
```{r eval=FALSE, include=TRUE}
#smoking at 8 months = f620; 0=none, 1=light, 5=light to moderate, 10|15=moderate to heavy, 20+=heavy
#smoking at 21 months = g820; 0=none, 1=light, 5=light to moderate, 10|15=moderate to heavy, 20+=heavy
temp_var_8<-NA
temp_var_8[dat$f620==0]<-0
temp_var_8[dat$f620==1]<-1
temp_var_8[dat$f620==5]<-1
temp_var_8[dat$f620==10|dat$f620==15]<-2
temp_var_8[dat$f620>=20]<-3

temp_var_21<-NA
temp_var_21[dat$g820==0]<-0
temp_var_21[dat$g820==1]<-1
temp_var_21[dat$g820==5]<-1
temp_var_21[dat$g820==10|dat$g820==15]<-2
temp_var_21[dat$g820>=20]<-3

temp_df<-data.frame(temp_var_8,temp_var_21)
temp_df$temp_var_218<-apply(temp_df,1,max,na.rm=T)
temp_df$temp_var_218[is.infinite(temp_df$temp_var_218)]<-NA
dat$smoking_mother_postnatal_ordinal<-temp_df$temp_var_218

dat$smoking_mother_postnatal_ordinal[dat$smoking_mother_postnatal_ordinal==0]<-"None"
dat$smoking_mother_postnatal_ordinal[dat$smoking_mother_postnatal_ordinal==1]<-"Light"
dat$smoking_mother_postnatal_ordinal[dat$smoking_mother_postnatal_ordinal==2]<-"Moderate"
dat$smoking_mother_postnatal_ordinal[dat$smoking_mother_postnatal_ordinal==3]<-"Heavy"

dat$smoking_mother_postnatal_ordinal<-factor(dat$smoking_mother_postnatal_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)

rm(temp_df,temp_var_21,temp_var_8)

```

Postnatal: any vs no smoking after birth of child, up to 2 years (first 1000 days)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_postnatal_binary <- NA
dat$smoking_mother_postnatal_binary[dat$smoking_mother_postnatal_ordinal=="None"]<-0
dat$smoking_mother_postnatal_binary[dat$smoking_mother_postnatal_ordinal%in% c("Light","Moderate","Heavy")]<-1
```

### Paternal
Any time up to birth: ever vs never smoker: binary variable

* self report, but there were a small number (33) where mother said in b683 that father is current smoker, but partner has reported they are a never smoker... Not going to do anything about this because the number is small and the mother may have been answering for a different partner(?), and we don't know the error rate (and can't do anything about the error) where fathers have said they are ever smokers and mothers have said they are not current smokers.
```{r eval=FALSE, include=TRUE}
dat$smoking_father_ever_life_binary <- NA
dat$smoking_father_ever_life_binary[dat$pb071==1]<-1
dat$smoking_father_ever_life_binary[dat$pb071==2]<-0
```

Early-onset: Smoking before or including age 11 vs no smoking before age 11

* concordance between maternal report and paternal report is ok, but there are some discrepancies and the paternal reported data is more complete, so stick with this.
```{r eval=FALSE, include=TRUE}
#self-report
dat$smoking_father_before11_binary <- NA
dat$smoking_father_before11_binary[dat$pb072<=11 & dat$pb072>0] <- 1
dat$smoking_father_before11_binary[dat$pb072>11] <- 0
dat$smoking_father_before11_binary[dat$smoking_father_ever_life_binary==0] <-0 #never smoker

#maternal-report
dat$smoking_father_before11_binary_mreport <-NA
dat$smoking_father_before11_binary_mreport[dat$b691<=11 & dat$b691>0] <- 1
dat$smoking_father_before11_binary_mreport[dat$b691>11] <- 0
dat$smoking_father_before11_binary_mreport[dat$smoking_father_ever_life_binary==0] <-0 #never smoker
```
Pre-conception/first trimester: Any vs no smoking in the three months prior to pregnancy or first trimester

* There were a few options here: pb077 asked if they smoked regularly in the last 9 months, but depending on when they filled out the questionnaire this could have been post-conception. pb075 and pb076 asks years and months since giving up smoking, but when I tried to derive the variable the n was very low. b683 is maternal report of partner current smoking, but not attached to a date. pb078 is times smoked per day at start of pregnancy, which is likely to be the same as preconception, so I think go with that.
* It wasn't possible to distinguish between pre-conception and first trimester
```{r eval=FALSE, include=TRUE}
dat$smoking_father_startpregnancy_binary <- NA
dat$smoking_father_startpregnancy_binary[dat$pb078==0]<-0
dat$smoking_father_startpregnancy_binary[dat$pb078 %in% 1:30]<-1

dat$smoking_father_startpregnancy_binary_mreport <-NA
dat$smoking_father_startpregnancy_binary_mreport[dat$b683==1]<-0
dat$smoking_father_startpregnancy_binary_mreport[dat$b683>1]<-1
```

Pre-conception/first trimester: Number of cigarettes per day (ordered categorical)
```{r eval=FALSE, include=TRUE}
dat$smoking_father_startpregnancy_ordinal <- NA
dat$smoking_father_startpregnancy_ordinal[dat$pb078==0]<-"None"
dat$smoking_father_startpregnancy_ordinal[dat$pb078 == 1]<-"Light"
dat$smoking_father_startpregnancy_ordinal[dat$pb078 == 5]<-"Light"
dat$smoking_father_startpregnancy_ordinal[dat$pb078 == 10 | dat$pb078 == 15]<-"Moderate"
dat$smoking_father_startpregnancy_ordinal[dat$pb078 >= 20]<-"Heavy"

dat$smoking_father_startpregnancy_ordinal<-factor(dat$smoking_father_startpregnancy_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)

dat$smoking_father_startpregnancy_ordinal_mreport <- NA
dat$smoking_father_startpregnancy_ordinal_mreport[dat$b685==0]<-"None"
dat$smoking_father_startpregnancy_ordinal_mreport[dat$b685 == 1|dat$b685==5]<-"Light"
dat$smoking_father_startpregnancy_ordinal_mreport[dat$b685 == 5]<-"Light"
dat$smoking_father_startpregnancy_ordinal_mreport[dat$b685 == 10 | dat$b685 == 15]<-"Moderate"
dat$smoking_father_startpregnancy_ordinal_mreport[dat$b685 >= 20]<-"Heavy"

dat$smoking_father_startpregnancy_ordinal_mreport<-factor(dat$smoking_father_startpregnancy_ordinal_mreport,levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Number of cigarettes per day in the second trimester (ordered categorical)

* pb079 asks how much smoking per day in past 2 weeks, and was mostly answered during the second trimester
```{r eval=FALSE, include=TRUE}
dat$smoking_father_secondtrim_ordinal <-NA
dat$smoking_father_secondtrim_ordinal[dat$pb079==0 & dat$time_qPB_partner_gestation %in% 13:26] <-"None"
dat$smoking_father_secondtrim_ordinal[dat$pb079==1 & dat$time_qPB_partner_gestation %in% 13:26] <-"Light"
dat$smoking_father_secondtrim_ordinal[dat$pb079==5 & dat$time_qPB_partner_gestation %in% 13:26] <-"Light"
dat$smoking_father_secondtrim_ordinal[dat$pb079%in% 10:15 & dat$time_qPB_partner_gestation %in% 13:26] <-"Moderate"
dat$smoking_father_secondtrim_ordinal[dat$pb079>=20 & dat$time_qPB_partner_gestation %in% 13:26] <-"Heavy"

dat$smoking_father_secondtrim_ordinal<-factor(dat$smoking_father_secondtrim_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking in the second trimester
```{r eval=FALSE, include=TRUE}
dat$smoking_father_secondtrim_binary <- NA
dat$smoking_father_secondtrim_binary[dat$smoking_father_secondtrim_ordinal=="None"]<-0
dat$smoking_father_secondtrim_binary[dat$smoking_father_secondtrim_ordinal %in% c("Light","Moderate","Heavy")]<-1
```

Pregnancy: Number of cigarettes per day in the third trimester (ordered categorical)

* pc260 asks how many times per day did you smoke in the last 2 months of pregnancy
```{r eval=FALSE, include=TRUE}
dat$smoking_father_thirdtrim_ordinal <- NA
dat$smoking_father_thirdtrim_ordinal[dat$pc260==0]<-"None"
dat$smoking_father_thirdtrim_ordinal[dat$pc260 == 1]<-"Light"
dat$smoking_father_thirdtrim_ordinal[dat$pc260 == 5]<-"Light"
dat$smoking_father_thirdtrim_ordinal[dat$pc260 == 10 | dat$pc260 == 15]<-"Moderate"
dat$smoking_father_thirdtrim_ordinal[dat$pc260 >= 20]<-"Heavy"

dat$smoking_father_thirdtrim_ordinal<-factor(dat$smoking_father_thirdtrim_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking in the third trimester
```{r eval=FALSE, include=TRUE}
dat$smoking_father_thirdtrim_binary <- NA
dat$smoking_father_thirdtrim_binary[dat$smoking_father_thirdtrim_ordinal=="None"]<-0
dat$smoking_father_thirdtrim_binary[dat$smoking_father_thirdtrim_ordinal %in% c("Light","Moderate","Heavy")]<-1
```

Pregnancy: Any smoking vs no smoking at any time during pregnancy
```{r eval=FALSE, include=TRUE}
dat$smoking_father_ever_pregnancy_binary<-NA
dat$smoking_father_ever_pregnancy_binary[dat$smoking_father_startpregnancy_binary==0 & 
                                    dat$smoking_father_secondtrim_binary==0 &
                                    dat$smoking_father_thirdtrim_binary==0] <- 0
dat$smoking_father_ever_pregnancy_binary[dat$smoking_father_startpregnancy_binary==1 | 
                                    dat$smoking_father_secondtrim_binary==1 |
                                    dat$smoking_father_thirdtrim_binary==1] <- 1
```

Passive smoke during pregnancy: NOT AVAILABLE

Postnatal: ordinal smoking after birth of child, up to 2 years (first 1000 days)
```{r eval=FALSE, include=TRUE}
#smoking at 8 months = pd620; 0=none, 1=light, 5=light to moderate, 10|15=moderate to heavy, 20+=heavy
#smoking at 21 months = pe450; 0=none, 1=light, 5=light to moderate, 10|15=moderate to heavy, 20+=heavy
temp_var_8<-NA
temp_var_8[dat$pd620==0]<-0
temp_var_8[dat$pd620==1]<-1
temp_var_8[dat$pd620==5]<-1
temp_var_8[dat$pd620==10|dat$pd620==15]<-2
temp_var_8[dat$pd620>=20]<-3

temp_var_21<-NA
temp_var_21[dat$pe450==0]<-0
temp_var_21[dat$pe450==1]<-1
temp_var_21[dat$pe450==5]<-1
temp_var_21[dat$pe450==10|dat$pe450==15]<-2
temp_var_21[dat$pe450>=20]<-3

temp_df<-data.frame(temp_var_8,temp_var_21)
temp_df$temp_var_218<-apply(temp_df,1,max,na.rm=T)
temp_df$temp_var_218[is.infinite(temp_df$temp_var_218)]<-NA
dat$smoking_father_postnatal_ordinal<-temp_df$temp_var_218

dat$smoking_father_postnatal_ordinal[dat$smoking_father_postnatal_ordinal==0]<-"None"
dat$smoking_father_postnatal_ordinal[dat$smoking_father_postnatal_ordinal==1]<-"Light"
dat$smoking_father_postnatal_ordinal[dat$smoking_father_postnatal_ordinal==2]<-"Moderate"
dat$smoking_father_postnatal_ordinal[dat$smoking_father_postnatal_ordinal==3]<-"Heavy"

dat$smoking_father_postnatal_ordinal<-factor(dat$smoking_father_postnatal_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)

rm(temp_df,temp_var_21,temp_var_8)
```

Postnatal: any vs no smoking after birth of child, up to 2 years (first 1000 days)
```{r eval=FALSE, include=TRUE}
dat$smoking_father_postnatal_binary <- NA
dat$smoking_father_postnatal_binary[dat$smoking_father_postnatal_ordinal=="None"]<-0
dat$smoking_father_postnatal_binary[dat$smoking_father_postnatal_ordinal%in% c("Light","Moderate","Heavy")]<-1
```

## Remove original variables
Remove the original (uncleaned, underived) variables from the dataset
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%parent_smoking_vars)]
```

# Parents' alcohol {.tabset}

## Ideal variables
Ideal harmonised alcohol exposures for both parents are:

*	Pre-conception: Any vs no drinking in the three months prior to pregnancy
*	Pre-conception: Number of units per day in the three months prior to pregnancy (ordered categorical)
*	Pregnancy: Any drinking vs no drinking at any time during pregnancy
*	Pregnancy: Any drinking vs no drinking in the first trimester
*	Pregnancy: Any drinking vs no drinking in the second trimester
*	Pregnancy: Any drinking vs no drinking in the third trimester
*	Pregnancy: Number of units per day in the first trimester (ordered categorical)
*	Pregnancy: Number of units per day in the second trimester (ordered categorical)
*	Pregnancy: Number of units per day in the third trimester (ordered categorical)
* Binge drinking in first trimester (binary)
* Binge drinking in second trimester (binary)
* Binge drinking in third trimester (binary)
* Binge drinking at any point during pregnancy
* Postnatal alcohol use ordinal, up to 2 years  21 months (first 1000 days)
* Postnatal alcohol up to age 2 binary

## Set up meta-data
To get a complete list of relevant variables, we can use the alspac package to search the variable catalogue. So we start with a long list of variables in the meta_data object and then reduce it down to the most relevant ones.

Select variables containing key words and completed by mothers or partners:
```{r eval=FALSE, include=TRUE}
#get a list of alcohol questions asked to parents:
parent_alcohol_meta_data <- findVars("alcohol", "binge", "alcoholic", "unit", "beer", "lager", "wine", "spirit", "sherry", "port")
parent_alcohol_meta_data_mother <-parent_alcohol_meta_data[parent_alcohol_meta_data$cat3=="Mother",]
parent_alcohol_meta_data_father <-parent_alcohol_meta_data[parent_alcohol_meta_data$cat3=="Partner",]
```

Questions regarding alcohol before or during pregnancy (exposures) are going to be in the early questionnaires described above: A, B, C, D and E (PA and PB for dads). Postnatal alcohol will be up to questionnaire G for mothers and PE for partners.
```{r eval=FALSE, include=TRUE}
#select questions asked in questionnaires A-R or PA to PP:
parent_alcohol_meta_data_father <-parent_alcohol_meta_data_father[grep(lapply(parent_alcohol_meta_data_father$obj,substring,1,2),pattern="pa|pb|pc|pd|pe|pf|pg|ph|pi|pj|pk|pl|pm|pn|po|pp"),]

parent_alcohol_meta_data_mother <-parent_alcohol_meta_data_mother[grep(lapply(parent_alcohol_meta_data_mother$obj,substring,1,1),pattern="a|b|c|d|e|f|g|h|i|j|k|l|m|n|o|p|q|r"),]
```

At this point, it's necessary to go through the table (parent_alcohol_meta_data) manually, referring to the ALSPAC documentation where necessary, to decide which variables to use and for what.

View(parent_alcohol_meta_data_mother)

### Maternal

**Any time up to birth: drinking vs no drinking: binary variable**

* b720 Alcohol consumption before this pregnancy (includes "never" response)
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c("b720")
```

**Early-onset: Drinking before age 11 vs no drinking before age 11**

* Not available

**Pre-conception: Any vs no drinking in the three months prior to pregnancy**

* b720 Alcohol consumption before this pregnancy (already in list)

**Pre-conception: Number of units per day in the three months prior to pregnancy (ordered categorical)**

* not available, only b720 that's already in list

**Pregnancy: Any drinking vs no drinking at any time during pregnancy**

* Can be derived from the trimester specific alcohol variables (a261, b721, c373):
* a261 Measures of alcohol per week
* b721 alcohol consumption 1-3 months pregnancy
* c373 total alcoholic units per week
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("a261", "b721", "c373"))
```

**Pregnancy: Any drinking vs no drinking in the first trimester**

* a261 Measures of alcohol per week (already in list)

**Pregnancy: Any drinking vs no drinking in the second trimester**

* b721 alcohol consumption 1-3 months pregnancy (already in list)
* b722 Alcohol consumption baby 1ST moved
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("b722"))
```

**Pregnancy: Any drinking vs no drinking in the third trimester**

* c373 total alcoholic units per week (already in list)

**Pregnancy: Number of units per day in the first trimester (ordered categorical)**

*a261 Measures of alcohol per week (already in list)

**Pregnancy: Number of units per day in the second trimester (ordered categorical)**

*	Total the below:
* b754 total beer per week
*	b757 total wine per week
*	b760 total spirit per week
*	b769 Total other alcohol measures per week
* b775 total ther drinking per week
* b721 alcohol consumption 1-3 months pregnancy (already in list)
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("b754","b757","b760","b769", "b775"))
```

**Pregnancy: Number of units per day in the third trimester (ordered categorical)**

*c373 total weekly units (already in list)
*e220 FREQ of alcohol use in last 2MTHS of PREG
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("e220"))
```

**Binge drinking in first trimester**

* Not available

**Binge drinking in second trimester**

* b723 Days of 2PTS beer or EQUIV in past MTH
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("b723"))
```

**Binge drinking in third trimester**

* c360 Binges in past month
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("c360"))
```

**Finding variables on maternal alcohol (postnatal)**

* e221 FREQ of alcohol use since birth
* e222 NO days of 4units alcohol in past MTH
* f601 NO days last MTH had 4units alcohol
* f625 Alcohol consumption
* f626 NO days last MTH had 4units alcohol
* g822 Quantity of alcohol mum drinks
* g823 Quantity of alcohol Mum drinks
* g825 Days in past month had equiv 2 PT beer

```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("e221", "e222", "f601", "f625", "f626", "g822", "g823", "g825"))
```

### Paternal
Questions regarding alcohol before or during pregnancy (exposures) are going to be in the early questionnaires described above: PA and PB

View(parent_alcohol_meta_data_father)

**Any time: drinking vs no drinking: binary variable**

* pb099 how often drank alcohol pre-pregnancy (included a never response)
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("pb099"))
```

**Early-onset: Drinking before age 11 vs no drinking before age 11**

* Not available

**Pre-conception: Any vs no drinking in the three months prior to pregnancy**

* *Not available

**Pre-conception: Number of units per day in the three months prior to pregnancy**
(ordered categorical)
* *Not available

**Pregnancy: Any drinking vs no drinking at any time during pregnancy**
* b730 PTNRS alcohol consumption
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("b730"))
```

**Pregnancy: Any drinking vs no drinking in the first trimester**

*Not available

**Pregnancy: Any drinking vs no drinking in the second trimester**

* b730 Partners alcohol consumption (categorical, contains a never response) (already in list)
* pb100 Alcohol consumption in the past three months
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("pb100"))
```

**Pregnancy: Any drinking vs no drinking in the third trimester**

* pc280 Drank alcohol in last two months of pregnancy
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("pc280"))
```

**Pregnancy: Number of units per day in the first trimester (ordered categorical)**

* Not available

**Pregnancy: Number of units per day in the second trimester (ordered categorical)**

* pb100 how often drank alcohol in past 3 months (ordered categorical) (already in list)

**Pregnancy: Number of units per day in the third trimester (ordered categorical)**

* pc280 Drank alcohol in last two months of pregnancy (already in list)

**Binge drinking in pregnancy**

* pb101 Binge drinking in past month
```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("pb101"))
```

**Postnatal**

* pc281 Drunk alcohol since birth
* pd625 Frequency Alcoholic Drinks Drunk
* pe452 Amount Of Alcohol Drunk
* pe411 Binges past month
* f600 Partners alcohol consumption
* g750 Frequency partner drinks alcohol
* h602 frequency partner drinks alcohol
* p3190 Frequency partner drinks

```{r eval=FALSE, include=TRUE}
parent_alcohol_vars <-c(parent_alcohol_vars,c("pc281","pd625","pe452","pe411","f600","g750","h602","p3190"))
```

### Set up final meta data object
```{r eval=FALSE, include=TRUE}
parent_alcohol_meta_data <- findVars(unique(parent_alcohol_vars)) #create a table with all the necessary meta data for each variable

parent_alcohol_meta_data <- subset(parent_alcohol_meta_data,subset=tolower(name) %in% parent_alcohol_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables 
```{r eval=FALSE, include=TRUE}
parent_alcohol_data <-extractVars(parent_alcohol_meta_data[parent_alcohol_meta_data$name %in% unique(parent_alcohol_vars),]) #extracting individual level data and performing withdrawal of consent

parent_alcohol_data<-parent_alcohol_data[,c("aln","qlet",parent_alcohol_vars)]
dat <- left_join(dat,parent_alcohol_data,by=c("aln","qlet")) 
```

## Derive clean variables

### Maternal

**Pre-conception: Number of units per day in the three months prior to pregnancy (ordered categorical)**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_preconception_ordinal <-NA
dat$alcohol_mother_preconception_ordinal[dat$b720==1]<-"None"
dat$alcohol_mother_preconception_ordinal[dat$b720==2]<-"Light" # <1glass per week
dat$alcohol_mother_preconception_ordinal[dat$b720%in%c(3)]<-"Moderate" #1+glasses pwk, but not daily
dat$alcohol_mother_preconception_ordinal[dat$b720%in%c(4,5,6)]<-"Heavy" #daily drinking, 1-2pd, 3-9pd, 10+pd
dat$alcohol_mother_preconception_ordinal<-factor(dat$alcohol_mother_preconception_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

**Pregnancy: Number of units per day in the first trimester (ordered categorical)**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_firsttrim_ordinal <-NA
dat$alcohol_mother_firsttrim_ordinal[dat$b721==1]<-"None"
dat$alcohol_mother_firsttrim_ordinal[dat$b721==2]<-"Light" #i.e. <1 glass pw
dat$alcohol_mother_firsttrim_ordinal[dat$b721%in%c(3)]<-"Moderate" #1+glasses pwk, but not daily
dat$alcohol_mother_firsttrim_ordinal[dat$b721%in%c(4,5,6)]<-"Heavy" #daily drinking, 1-2pd, 3-9pd, 10+pd
dat$alcohol_mother_firsttrim_ordinal<-factor(dat$alcohol_mother_firsttrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

**Pregnancy: Number of units per day in the second trimester (ordered categorical)**

The question asked was actually "how much were you drinking at around the time you first felt the baby move, or if you haven't felt the baby move yet, the last 2 weeks". People usually feel the baby kick between 16 and 25 weeks (first trim), so we can pretty safely assume this question relates to the second trimester.
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_secondtrim_ordinal <-NA
dat$alcohol_mother_secondtrim_ordinal[dat$b722==1]<-"None"
dat$alcohol_mother_secondtrim_ordinal[dat$b722==2]<-"Light" #i.e. <1 glass pw
dat$alcohol_mother_secondtrim_ordinal[dat$b722%in%c(3)]<-"Moderate" #1+glasses pwk, but not daily
dat$alcohol_mother_secondtrim_ordinal[dat$b722%in%c(4,5,6)]<-"Heavy" #daily drinking, 1-2pd, 3-9pd, 10+pd
dat$alcohol_mother_secondtrim_ordinal<-factor(dat$alcohol_mother_secondtrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

**Pregnancy: Number of units per day in the third trimester (ordered categorical)**
```{r eval=FALSE, include=TRUE}
#Frequency of alcohol in last 2 months of pregnancy
dat$alcohol_mother_thirdtrim_ordinal <-NA
dat$alcohol_mother_thirdtrim_ordinal[dat$e220==1]<-"None"
dat$alcohol_mother_thirdtrim_ordinal[dat$e220==2]<-"Light" #i.e. <1 glass pw
dat$alcohol_mother_thirdtrim_ordinal[dat$e220%in%c(3)]<-"Moderate" #1+glasses pwk, but not daily
dat$alcohol_mother_thirdtrim_ordinal[dat$e220%in%c(4,5,6)]<-"Heavy" #i.e DAILY drinking 1-2 glasses pd, 3-9 glasses per day, 10+ glasses per day
dat$alcohol_mother_thirdtrim_ordinal<-factor(dat$alcohol_mother_thirdtrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

**Pregnancy: Any drinking vs no drinking preconception**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_preconception_binary <- NA
dat$alcohol_mother_preconception_binary[dat$alcohol_mother_preconception_ordinal=="None"] <- 0
dat$alcohol_mother_preconception_binary[dat$alcohol_mother_preconception_ordinal%in%c("Light","Moderate","Heavy")] <- 1
```

**Pregnancy: Any drinking vs no drinking in the first trimester**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_firsttrim_binary <- NA
dat$alcohol_mother_firsttrim_binary[dat$alcohol_mother_firsttrim_ordinal=="None"] <- 0
dat$alcohol_mother_firsttrim_binary[dat$alcohol_mother_firsttrim_ordinal%in%c("Light","Moderate","Heavy")] <- 1
```

**Pregnancy: Any drinking vs no drinking in the second trimester**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_secondtrim_binary <- NA
dat$alcohol_mother_secondtrim_binary[dat$alcohol_mother_secondtrim_ordinal=="None"] <- 0
dat$alcohol_mother_secondtrim_binary[dat$alcohol_mother_secondtrim_ordinal%in%c("Light","Moderate","Heavy")] <- 1
```

**Pregnancy: Any drinking vs no drinking in the third trimester**

```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_thirdtrim_binary <- NA
dat$alcohol_mother_thirdtrim_binary[dat$alcohol_mother_thirdtrim_ordinal=="None"] <- 0
dat$alcohol_mother_thirdtrim_binary[dat$alcohol_mother_thirdtrim_ordinal%in%c("Light","Moderate","Heavy")] <- 1
```

**Pregnancy: Any drinking vs no drinking at any time during pregnancy**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_ever_pregnancy_binary<-NA
dat$alcohol_mother_ever_pregnancy_binary[dat$alcohol_mother_firsttrim_binary==0 &dat$alcohol_mother_secondtrim_binary==0 & dat$alcohol_mother_thirdtrim_binary == 0] <-0
dat$alcohol_mother_ever_pregnancy_binary[dat$alcohol_mother_firsttrim_binary==1 |dat$alcohol_mother_secondtrim_binary ==1 |dat$alcohol_mother_thirdtrim_binary ==1] <-1
```

**Binge drinking in first trimester**

```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_binge1_binary<-NA
dat$alcohol_mother_binge1_binary[dat$b723==0]<- 0
dat$alcohol_mother_binge1_binary[dat$b723>0]<- 1
```

**Binge drinking in second trimester**

Not available

**Binge drinking in third trimester**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_binge3_binary<-NA
dat$alcohol_mother_binge3_binary[dat$c360==0]<- 0
dat$alcohol_mother_binge3_binary[dat$c360>0]<- 1
```

**Binge drinking at any point during pregnancy**
```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_bingepreg_binary <-NA
dat$alcohol_mother_bingepreg_binary[dat$alcohol_mother_binge1_binary==0&dat$alcohol_mother_binge3_binary==0] <-0
dat$alcohol_mother_bingepreg_binary[dat$alcohol_mother_binge1_binary==1|dat$alcohol_mother_binge3_binary==1] <-1
```

**Postnatal: ordinal alcohol use after birth of child up to 2 years (first 1000 days)**
```{r eval=FALSE, include=TRUE}
#alcohol use at 8 weeks = e221; 0=none, 1=light, 2=light to moderate, 3= Heavy
#alcohol use at 8 months = f625; 0=none, 1=light, 2=light to moderate, 3=heavy
#alcohol use at 21 months = g822; 0=none, 1=light, 2=light to moderate, 3=heavy
temp_var_8w<-NA
temp_var_8w[dat$e221==1]<-0
temp_var_8w[dat$e221==2]<-1
temp_var_8w[dat$e221==3]<-2
temp_var_8w[dat$e221%in%c(4,5,6)]<-3

temp_var_8m<-NA
temp_var_8m[dat$f625==1]<-0
temp_var_8m[dat$f625==2]<-1
temp_var_8m[dat$f625==3]<-2
temp_var_8m[dat$f625%in%c(4,5,6)]<-3

temp_var_21m<-NA
temp_var_21m[dat$g822==1]<-0
temp_var_21m[dat$g822==2]<-1
temp_var_21m[dat$g822==3]<-2
temp_var_21m[dat$g822%in%c(4,5,6)]<-3

temp_df<-data.frame(temp_var_8w,temp_var_8m,temp_var_21m)
temp_df$temp_var_test<-apply(temp_df,1,max,na.rm=T)
temp_df$temp_var_test[is.infinite(temp_df$temp_var_test)]<-NA
dat$alcohol_mother_postnatal_ordinal<-temp_df$temp_var_test

dat$alcohol_mother_postnatal_ordinal[dat$alcohol_mother_postnatal_ordinal==0]<-"None"
dat$alcohol_mother_postnatal_ordinal[dat$alcohol_mother_postnatal_ordinal==1]<-"Light"
dat$alcohol_mother_postnatal_ordinal[dat$alcohol_mother_postnatal_ordinal==2]<-"Moderate"
dat$alcohol_mother_postnatal_ordinal[dat$alcohol_mother_postnatal_ordinal==3]<-"Heavy"

dat$alcohol_mother_postnatal_ordinal<-factor(dat$alcohol_mother_postnatal_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)

rm(temp_df,temp_var_21m,temp_var_8m,temp_var_8w)
```

**Postnatal: any vs no drinking after birth of child up to 2 years**

```{r eval=FALSE, include=TRUE}
dat$alcohol_mother_postnatal_binary <- NA
dat$alcohol_mother_postnatal_binary[dat$alcohol_mother_postnatal_ordinal=="None"]<-0
dat$alcohol_mother_postnatal_binary[dat$alcohol_mother_postnatal_ordinal%in% c("Light","Moderate","Heavy")]<-1
```

### Paternal
Questions regarding alcohol before or during pregnancy (exposures) are going to be in the early questionnaires described above: PA and PB

**Early-onset: Drinking before age 11 vs no drinking before age 11**

* Not available

**Pre-conception: Any vs no drinking in the three months prior to pregnancy**

```{r eval=FALSE, include=TRUE}
dat$alcohol_father_preconception_binary <- NA
dat$alcohol_father_preconception_binary[dat$pb099==1]<-0
dat$alcohol_father_preconception_binary[dat$pb099%in%c(2,3,4,5,6)]<-1
```

**Pre-conception: Number of units per day in the three months prior to pregnancy (ordered categorical)**
```{r eval=FALSE, include=TRUE}
dat$alcohol_father_preconception_ordinal <-NA
dat$alcohol_father_preconception_ordinal[dat$pb099==1]<-"None"
dat$alcohol_father_preconception_ordinal[dat$pb099==2]<-"Light" # <1glass per week
dat$alcohol_father_preconception_ordinal[dat$pb099%in%c(3)]<-"Moderate" #1+glasses pwk, but not daily
dat$alcohol_father_preconception_ordinal[dat$pb099%in%c(4,5,6)]<-"Heavy" #daily drinking, 1-2pd, 3-9pd, 10+pd
dat$alcohol_father_preconception_ordinal<-factor(dat$alcohol_father_preconception_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

**Pregnancy: Number of units per day in the first trimester (ordered categorical)**

* Not available

**Pregnancy: Number of units per day in the second trimester (ordered categorical)**
```{r eval=FALSE, include=TRUE}
dat$alcohol_father_secondtrim_ordinal <-NA
dat$alcohol_father_secondtrim_ordinal[dat$pb100==1]<-"None"
dat$alcohol_father_secondtrim_ordinal[dat$pb100==2]<-"Light" # <1glass per week
dat$alcohol_father_secondtrim_ordinal[dat$pb100%in%c(3)]<-"Moderate" #1+glasses pwk, but not daily
dat$alcohol_father_secondtrim_ordinal[dat$pb100%in%c(4,5,6)]<-"Heavy" #daily drinking, 1-2pd, 3-9pd, 10+pd
dat$alcohol_father_secondtrim_ordinal<-factor(dat$alcohol_father_secondtrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

**Pregnancy: Number of units per day in the third trimester (ordered categorical)**
```{r eval=FALSE, include=TRUE}
dat$alcohol_father_thirdtrim_ordinal <-NA
dat$alcohol_father_thirdtrim_ordinal[dat$pc280==1]<-"None"
dat$alcohol_father_thirdtrim_ordinal[dat$pc280==2]<-"Light" # <1glass per week
dat$alcohol_father_thirdtrim_ordinal[dat$pc280%in%c(3)]<-"Moderate" #1+glasses pwk, but not daily
dat$alcohol_father_thirdtrim_ordinal[dat$pc280%in%c(4,5,6)]<-"Heavy" #daily drinking, 1-2pd, 3-9pd, 10+pd
dat$alcohol_father_thirdtrim_ordinal<-factor(dat$alcohol_father_thirdtrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

**Pregnancy: Any drinking vs no drinking in the first trimester**

*Not available

**Pregnancy: Any drinking vs no drinking in the second trimester**

```{r eval=FALSE, include=TRUE}
dat$alcohol_father_secondtrim_binary <- NA
dat$alcohol_father_secondtrim_binary[dat$pb100==1] <- 0
dat$alcohol_father_secondtrim_binary[dat$pb100>1] <- 1
```

**Pregnancy: Any drinking vs no drinking in the third trimester**
```{r eval=FALSE, include=TRUE}
dat$alcohol_father_thirdtrim_binary <- NA
dat$alcohol_father_thirdtrim_binary[dat$pc280==1] <- 0
dat$alcohol_father_thirdtrim_binary[dat$pc280>1] <- 1
```

**Pregnancy: Any drinking vs no drinking at any time during pregnancy**
```{r eval=FALSE, include=TRUE}
dat$alcohol_father_ever_pregnancy_binary<-NA
dat$alcohol_father_ever_pregnancy_binary[dat$alcohol_father_secondtrim_binary==0 & dat$alcohol_father_thirdtrim_binary == 0] <-0
dat$alcohol_father_ever_pregnancy_binary[dat$alcohol_father_secondtrim_binary ==1 |dat$alcohol_father_thirdtrim_binary ==1] <-1

dat$alcohol_father_ever_pregnancy_binary_mreport<-NA
dat$alcohol_father_ever_pregnancy_binary_mreport[dat$b730==1]<-0
dat$alcohol_father_ever_pregnancy_binary_mreport[dat$b730>1]<-1
```

**Binge drinking**

Only available in second trimester, so take this for the whole of pregnancy

```{r eval=FALSE, include=TRUE}
dat$alcohol_father_bingepreg_binary<-NA
dat$alcohol_father_bingepreg_binary[dat$pb101==0]<- 0
dat$alcohol_father_bingepreg_binary[dat$pb101>0]<- 1
```

**Postnatal: ordinal alcohol use after birth of child up to 2 years (first 1000 days)**
```{r eval=FALSE, include=TRUE}
#drunk alcohol since birth asked at 8 weeks= pc281; 0=none,1=light,2=light to moderate, 3=heavy
#how much alcohol drink at 8 months = pd625; 0=none,1=light,2=light to moderate, 3=heavy
#how much alochol drink at 1 year 9m = pe452; 0=none,1=light,2=light to moderate, 3=heavy

temp_var_pc<-NA
temp_var_pc[dat$pc281==1]<-0
temp_var_pc[dat$pc281%in%c(2)]<-1
temp_var_pc[dat$pc281==3]<-2
temp_var_pc[dat$pc281%in%c(4,5,6)]<-3

temp_var_pd<-NA
temp_var_pd[dat$pd625==5]<-0
temp_var_pd[dat$pd625%in%c(2,3)]<-1
temp_var_pd[dat$pd625==2]<-2
temp_var_pd[dat$pd625==1]<-3

temp_var_pe<-NA
temp_var_pe[dat$pe452==1]<-0
temp_var_pe[dat$pe452%in%c(2)]<-1
temp_var_pe[dat$pe452==3]<-2
temp_var_pe[dat$pe452%in%c(4,5,6)]<-3

temp_df<-data.frame(temp_var_pc,temp_var_pd,temp_var_pe)
temp_df$temp_var_test<-apply(temp_df,1,max,na.rm=T)
temp_df$temp_var_test[is.infinite(temp_df$temp_var_test)]<-NA
dat$alcohol_father_postnatal_ordinal<-temp_df$temp_var_test

dat$alcohol_father_postnatal_ordinal[dat$alcohol_father_postnatal_ordinal==0]<-"None"
dat$alcohol_father_postnatal_ordinal[dat$alcohol_father_postnatal_ordinal==1]<-"Light"
dat$alcohol_father_postnatal_ordinal[dat$alcohol_father_postnatal_ordinal==2]<-"Moderate"
dat$alcohol_father_postnatal_ordinal[dat$alcohol_father_postnatal_ordinal==3]<-"Heavy"

dat$alcohol_father_postnatal_ordinal<-factor(dat$alcohol_father_postnatal_ordinal,levels=c("None","Light","Moderate","Heavy"),ordered=T)

rm(temp_df,temp_var_pe,temp_var_pd,temp_var_pc)
```

**Postnatal: any vs no drinking after birth of child up to 2 years**

```{r eval=FALSE, include=TRUE}
dat$alcohol_father_postnatal_binary <- NA
dat$alcohol_father_postnatal_binary[dat$alcohol_father_postnatal_ordinal=="None"]<-0
dat$alcohol_father_postnatal_binary[dat$alcohol_father_postnatal_ordinal%in% c("Light","Moderate","Heavy")]<-1
```

## Remove original variables
Remove original variables we don't need anymore:
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%parent_alcohol_vars)]
```

# Parents' caffeine {.tabset}
## Ideal variables 

Caffeine from tea, coffee and cola only.

Ideal harmonised caffeine exposures for both parents are:

* Pre-conception: Any vs no caffeine in the three months prior to pregnancy
* Pre-conception: Amount of caffeine (mg/day) per day in the three months prior to pregnancy

* Pregnancy: caffeine from coffee in first trimester (mg/day)
* Pregnancy: caffeine from tea in first trimester (mg/day)
* Pregnancy: caffeine from cola in first trimester (mg/day)
* Pregnancy: total caffeine in first trimester (mg/day)

* Pregnancy: caffeine from coffee in second trimester (mg/day)
* Pregnancy: caffeine from tea in second trimester (mg/day)
* Pregnancy: caffeine from cola in second trimester (mg/day)
* Pregnancy: total caffeine in second trimester (mg/day)

* Pregnancy: caffeine from coffee in third trimester (mg/day)
* Pregnancy: caffeine from tea in third trimester (mg/day)
* Pregnancy: caffeine from cola in third trimester (mg/day)
* Pregnancy: total caffeine in third trimester (mg/day)

* Pregnancy: Any caffeine vs no caffeine in the first trimester
* Pregnancy: Any caffeine vs no caffeine in the second trimester
* Pregnancy: Any caffeine vs no caffeine in the third trimester

* Pregnancy: Average caffeine per day in pregnancy
* Pregnancy: Any caffeine vs no caffeine in pregnancy

* Postnatal: caffeine from coffee up to age 2 (mg/day)
* Postnatal: caffeine from tea up to age 2 (mg/day)
* Postnatal: caffeine from cola up to age 2 (mg/day)
* Postnatal: total caffeine up to age 2 (mg/day)
* Postnatal: any vs none

## Set up meta-data
To get a complete list of relevant variables, we can use the alspac package to search the variable catalogue. So we start with a long list of variables in the meta_data object and then reduce it down to the most relevant ones.

Select variables containing key words and completed by mothers or partners:
```{r eval=FALSE, include=TRUE}
#get a list of caffeine questions asked to parents:
parent_caffeine_meta_data <- findVars("caffeine", "coffee", "tea", "cola")
parent_caffeine_meta_data_mother <-parent_caffeine_meta_data[parent_caffeine_meta_data$cat3=="Mother",]
parent_caffeine_meta_data_father <-parent_caffeine_meta_data[parent_caffeine_meta_data$cat3=="Partner",]
```
Questions regarding caffeine before or during pregnancy (exposures) are going to be in the early questionnaires described above: A, B, C, D and E (PA and PB for dads). Postnatal caffeine will be up to questionnaire G for mothers and PE for partners.
```{r eval=FALSE, include=TRUE}
#select questions asked in questionnaires A-G or PA to PE:
parent_caffeine_meta_data_mother <-parent_caffeine_meta_data_mother[grep(lapply(parent_caffeine_meta_data_mother$obj,substring,1,1),pattern="a|b|c|d|e|f|g"),]

parent_caffeine_meta_data_father <-parent_caffeine_meta_data_father[grep(lapply(parent_caffeine_meta_data_father$obj,substring,1,2),pattern="pa|pb|pc|pd|pe|pf|pg|ph|pi|pj|pk|pl|pm|pn|po|pe"),]

```
At this point, it's necessary to go through the table (parent_caffeine_meta_data_xxx) manually, referring to the ALSPAC documentation where necessary, to decide which variables to use and for what.

### Maternal

**Early-onset: caffeine before age 11 vs no caffeine before age 11**

* not available 

**Pre-conception: Any vs no caffeine in the three months prior to pregnancy**

* not available 

**Pre-conception: Amount of caffeine (categorised mg/day) per day in the three months prior to pregnancy (ordered categorical)**

* not available 

**Pregnancy: Any caffeine vs no caffeine at any time during pregnancy**

* Can be derived from the trimester specific caffeine variables

**Pregnancy: Any caffeine vs no caffeine in the first trimester**

* Can be derived from the trimester specific caffeine variables

**Pregnancy: Any caffeine vs no caffeine in the second trimester**

* Can be derived from the trimester specific caffeine variables

**Pregnancy: Any caffeine vs no caffeine in the third trimester**

* Can be derived from the trimester specific caffeine variables

**Pregnancy: Amount of caffeine (categorised mg/day) per day in the first trimester (ordered categorical)**

* a228 Cups of coffee per week (use to make binary)
* a222 Cups of tea per week
* a225 Cups of decaff tea per week
* a231 Cups of decaff. coffee per week
* a234 No. cans of cola per week
* a237 Cans of decaff. cola per week
```{r eval=FALSE, include=TRUE}
parent_caffeine_vars <-c("a228", "a222", "a225", "a231", "a234", "a237")
```

**Pregnancy: Amount of caffeine (categorised mg/day) per day in the second trimester (ordered categorical)**

* b748 total coffee per week
* b742 Total tea per week
* b745 Total DECAF tea per week
* b751 Total DECAF coffee per week
* b763 Total cola drinking per week
* b766 Total DECAF cola per week

```{r eval=FALSE, include=TRUE}
parent_caffeine_vars <-c(parent_caffeine_vars,c("b748", "b742", "b745", "b751", "b763", "b766"))
```

**Pregnancy: Amount of caffeine (categorised mg/day) per day in the third trimester (ordered categorical)**
The total weekly caffeine dose is calculated by lower amounts of caffeine per cup than used by Treur et al. Therefore will calculate own and unlikely use ALSPAC's total

* c300 Cups of tea per day
* c303 CUPS OF DECAFF. TEA (NEW C12C)
* c305 cups of coffee per day
* c309 Cups Of Real Coffee (New C12G)
* c310 Drinks of cola PWK
* c312 Drinks Of Decaff. Cola Pwk (New C13B)
* c317 Weekly caffeine dose
```{r eval=FALSE, include=TRUE}
parent_caffeine_vars <-c(parent_caffeine_vars,c("c300", "c303", "c305", "c309", "c310", "c312", "c317"))
```

**Postnatal**

Questions regarding caffeine after pregnancy (postnatal mediator) are going to be in the postnatal questionnaires, E to R:

* e157 Number of ord real coffee daily
* e150	NO DECAF cola cans PWK
* e151	NO ORD cola cans PWK
* e152	NO cups of DECAF tea daily
* e153 NO cups of ORD tea daily
* e154	NO cups of DECAF INST coffee daily
* e155	NO cups of ORD INST coffee daily
* e156	NO cups of DECAF real coffee daily
* e157	NO cups of ORD real coffee daily

```{r eval=FALSE, include=TRUE}
parent_caffeine_vars <-c(parent_caffeine_vars,c("e157","e150","e151","e152","e154","e155","e153","e156"))
```

### Paternal

**Any time: caffeine vs no caffeine: binary variable**

* Can be derived from the trimester specific caffeine variables

**Early-onset: caffeine before age 11 vs no caffeine before age 11**

* not available 

**Pre-conception: Any vs no caffeine in the three months prior to pregnancy**

* not available 

**Pre-conception: Amount of caffeine (categorised mg/day) per day in the three months prior to pregnancy (ordered categorical)**

* not available 

**Pregnancy: Any caffine vs no caffeine at any time during pregnancy**

* Can be derived from the trimester specific caffeine variables

**Pregnancy: Any caffeine vs no caffeine in the first trimester**

* not available 

**Pregnancy: Any caffeine vs no caffeine in the second trimester**

* Can be derived from the trimester specific caffeine variables

**Pregnancy: Any caffeine vs no caffeine in the third trimester**

* not available 

**Pregnancy: Amount of caffeine (categorised mg/day) per day in the first trimester (ordered categorical)**

* not available 

**Pregnancy: Amount of caffeine (categorised mg/day) per day in the second trimester (ordered categorical)**

* pb054 Cups of tea per day
* pb057 No. cups tea/day wh. are DECAF (150791+)
* pb058 cups of coffee per day (use to make binary var)
* pb061 No.cups coffee/day wh.are DECAF(150791+)
* pb062 No. cups real coffee/day (150791+)
* pb063 No.cups realcoffee wh.are DECAF(150791+)
* pb064 Drinks of cola PWK
* pb066 No.drinks cola which are DECAF (150791+)

```{r eval=FALSE, include=TRUE}
parent_caffeine_vars <-c(parent_caffeine_vars,c("pb054", "pb057", "pb058", "pb060a", "pb061", "pb062", "pb063", "pb064", "pb066","pb056","pb056a","pb060","pb065","pb065a"))
```

**Pregnancy: Amount of caffeine (categorised mg/day) per day in the third trimester (ordered categorical)**

* not available 

**Paternal postnatal**

*not available

### Set up final meta data object
```{r eval=FALSE, include=TRUE}
parent_caffeine_meta_data <- findVars(unique(parent_caffeine_vars)) #create a table with all the necessary meta data for each variable

parent_caffeine_meta_data <- subset(parent_caffeine_meta_data,subset=tolower(name) %in% parent_caffeine_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
```{r eval=FALSE, include=TRUE}
parent_caffeine_data <- extractVars(parent_caffeine_meta_data[parent_caffeine_meta_data$name %in% unique(parent_caffeine_vars),]) #extracting individual level data and performing withdrawal of consent

parent_caffeine_data<-parent_caffeine_data[,c("aln","qlet",parent_caffeine_vars)]
dat <- left_join(dat,parent_caffeine_data,by=c("aln","qlet")) 
```

## Derive clean variables

### Maternal

**First trimester coffee mg per day**

The n is quite low here because we have to limit to mothers who filled in the questionnaire in the first trimester (so use average over three trimesters to derive any/no caffeine in pregnancy rather than enforcing that this variable has to be 0 for no caffeine)
```{r eval=FALSE, include=TRUE}
#cups of coffee per week
dat$caffeine_mother_coffee_firsttrim_continuous <-dat$a228
dat$caffeine_mother_coffee_firsttrim_continuous[dat$a228<0]<-NA
dat$caffeine_mother_coffee_firsttrim_continuous[dat$time_qA_mother_gestation>12]<-NA
# cups of coffee multiplied by 57 (mg caffeine in 1 cup coffee)
dat$caffeine_mother_coffee_firsttrim_continuous<-dat$caffeine_mother_coffee_firsttrim_continuous*57
# mg per day
dat$caffeine_mother_coffee_firsttrim_continuous<-round(dat$caffeine_mother_coffee_firsttrim_continuous/7)
```

**First trimester tea mg per day**

```{r eval=FALSE, include=TRUE}
#cups of tea per week
dat$caffeine_mother_tea_firsttrim_continuous <-dat$a222
dat$caffeine_mother_tea_firsttrim_continuous[dat$a222<0]<-NA
dat$caffeine_mother_tea_firsttrim_continuous[dat$time_qA_mother_gestation>12]<-NA
# cups of tea multiplied by 27 (mg caffeine in 1 cup tea)
dat$caffeine_mother_tea_firsttrim_continuous<-dat$caffeine_mother_tea_firsttrim_continuous*27
# mg per day
dat$caffeine_mother_tea_firsttrim_continuous<-round(dat$caffeine_mother_tea_firsttrim_continuous/7)
```

**First trimester cola mg per day**

```{r eval=FALSE, include=TRUE}
#cups of cola per week
dat$caffeine_mother_cola_firsttrim_continuous <-dat$a234
dat$caffeine_mother_cola_firsttrim_continuous[dat$a234<0]<-NA
dat$caffeine_mother_cola_firsttrim_continuous[dat$time_qA_mother_gestation>12]<-NA
# cups of cola multiplied by 27 (mg caffeine in 1 cup cola)
dat$caffeine_mother_cola_firsttrim_continuous<-dat$caffeine_mother_cola_firsttrim_continuous*20
# mg per day
dat$caffeine_mother_cola_firsttrim_continuous<-round(dat$caffeine_mother_cola_firsttrim_continuous/7)
```

**First trimester total caffeine mg per day**
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_firsttrim_continuous <- rowSums(dat[,c("caffeine_mother_cola_firsttrim_continuous","caffeine_mother_coffee_firsttrim_continuous","caffeine_mother_tea_firsttrim_continuous")],na.rm=TRUE)
dat$caffeine_mother_total_firsttrim_continuous[which(is.na(dat$caffeine_mother_coffee_firsttrim_continuous) & is.na(dat$caffeine_mother_tea_firsttrim_continuous) & is.na(dat$caffeine_mother_cola_firsttrim_continuous))]<-NA
```

**Second trimester tea mg per day**

```{r eval=FALSE, include=TRUE}
#TEA
dat$caffeine_mother_tea_secondtrim_continuous<-dat$b742
dat$caffeine_mother_tea_secondtrim_continuous[dat$b742<0]<-NA
dat$caffeine_mother_tea_secondtrim_continuous<-dat$caffeine_mother_tea_secondtrim_continuous*27 #tranform cups to mg per week
dat$caffeine_mother_tea_secondtrim_continuous<-dat$caffeine_mother_tea_secondtrim_continuous/7 #tranform weekly caffeine to daily
```

**Second trimester coffee mg per day**
```{r eval=FALSE, include=TRUE}
#COFFEE
dat$caffeine_mother_coffee_secondtrim_continuous<-dat$b748
dat$caffeine_mother_coffee_secondtrim_continuous[dat$b748<0]<-NA
dat$caffeine_mother_coffee_secondtrim_continuous<-dat$caffeine_mother_coffee_secondtrim_continuous*57 #tranform cups to mg per week
dat$caffeine_mother_coffee_secondtrim_continuous<-dat$caffeine_mother_coffee_secondtrim_continuous/7 #tranform weekly caffeine to daily
```

**Second trimester cola mg per day**
```{r eval=FALSE, include=TRUE}
#COLA
dat$caffeine_mother_cola_secondtrim_continuous<-dat$b763
dat$caffeine_mother_cola_secondtrim_continuous[dat$b763<0]<-NA
dat$caffeine_mother_cola_secondtrim_continuous<-dat$caffeine_mother_cola_secondtrim_continuous*20 #tranform cups to mg per week
dat$caffeine_mother_cola_secondtrim_continuous<-dat$caffeine_mother_cola_secondtrim_continuous/7 #tranform weekly caffeine to daily
```

**Second trimester total caffeine mg per day**
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_secondtrim_continuous <- rowSums(dat[,c("caffeine_mother_cola_secondtrim_continuous","caffeine_mother_coffee_secondtrim_continuous","caffeine_mother_tea_secondtrim_continuous")],na.rm=TRUE)
dat$caffeine_mother_total_secondtrim_continuous[which(is.na(dat$caffeine_mother_coffee_secondtrim_continuous) & is.na(dat$caffeine_mother_tea_secondtrim_continuous) & is.na(dat$caffeine_mother_cola_secondtrim_continuous))]<-NA
```

**Third trimester tea caffeine mg per day**

```{r eval=FALSE, include=TRUE}
#TEA (cups of caffeinated tea per day)
dat$caffeine_mother_tea_thirdtrim_continuous<-dat$c300
dat$caffeine_mother_tea_thirdtrim_continuous[dat$c300<0]<-NA
dat$caffeine_mother_tea_thirdtrim_continuous<-dat$caffeine_mother_tea_thirdtrim_continuous*27 #tranform cups to mg per day
```

**Third trimester coffee caffeine mg per day**

```{r eval=FALSE, include=TRUE}
#COFFEE (cups of caffeinated coffee per day)
dat$caffeine_mother_coffee_thirdtrim_continuous<-dat$c305
dat$caffeine_mother_coffee_thirdtrim_continuous[dat$c305<0]<-NA
dat$caffeine_mother_coffee_thirdtrim_continuous<-dat$caffeine_mother_coffee_thirdtrim_continuous*57 #tranform cups to mg per day
```

**Third trimester cola caffeine mg per day**

```{r eval=FALSE, include=TRUE}
#COLA (cups of caffeinated cola per week)
dat$caffeine_mother_cola_thirdtrim_continuous<-dat$c310
dat$caffeine_mother_cola_thirdtrim_continuous[dat$c310<0]<-NA
dat$caffeine_mother_cola_thirdtrim_continuous<-dat$caffeine_mother_cola_thirdtrim_continuous*20 #tranform cups to mg per week
dat$caffeine_mother_cola_thirdtrim_continuous<-dat$caffeine_mother_cola_thirdtrim_continuous/7 #transform to mg per day
```

**Third trimester total caffeine mg per day**
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_thirdtrim_continuous <- rowSums(dat[,c("caffeine_mother_cola_thirdtrim_continuous","caffeine_mother_coffee_thirdtrim_continuous","caffeine_mother_tea_thirdtrim_continuous")],na.rm=TRUE)
dat$caffeine_mother_total_thirdtrim_continuous[which(is.na(dat$caffeine_mother_coffee_thirdtrim_continuous) & is.na(dat$caffeine_mother_tea_thirdtrim_continuous) & is.na(dat$caffeine_mother_cola_thirdtrim_continuous))]<-NA
```

**Any vs none in first trim **
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_firsttrim_binary <- NA
dat$caffeine_mother_total_firsttrim_binary[dat$caffeine_mother_total_firsttrim_continuous==0]<-0
dat$caffeine_mother_total_firsttrim_binary[dat$caffeine_mother_total_firsttrim_continuous>0]<-1
```

**Any vs none in second trim **
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_secondtrim_binary <- NA
dat$caffeine_mother_total_secondtrim_binary[dat$caffeine_mother_total_secondtrim_continuous==0]<-0
dat$caffeine_mother_total_secondtrim_binary[dat$caffeine_mother_total_secondtrim_continuous>0]<-1
```

**Any vs none in third trim **
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_thirdtrim_binary <- NA
dat$caffeine_mother_total_thirdtrim_binary[dat$caffeine_mother_total_thirdtrim_continuous==0]<-0
dat$caffeine_mother_total_thirdtrim_binary[dat$caffeine_mother_total_thirdtrim_continuous>0]<-1
```

**Average per day in pregnancy**
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_ever_pregnancy_continuous <- rowMeans(dat[,c("caffeine_mother_total_thirdtrim_continuous","caffeine_mother_total_secondtrim_continuous","caffeine_mother_total_firsttrim_continuous")],na.rm=TRUE)
```

**Any vs none any time in pregnancy **
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_ever_pregnancy_binary <- NA
dat$caffeine_mother_total_ever_pregnancy_binary[dat$caffeine_mother_total_ever_pregnancy_continuous==0]<-0
dat$caffeine_mother_total_ever_pregnancy_binary[dat$caffeine_mother_total_ever_pregnancy_continuous>0]<-1
```

**Postnatal tea caffeine mg per day**

```{r eval=FALSE, include=TRUE}
#TEA (cups of caffeinated tea per day)
dat$caffeine_mother_tea_postnatal_continuous<-dat$e153
dat$caffeine_mother_tea_postnatal_continuous[dat$e153<0]<-NA
dat$caffeine_mother_tea_postnatal_continuous<-dat$caffeine_mother_tea_postnatal_continuous*27 #tranform cups to mg per day
```

**Postnatal coffee caffeine mg per day**

```{r eval=FALSE, include=TRUE}
#COFFEE (cups of caffeinated coffee per day)
dat$caffeine_mother_coffee_postnatal_continuous<-dat$e155
dat$caffeine_mother_coffee_postnatal_continuous[dat$e155<0]<-NA
dat$caffeine_mother_coffee_postnatal_continuous<-dat$caffeine_mother_coffee_postnatal_continuous*57 #tranform cups to mg per day
```

**Postnatal cola caffeine mg per day**

```{r eval=FALSE, include=TRUE}
#COLA (cups of caffeinated cola per week)
dat$caffeine_mother_cola_postnatal_continuous<-dat$e151
dat$caffeine_mother_cola_postnatal_continuous[dat$e151<0]<-NA
dat$caffeine_mother_cola_postnatal_continuous<-dat$caffeine_mother_cola_postnatal_continuous*20 #tranform cups to mg per week
dat$caffeine_mother_cola_postnatal_continuous<-dat$caffeine_mother_cola_postnatal_continuous/7 #transform to mg per day
```

**Postnatal total caffeine mg per day**
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_postnatal_continuous <- rowSums(dat[,c("caffeine_mother_cola_postnatal_continuous","caffeine_mother_coffee_postnatal_continuous","caffeine_mother_tea_postnatal_continuous")],na.rm=TRUE)
dat$caffeine_mother_total_postnatal_continuous[which(is.na(dat$caffeine_mother_coffee_postnatal_continuous) & is.na(dat$caffeine_mother_tea_postnatal_continuous) & is.na(dat$caffeine_mother_cola_postnatal_continuous))]<-NA
```

**Any vs none postnatally**
```{r eval=FALSE, include=TRUE}
dat$caffeine_mother_total_postnatal_binary <- NA
dat$caffeine_mother_total_postnatal_binary[dat$caffeine_mother_total_postnatal_continuous==0]<-0
dat$caffeine_mother_total_postnatal_binary[dat$caffeine_mother_total_postnatal_continuous>0]<-1
```

### Paternal

For dads, we only have one time point (in the second trimester), which can be used for "any time in pregnancy". The problem is that the questions on how many cups of tea/coffee/cola were decaf were different in different versions of the questionnaire. In the beginning, they were asked to say whether "some" "all" or "none" of their beverages were decaf, but in later versions they were asked to give the number of cups that were decaf. If we just take the question from the latter version, we end up with a lot of missing data, so we need to consider all versions of that question.

**Second trimester tea mg per day**

```{r eval=FALSE, include=TRUE}
#TEA
dat$caffeine_father_tea_secondtrim_continuous<-dat$pb054 #cups of tea per day)
dat$caffeine_father_tea_secondtrim_continuous[dat$pb054<0]<-NA
dat$caffeine_father_tea_secondtrim_continuous[dat$pb056a==1]<-0#set to no caffeine if say always or usually drink decaf
dat$caffeine_father_tea_secondtrim_continuous[dat$pb056==1]<-0#set to no caffeine if say always or usually drink decaf
dat$pb057[dat$pb056a==3]<-0#set decaf to 0 if say never drink decaf
dat$pb057[dat$pb056==3]<-0#set decaf to 0 if say never drink decaf
dat$pb057[dat$pb057<0]<-NA #set missing to NA in n decaf cups 
dat$caffeine_father_tea_secondtrim_continuous<-dat$caffeine_father_tea_secondtrim_continuous-dat$pb057
dat$caffeine_father_tea_secondtrim_continuous[dat$caffeine_father_tea_secondtrim_continuous<0] <-NA
dat$caffeine_father_tea_secondtrim_continuous<-dat$caffeine_father_tea_secondtrim_continuous*27 #tranform cups to mg per day
```

**Second trimester coffee mg per day**
```{r eval=FALSE, include=TRUE}
#COFFEE
dat$caffeine_father_coffee_secondtrim_continuous<-dat$pb058 #cups of coffee per day)
dat$caffeine_father_coffee_secondtrim_continuous[dat$pb058<0]<-NA
dat$caffeine_father_coffee_secondtrim_continuous[dat$pb060a==1]<-0#set to no caffeine if say always or usually drink decaf
dat$caffeine_father_coffee_secondtrim_continuous[dat$pb060==1]<-0#set to no caffeine if say always or usually drink decaf
dat$pb061[dat$pb060a==3]<-0#set decaf to 0 if say never drink decaf
dat$pb061[dat$pb060==3]<-0#set decaf to 0 if say never drink decaf
dat$pb061[dat$pb061<0]<-NA #set missing to NA in n decaf cups 
dat$caffeine_father_coffee_secondtrim_continuous<-dat$caffeine_father_coffee_secondtrim_continuous-dat$pb061
dat$caffeine_father_coffee_secondtrim_continuous[dat$caffeine_father_coffee_secondtrim_continuous<0] <-NA
dat$caffeine_father_coffee_secondtrim_continuous<-dat$caffeine_father_coffee_secondtrim_continuous*57 #transform cups to mg per day
```

**Second trimester cola mg per day**
```{r eval=FALSE, include=TRUE}
#COLA
dat$caffeine_father_cola_secondtrim_continuous<-dat$pb064 #cups of cola per weel)
dat$caffeine_father_cola_secondtrim_continuous[dat$pb064<0]<-NA
dat$caffeine_father_cola_secondtrim_continuous[dat$pb065a==1]<-0#set to no caffeine if say always or usually drink decaf
dat$caffeine_father_cola_secondtrim_continuous[dat$pb065==1]<-0#set to no caffeine if say always or usually drink decaf
dat$pb066[dat$pb065a==3]<-0#set decaf to 0 if say never drink decaf
dat$pb066[dat$pb065==3]<-0#set decaf to 0 if say never drink decaf
dat$pb066[dat$pb066<0]<-NA #set missing to NA in n decaf cups 
dat$caffeine_father_cola_secondtrim_continuous<-dat$caffeine_father_cola_secondtrim_continuous-dat$pb066
dat$caffeine_father_cola_secondtrim_continuous[dat$caffeine_father_cola_secondtrim_continuous<0] <-NA
dat$caffeine_father_cola_secondtrim_continuous<-dat$caffeine_father_cola_secondtrim_continuous*20 #tranform cups to mg per week
dat$caffeine_father_cola_secondtrim_continuous<-dat$caffeine_father_cola_secondtrim_continuous/7 #per day
```

**Second trimester total caffeine mg per day**
```{r eval=FALSE, include=TRUE}
dat$caffeine_father_total_secondtrim_continuous <- rowSums(dat[,c("caffeine_father_cola_secondtrim_continuous","caffeine_father_coffee_secondtrim_continuous","caffeine_father_tea_secondtrim_continuous")],na.rm=TRUE)
dat$caffeine_father_total_secondtrim_continuous[which(is.na(dat$caffeine_father_coffee_secondtrim_continuous) & is.na(dat$caffeine_father_tea_secondtrim_continuous) & is.na(dat$caffeine_father_cola_secondtrim_continuous))]<-NA
```

**Any vs none in second trimester**
```{r eval=FALSE, include=TRUE}
dat$caffeine_father_total_secondtrim_binary <- NA
dat$caffeine_father_total_secondtrim_binary[dat$caffeine_father_total_secondtrim_continuous==0]<-0
dat$caffeine_father_total_secondtrim_binary[dat$caffeine_father_total_secondtrim_continuous>0]<-1
```

**Any in pregnancy** (using only the second trimester)
```{r eval=FALSE, include=TRUE}
dat$caffeine_father_total_ever_pregnancy_binary <- dat$caffeine_father_total_secondtrim_binary
dat$caffeine_father_total_ever_pregnancy_continuous <- dat$caffeine_father_total_secondtrim_continuous
```

## Remove original variables
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%parent_caffeine_vars)]
```

# Parents' physical activity {.tabset}

## Ideal variables

* Maternal and paternal PA before pregnancy
* Maternal and paternal PA in the first trimester
* Maternal and paternal PA in the second trimester
* Maternal and paternal PA in the third trimester
* Maternal and paternal PA any time in pregnancy
* Maternal and paternal PA in the postnatal period up to age 2

## Set up meta-data

Because I added this exposure after the others, I just searched the variable catalogue and identified the following variables (i.e. no need to use the alspac package):

* pb013 - PA once per week at present
* pb014 - hours per week PA at present

* b555 - PA in year before pregnancy (exertion - specifically re job though so not included)
* b556 - PA in first trim (exertion - specifically re job though so not included)
* b557 - PA in second trim (exertion - specifically re job though so not included)
* b634 - PA once per week at present (long enough to work up a sweat)
* b635 - hours per week PA at present
* b862 - index of leisure time (will be too difficult to harmonise with other cohorts)
* c503 - PA once per week at present
* c504 - hours per week PA at present
* g761 - exercises once per week at present (21 m)
* g762 - days pwk exercise at present (21 m)

```{r eval=FALSE, include=TRUE}
phys_act_vars <- c("pb013","pb014","b555","b556","b557","b635","b634","b862","c503","c504","g761","g762")

pa_meta_data <- findVars(unique(phys_act_vars)) #create a table with all the necessary meta data for each variable

pa_meta_data <- subset(pa_meta_data,subset=tolower(name) %in% phys_act_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
```{r eval=FALSE, include=TRUE}
pa_data <- extractVars(pa_meta_data[pa_meta_data$name %in% unique(phys_act_vars),]) #extracting individual level data and performing withdrawal of consent

pa_data<-pa_data[,c("aln","qlet",phys_act_vars)]
dat <- left_join(dat,pa_data,by=c("aln","qlet")) 
```

## Derive clean variables

From Liu et al 2011 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3898349/: 
"ALSPAC has included a series of physical activity questions, which were derived by the study team and tested by Wildschut et al. [29] Information on the frequency of womens physical activity during pregnancy was collected using identical questions in both the 18 and 32-week antenatal self-administrated questionnaire. Women were asked nowadays, at least once a week do you engage in any regular activity like brisk walking, gardening, housework, jogging, cycling etc., long enough to work up a sweat? We defined these activities as strenuous on the basis that the participants were specifically asked to respond yes only if the activity was done to a level that induced sweating. Those who answered Yes were asked to report their number of hours per week spent in such an activity. Because the duration of strenuous physical activity was reported in hours not in minutes, we chose a cut-off point (3 h per week) to dichotomize women into two groups. The cut-off point is used to approximate the level of exercise at moderate intensity for 30 min or more a day for most days of the week recommended by ACOG for healthy pregnant women [24] and recognized by RCOG [26]."

* Maternal and paternal PA in the second trimester

```{r eval=FALSE, include=TRUE}
dat$physact_mother_secondtrim_ordinal <- NA
dat$physact_mother_secondtrim_ordinal[dat$b635==0|dat$b634==2]<-"Inactive" #0
dat$physact_mother_secondtrim_ordinal[which(dat$b635>0&dat$b635<3)]<-"Somewhat" #<3
dat$physact_mother_secondtrim_ordinal[dat$b635>=3]<-"Active" #>=3
dat$physact_mother_secondtrim_ordinal<-factor(dat$physact_mother_secondtrim_ordinal,levels=c("Inactive","Somewhat","Active"),ordered=T)

dat$physact_father_secondtrim_ordinal <- NA
dat$physact_father_secondtrim_ordinal[dat$pb014==0|dat$pb013==2]<-"Inactive" #0
dat$physact_father_secondtrim_ordinal[which(dat$pb014>0&dat$pb014<3)]<-"Somewhat" #<3
dat$physact_father_secondtrim_ordinal[dat$pb014>=3]<-"Active" #>=3
dat$physact_father_secondtrim_ordinal<-factor(dat$physact_father_secondtrim_ordinal,levels=c("Inactive","Somewhat","Active"),ordered=T)
```

* Maternal PA in the third trimester

```{r eval=FALSE, include=TRUE}
dat$physact_mother_thirdtrim_ordinal <- NA
dat$physact_mother_thirdtrim_ordinal[dat$c504==0|dat$c503==2]<-"Inactive" #0
dat$physact_mother_thirdtrim_ordinal[dat$c504>0&dat$c504<3]<-"Somewhat" #<3
dat$physact_mother_thirdtrim_ordinal[dat$c504>=3]<-"Active" #>=3
dat$physact_mother_thirdtrim_ordinal<-factor(dat$physact_mother_thirdtrim_ordinal,levels=c("Inactive","Somewhat","Active"),ordered=T)
```

* Maternal PA in the postnatal period up to age 2
```{r eval=FALSE, include=TRUE}
dat$physact_mother_postnatal_ordinal <- NA
dat$physact_mother_postnatal_ordinal[dat$g761==2]<-"Inactive" #0
dat$physact_mother_postnatal_ordinal[dat$g762>0&dat$g762<3]<-"Somewhat" #<3
dat$physact_mother_postnatal_ordinal[dat$g762>=3]<-"Active" #>=3
dat$physact_mother_postnatal_ordinal<-factor(dat$physact_mother_postnatal_ordinal,levels=c("Inactive","Somewhat","Active"),ordered=T)
```

## Remove original variables
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%phys_act_vars)]
```

# Parents' mediterranean diet

The following code was developed in NutriPROGRAM so has a different format/approach to the rest of the code in this document. But it does the job.

```{r eval=FALSE, include=TRUE}
var.names <- c(#FFQ
#start of weekly variables
"c200", #FREQ of eating sausages or burgers
"c201", # FREQ of eating pies or pasties
"c202", #FREQ of eating meat
"c203", #FREQ of eating poultry
"c204", #FREQ of eating organ
"c205", #FREQ of eating white fish
"c206", #FREQ of eating oily fish
"c207", #FREQ of eating shellfish
"c208", #FREQ of eating eggs or quiche
"c209", #FREQ of eating cheese
"c210", #FREQ of eating pizza
"c211", #FREQ of eating chips
"c215", #FREQ of eating roast potatoes
"c216", #FREQ of eating boiled or baked potatoes
"c217", #FREQ of eating boiled rice
"c218", #FREQ of eating pasta
"c219", #FREQ of eating crisps
"c220", #FREQ of eating fried food
"c221", #Eat fat on meat
"c222", #FREQ of eating baked beans
"c223", #FREQ of eating peas corn or SIM
"c224", #FREQ of eating cabbage or SIM
"c225", #FREQ of eating other green VEG
"c226", #FREQ of eating carrots
"c227", #FREQ of eating root VEG not INC carrots
"c228", #FREQ of eating salad
"c229", #FREQ of eating fresh fruit
"c230", #FREQ of drinking tinned juice
"c231", #FREQ of drinking pure non tinned juice
"c232", #FREQ of eating pudding
"c233", #FREQ of eating oat cereals
"c234", #FREQ of eating bran cereals
"c235", #FREQ of eating other cereals
"c236", #FREQ of eating cakes or buns
"c237", #FREQ of eating crispbreads
"c238", #FREQ of eating biscuits
"c239", #FREQ of eating chocolate bars
"c240", #FREQ of eating pulses
"c241", #FREQ of eating nuts
"c242", #FREQ of eating bean curd
"c243", #FREQ of eating tahini
"c244", #FREQ of eating soya or SIM non meat
"c245", #FREQ of eating chocolate
"c246", #FREQ of eating sweets
#end of weekly variables
"c247", #FREQ of choosing diet soft drink
"c250", #Slices of bread eaten per day
"c275", #Slices of bread with fat daily
"c251", #Take away main meals per MTH
"c252", #Eat white bread most days
"c253", #Eat brown or granary bread
"c254", #Eat wholemeal bread
"c255", #Eat chappatis most days
"c256", #Dont usually eat any bread
"c260", #Mainly use butter etc on bread & VEG
"c261", #Mainly use butter etc for frying
"c263", #Mainly use hard or soft MARG on bread
"c264", #Mainly use hard or soft MARG for frying
"c265", #Mainly use POLYUNSAT MARG on bread & VEG
"c266", #Mainly use POLYUNSAT MARG for frying
"c267", #Mainly use low fat spread on bread & VEG
"c268", #Mainly use low fat spread for frying
"c269", #Mainly use sunflower oil or SIM on VEG
"c270", #Mainly fry with sunflower oil or SIM
"c271", #Mainly use other VEG oil on bread & VEG
"c272", #Mainly use other VEG oil for frying
"c273", #Other fat used mainly on bread & VEG
"c274", #Other fat used mainly for frying
"c276", #Use full fat milk
"c277", #Use semi skimmed milk
"c278", #Use skimmed milk
"c279", #Use sterilised milk
"c280", #Use dried milk
"c281", #Use goat or sheep milk
"c282", #Use soya milk
"c283", #Use other milk
#for "milkvol" - start
"c284", #Milk in tea
"c285", #Milk in coffee	
"c287", #Milk as pudding
"c288", #Milk to drink on its own
"c289", #Milk as a milky drink
"c286", #Milk on breakfast cereal
#end
"c300", #Cups of caffeinated tea per day
"c301", #Spoons of sugar per cup of tea
"c305", #Cups of coffee per day
"c306", #Spoons of sugar per cup of coffee
"c310", #Drinks of cola PWK
"c373", #
				#covariates
				"kz021", ## gender; 1 =male; 2= female
                "c3804",# DV: Daily energy intake (kJ) FFQ for mothers: 32-week gestation
)

#########################################
# Read in data: 
#########################################

alspac.vars <- findVars(var.names) 

#remove variables that appear more than once (keep the main version from Current data)
alspac.vars<-alspac.vars[-which(alspac.vars$name%in%c("c645a", "c804","kz021", "mz010", "sex")&alspac.vars$cat1!="Current"),]

raw_dat <- extractVars(alspac.vars)
alspac.table <- raw_dat

# From here on I assume alspac.table contains the relevant data: one row per mother and 
# the columns aln and food freq variables c200-c373 listed above

# NOTE: I only have aln as the id in my dataset. If this is not the 
# correct identifier variable, you will need to change 
# keep_vars at line 197 and the sample.id for food.gp at line 313

# Check variable names: 
colnames(alspac.table) <- tolower(colnames(alspac.table))
names(alspac.table)


#########################################
# Missing Data
#########################################

# Missing data are negative values

# set missings to 0 for these variables
# (ie if missing assume not consumed at all)
vars=c(
"c200","c201","c202","c203","c204","c205","c206","c207","c208","c209",
"c210","c211","c215","c216","c217","c218","c219","c220","c221","c222",
"c223","c224","c225","c226","c227","c228","c229","c230","c231","c232",
"c233","c234","c235","c236","c237","c238","c239","c240","c241","c242",
"c243","c244","c245","c246","c247","c251","c252","c253","c254","c255",
"c256","c260","c261","c263","c264","c265","c266","c267","c268","c269",
"c270","c271","c272","c273","c274","c276","c277","c278","c279","c280",
"c281","c282","c283","c284","c285","c286","c287","c288","c289","c300",
"c301","c305","c306","c310","c373")

# replace missing values with 0

alspac.table[,vars]<-replace(alspac.table[,vars],alspac.table[,vars]<0,0)

#########################################
# Extract the relevant variables from ALSPAC data 
# and set up food.data data frame for storing the 
# interim variables - this keeps the orginal data 
# untouched.
#
# Food.data consists of frequency of eating variables
# recoded from cateogrical to midpoints, plus a couple 
# of interim calculations 
#

# ALSPAC variables to keep
keep_vars<-c("aln",
    "c200","c201", "c202", "c203", "c204", 
    "c205","c206", "c207", "c209","c210", 
    "c217","c218", "c220", "c222", "c223", 
    "c224", "c225", "c226", "c227", "c228", 
    "c229", "c232", "c233", "c234", "c235", 
    "c237", "c240", "c241", "c243", "c250", 
    "c252", "c253", "c254", "c255", "c260",
    "c261", "c263", "c264", "c265", "c266", 
    "c267", "c268", "c269", "c270", "c271", 
    "c272", "c273", "c274", "c275", "c284", 
    "c285", "c286", "c287", "c288", "c289", 
    "c300", "c305", "c373")

food.data <- alspac.table[,keep_vars]
#########################################################
# convert all categorical frequency variables into number 
# of times eaten per week
#########################################################

# FREQ
# 1  Never or rarely    0
# 2  Once in 2WKS       0.5
# 3  1-3 times PWK      2
# 4  4-7 times PWK      5.5
# 5  >once a day        10

vars<-c("c200","c201", "c202", "c203", "c204", 
        "c205","c206", "c207", "c209", "c210", 
        "c217","c218", "c222", "c223", "c224", 
        "c225", "c226", "c227", "c228", "c229", 
        "c232", "c233", "c234", "c235", "c237", 
        "c240", "c241", "c243")

food.data[,vars]<-sapply(alspac.table[,vars], recode,
                         "1"=0,
                         "2"=0.5,
                         "3"=2,
                         "4"=5.5,
                         "5"=10)

# c250 slices of bread
# missing       1.5
# 1 <1          0
# 2 1-2         1.5
# 3 3-4         3.5
# 4 5 or more   6

food.data$c250[alspac.table$c250<0] <- 1.5
food.data$c250<-recode(alspac.table$c250, 
                         "0"=0,
                         "1"=0,
                         "2"=1.5,
                         "3"=3.5,
                         "4"=6)

# c275 slices of bread with fat 
# missing   2
# values>7  6
food.data$c275[alspac.table$c275<0] <- 2
food.data$c275[alspac.table$c275>7] <- 6

# c252-255 eat types of bread
# 1 Yes 1
# 2 No  0
food.data$c252[alspac.table$c252!=1] <- 0
food.data$c253[alspac.table$c253!=1] <- 0
food.data$c254[alspac.table$c254!=1] <- 0
food.data$c255[alspac.table$c255!=1] <- 0

#c284 c285 c287 c288 c289 - milk in tea, coffee, pudding, as drink, in milky drink
# 0 (recoded from missing) 0
# 1 Usually             1
# 2 Sometimes           0.5
# 3 Never               0
vars=c("c284","c285","c287","c288","c289") 
food.data[,vars]<-sapply(alspac.table[,vars], recode,
                         "0"=0,
                         "1"=1,
                         "2"=0.5,
                         "3"=0)

#c286 milk in cereal
# 0 (recoded from missing) 0
# 1 Usually             1
# 2 Sometimes           1
# 3 Never               0
food.data$c286 = recode(alspac.table$c286,
                        "0"=0, "1"=1, "2"=1,"3"=0)

# c269 & c270 oil on bread/for frying
# 1 Yes 1
# 2 No  0
food.data$c269[alspac.table$c269!=1] <- 0
food.data$c270[alspac.table$c270!=1] <- 0


# variables in food.data now have appropriate frequencies


###########################################                        
#Set up a data frame for the output: food.gp

# From EWAS Script:
# 'Necessary variable names are: "sample.id","vegetables",
# "legumes","fruitnuts","dairy","cereals","meat","fish",
# "oliveoil","alcohol","sex","mat.edu","mat.age",
# "mat.smoking","mat.bmi","mat.kcal","batch"
# -in that order
#
# I don't have all these variables, so food.gp contains the 
# first ten of these (ID plus nine food groups) in the correct 
# order.
# However, see (unchecked)  code at end from another file that 
# purports to be the remaining variables needed

food.gp=data.frame(aln=alspac.table$aln,
                   qlet=alspac.table$qlet,
                        vegetables=NA,
                        legumes=NA,
                        fruitnuts=NA,
                        dairy=NA,
                        cereals=NA,
                        meat=NA,
                        fish=NA,
                        oliveoil=NA,
                        alcohol=NA)

###########################################


###################### Vegetables
# Use variables: 
#   c224 leafy green    119g
#   c225 other green    132g
#   c226 carrots        60g
#   c227 other root veg 87g
#   c228 salad          80g
#   c210 pizza toppings   54.8g

food.gp$vegetables <- 119 * food.data$c224 +
                      132 * food.data$c225 +
                       60 * food.data$c226 +
                       87 * food.data$c227 +
                       80 * food.data$c228 +
                     54.8 * food.data$c210
# these are portions per week - convert to daily:
food.gp$vegetables <- food.gp$vegetables/7


###################### Legumes
# c222 baked beans 135g
# c223 legumes      74g
# c240 pulses       120g

food.gp$legumes <- 135 * food.data$c222 +
 			74 * food.data$c223 +
 			120 * food.data$c240 


# these are portions per week - convert to daily:
food.gp$legumes <- food.gp$legumes/7


###################### Fruit & Nuts
# c229 fresh fruit  118g
# c241 nuts          52g
# c243  tahini       10g

food.gp$fruitnuts <- 118 * food.data$c229 +
                      52 * food.data$c241 +
                      10 * food.data$c243

# these are portions per week - convert to daily:
food.gp$fruitnuts <- food.gp$fruitnuts/7

###################### Dairy

### milk:
# c284  in tea          30g per cup
# c300 cups/tea per day
# c285 in coffee        30g per cup
# c305 cups/coffee per day
# c286 in cereal        100g per cereal
# c233, c234 c235 - oat, bran and other cereals /week
# c287 in pudding       150g
# c232 no. puddings /week
# c288 as a drink       200g
# c289 in milky drink   260g
# no measure of frequency for these two beyond usually/sometimes -
# previous code has a factor 2/3 which I think is
# interpreting 'usually have a milk drink' as on 2 in 3 days
# 'sometimes' has a value of 0.5, so is thus on 1 in 3 days

# milk - do this as daily milk (so need to convert cer3al and puddings)
food.data$milk <- 30 * food.data$c284 * food.data$c300 +
        30 * food.data$c285 * food.data$c305 +
        (100 * food.data$c286 * (food.data$c233 + food.data$c234 + food.data$c235) )/7+
        (150 * food.data$c287 *food.data$c232)/7 +
        200 * food.data$c288 * 2/3 +
        260 * food.data$c289 * 2/3

### Other dairy:
# c209 cheese   46g
# c210 pizza    60.3g


# convert cheese and pizza to daily
food.gp$dairy <- (46 * food.data$c209 +
                60.3 * food.data$c210)/7 +
                food.data$milk

###################### Cereals


### bread:
# c250 portions eaten per day

# Types of bread eaten
# c252 white    36g
# c253 brown    36g
# c254 wholemeal    36g
# c255 chapati    68g
# c256 Dont usually eat any bread

#vector of portion sizes for each type of bread
b.portion <- matrix(c(36, 36, 36, 68),4,1)

# matrix of bread types eaten: rows are observations, 4 cols for bread types
# matrix contains 1 if type is eaten, 0 otherwise
b.mat <- as.matrix(food.data[,c("c252","c253","c254","c255")] )
# count number of different types of bread eaten
n.btype <- apply(b.mat ,1,sum,na.rm=T)

# scale bread matrix to be proportions  - 
# so if eat two types of bread, then half for each
# (avoid dividing by zero if there are none)
b.mat[n.btype>0,] <- b.mat[n.btype>0,]/n.btype[n.btype>0]

# calculate total portion size for a person, given the breads they eat
food.data$bread <- b.mat %*% b.portion 

# and calculate daily amount from portinos eaten per day
food.data$bread <- food.data$bread * food.data$c250

#### Other cereals

# c217 rice     135g
# c218 pasta    150g

# c210 pizza    32.1g

# c233 oat cereals      80g
# c234 bran cereals     36g
# c235 other cereals    33g

# c237 crispbread   20g


food.gp$cereals <- 135 * food.data$c217 +
                   150 * food.data$c218 +
                    32.1 * food.data$c210 +
                    80 * food.data$c233 +
                    36 * food.data$c234 +
                    33 * food.data$c235 +
                    20 * food.data$c237

#convert to daily
food.gp$cereals <- food.gp$cereals/7

# add daily bread
food.gp$cereals <- food.gp$cereals + food.data$bread


###################### Meat
# c200 sausages/burgers 80g
# c201 pies/pasties    129g
# c202 red meat         88g
# c203 poultry          100g
# c204 offal            98g

# amount of fat - not relevant


food.gp$meat <-  80 * food.data$c200 +
                129 * food.data$c201 +
                 88 * food.data$c202 +
                100 * food.data$c203 +
                 98 * food.data$c204

# these are portions per week - convert to daily:
food.gp$meat <- food.gp$meat/7


###################### Fish

# c205 white fish   130g
# c206 oily fish    119g
# c207 shellfish    89g

food.gp$fish <- 130 * food.data$c205 +
                119 * food.data$c206 +
                 89 * food.data$c207

# these are portions per week - convert to daily:
food.gp$fish <- food.gp$fish/7

###################### Olive Oil

# c269 oil on bread         10g
# c270 oil for frying       10g

# ignore oil on bread

#oil for frying:
# sunflower/soya/corn/olive oil
# Assume these occur in the proportions 30/30/30/10
# - so 1/10 of oil is assumed to be olive oil

# c261 butter for frying
# c264 marg for frying
# c266 polyunsat marg for frying
# c268 low fat spread for frying
# c270 sun/olive/etc oil for frying   <==========
# c272 other oil for frying
# c274 other fat for frying
#  - assume these are used for frying proportionately

# c220 freq eat fried food

# matrix of frying fats used: rows are observations, 7 cols for fat types
# matrix contains 1 if fat is used, 0 otherwise
f.mat <- as.matrix(food.data[,c("c261","c264","c266","c268", "c270", "c272", "c274")] )
# count number of different types of fat used
n.ftype <- apply(f.mat ,1,sum)
# scale fat matrix to be proportions  - 
f.mat[which(n.ftype>0),] <- f.mat[which(n.ftype>0),]/n.ftype[which(n.ftype>0)]

#only interested in the oil proportion:
oil.p <- f.mat[,5]

#assume 1/10 of this is olive oil:
oil.p <- oil.p/10

# calculate  total olive oil as 10g per portion each fried food occasion:
food.gp$oliveoil <- 10 * oil.p * food.data$c220

###################### Alcohol

# c373 alcohol units/wk 8g/unit

food.gp$alcohol  <- food.data$c373 * 8              
# convert to daily
food.gp$alcohol  <- food.gp$alcohol/7


############################################
# End of Med Diet calculations
############################################

# tidy up
# remove interim variables
rm(list=c("keep_vars","vars","nmiss","v",
     "b.mat","n.btype","b.portion",
     "f.mat","n.ftype","oil.p"))

############################################
# Additional code

pheno <- food.gp
pheno$mat.kcal <- alspac.table$c3804
pheno$mat.kcal[pheno$mat.kcal<0]<-NA
pheno$mat.kcal <- pheno$mat.kcal/4.184 # convert kj to kcal

################
# Construct food group tertiles
# Please comment out (using # at beginning of each line) if your cohort does not have sufficient variation (not a normal distribution) in consumption of a food item (e.g. legumes or olive oil)
# In that case, please see the next step of constructing alternative cut-offs
################

# first check the distribution of food items
require(ggplot2)
df <- melt(pheno[,-c(1:2)])
fooditems_hist <- ggplot(df,aes(value))+
  geom_histogram()+
  facet_wrap(.~variable,scales="free")
fooditems_summary <- df %>% group_by(variable) %>%
  summarise(mean=mean(value,na.rm=T),sd=sd(value,na.rm=T),median=median(value,na.rm=T),lowerQ=quantile(value,0.25,na.rm=T),upperQ=quantile(value,0.75,na.rm=T)) %>%
  kable()

# Now generate tertiles

# Vegetables
pheno$veg1000kcal <- (pheno$vegetables / pheno$mat.kcal * 1000)
pheno$vegrmed <- with(pheno, factor(findInterval(pheno$veg1000kcal, c(-Inf, quantile(pheno$veg1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
pheno$vegrmed <- revalue(as.character(pheno$vegrmed), c("Q1"="0", "Q2"="1", "Q3"="2"))
pheno$vegrmed <- as.numeric(pheno$vegrmed)

# Legumes
pheno$leg1000kcal <- (pheno$legumes / pheno$mat.kcal * 1000)
pheno$legrmed <- with(pheno, factor(findInterval(pheno$leg1000kcal, c(-Inf, quantile(pheno$leg1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
pheno$legrmed <- revalue(as.character(pheno$legrmed), c("Q1"="0", "Q2"="1", "Q3"="2"))
pheno$legrmed <- as.numeric(pheno$legrmed)

# Fruit incl nuts	
pheno$fruit1000kcal <- (pheno$fruitnuts / pheno$mat.kcal * 1000)
pheno$fruitrmed <- with(pheno, factor(findInterval(pheno$fruit1000kcal, c(-Inf, quantile(pheno$fruit1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
pheno$fruitrmed <- revalue(as.character(pheno$fruitrmed), c("Q1"="0", "Q2"="1", "Q3"="2"))
pheno$fruitrmed <- as.numeric(pheno$fruitrmed)

# Fish and seafood								
pheno$fish1000kcal <- (pheno$fish / pheno$mat.kcal * 1000)
pheno$fishrmed <- with(pheno, factor(findInterval(pheno$fish1000kcal, c(-Inf, quantile(pheno$fish1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
pheno$fishrmed <- revalue(as.character(pheno$fishrmed), c("Q1"="0", "Q2"="1", "Q3"="2"))
pheno$fishrmed <- as.numeric(pheno$fishrmed)

# Cereals								
pheno$cereal1000kcal <- (pheno$cereals / pheno$mat.kcal * 1000)
pheno$cerealrmed <- with(pheno, factor(findInterval(pheno$cereal1000kcal, c(-Inf, quantile(pheno$cereal1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
pheno$cerealrmed <- revalue(as.character(pheno$cerealrmed), c("Q1"="0", "Q2"="1", "Q3"="2"))
pheno$cerealrmed <- as.numeric(pheno$cerealrmed)

# Meat								
pheno$meat1000kcal <- (pheno$meat / pheno$mat.kcal * 1000)
pheno$meatrmed <- with(pheno, factor(findInterval(pheno$meat1000kcal, c(-Inf, quantile(pheno$meat1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
pheno$meatrmed <- revalue(as.character(pheno$meatrmed), c("Q1"="2", "Q2"="1", "Q3"="0"))
pheno$meatrmed <- as.numeric(pheno$meatrmed)

# Dairy								
pheno$dairy1000kcal <- (pheno$dairy / pheno$mat.kcal * 1000)
pheno$dairyrmed <- with(pheno, factor(findInterval(pheno$dairy1000kcal, c(-Inf, quantile(pheno$dairy1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
pheno$dairyrmed <- revalue(as.character(pheno$dairyrmed), c("Q1"="2", "Q2"="1", "Q3"="0"))
pheno$dairyrmed <- as.numeric(pheno$dairyrmed)

# Olive oil [does not show a normal distribution in ALSPAC, so this code not used]
##pheno$oliveoil1000kcal <- (pheno$oliveoil / pheno$mat.kcal * 1000)
##pheno$oliveoilrmed <- with(pheno, factor(findInterval(pheno$oliveoil1000kcal, c(-Inf, quantile(pheno$oliveoil1000kcal, probs=c(0.33333, .66667),na.rm=T), Inf)), labels=c("Q1","Q2","Q3")))
##pheno$oliveoilrmed <- revalue(as.character(pheno$oliveoilrmed), c("Q1"="0", "Q2"="1", "Q3"="2"))
##pheno$oliveoilrmed <- as.numeric(pheno$oliveoilrmed)

# Alcohol
pheno$alcoholrmed <- 0
pheno$alcoholrmed[pheno$alcohol>=5 & pheno$alcohol<=25] <- 2

################
# Construct alternative cut-offs if >33% has same intake, e.g. >33% of women in Generation R consumes no legumes
# In this case remove this specific food group above from "Construct food group tertiles"
################
# in ALSPAC, we have to do this for oliveoil
## Please note that the first "tertile" will be those with no oliveoil consumption, since >33% had olive oil intake of zero. The script won't run when one value crosses two tertiles
pheno$oliveoil1000kcal <- (pheno$oliveoil / pheno$mat.kcal * 1000)
pheno$oliveoilrmed_notzero <- ifelse(pheno$oliveoil1000kcal == 0, NA, pheno$oliveoil1000kcal)
pheno$oliveoilrmed <- with(pheno, factor(findInterval(pheno$oliveoilrmed_notzero, c(0.0000001, quantile(pheno$oliveoilrmed_notzero, probs=c(0.5), na.rm=T), Inf)), labels=c("1","2"))) ## CHECK WITH LEANNE AS I HAD TO CHANGE 0.01 TO 0.0000001
pheno$oliveoilrmed <- as.numeric(pheno$oliveoilrmed)
pheno$oliveoilrmed[pheno$oliveoil1000kcal==0] <- 0 #because more than 33% had olive oil intake of zero. The script won't run when one value crosses two tertiles
pheno$oliveoilrmed_notzero <-NULL

## Check tertiles
df <- melt(pheno[,grep("rmed",colnames(pheno))])
tertile.table <- kable(table(df$variable, df$value))

################
# Construct rMED
################
pheno$rmed <- pheno$vegrmed + pheno$legrmed + pheno$fruitrmed + pheno$fishrmed + pheno$cerealrmed + pheno$meatrmed + pheno$dairyrmed + pheno$oliveoilrmed + pheno$alcoholrmed

################
# Construct rMED2
################
pheno$rmed2 <- pheno$vegrmed + pheno$legrmed + pheno$fruitrmed + pheno$fishrmed + pheno$cerealrmed + pheno$meatrmed + pheno$dairyrmed + pheno$oliveoilrmed

# check distribution of med diet indices
require(ggplot2)
df <- melt(pheno[,c("rmed","rmed2")])
rmed_hist <- ggplot(df,aes(value))+
  geom_histogram()+
  facet_wrap(.~variable,scales="free")
rmed_summary <- df %>% group_by(variable) %>%
  summarise(mean=mean(value,na.rm=T),sd=sd(value,na.rm=T),median=median(value,na.rm=T),lowerQ=quantile(value,0.25,na.rm=T),upperQ=quantile(value,0.75,na.rm=T)) %>%
  kable()

names(pheno) <- c("aln","qlet",paste0("diet_mother_",names(pheno)[3:ncol(pheno)],"_ever_pregnancy_continuous"))
dat <- left_join(dat,pheno,by=c("aln","qlet"),all.x=T)

```

# Child's anthropometry & adiposity {.tabset}

## Ideal variables

Ideal harmonised anthropometry and adiposity measures for offspring are:

*	Birthweight (g)
*	SGA
*	LGA
* Low BW <2500g vs >=2500g & <=4500g
* High BW > 4500g vs >=2500g & <=4500g
*	Birth length (cm)
*	Head circumference (cm)
*	Height (cm)
*	Waist circumference (cm)
*	Weight (kg)
*	BMI (kg/m2)
*	WHO categories of overweight/obese/underweight vs normal weight
*	Fat mass index

Ages are defined as:

* at birth or in first year of life (>0 to <1)
* childhood stage 1 (around age 2; >=1 to <3)
* childhood stage 2 (around age 4; >=3 to <5)
* childhood stage 3 (around age 6; >=5 to <8)
* childhood stage 4 (around age 9; >=8 to <11)

## Set up meta-data
```{r eval=FALSE, include=TRUE}
#get a list of anthropometry measures for offspring
anthropometry_meta_data <- findVars("sga", "lga", "gestational", "height", "weight", "waist", "circumference", "BMI", "obese", "underweight", "fat", "fat mass percentage")
anthropometry_meta_data_child <-anthropometry_meta_data[anthropometry_meta_data$cat3 %in% c("Child","Child Based","Obstetric"),]

#select measures up to age 11
anthropometry_meta_data_child <-anthropometry_meta_data_child[grep(lapply(anthropometry_meta_data_child$obj,substring,1,3),pattern= "f07|f08|f09|f10|f11|cif|ka|kb|kc|kd|ke|kf|kg|kh|ki|kj|kl|km|kn|ko|kp|kr|ks|kt|ku|kv|kq|kr|ks|kt|ku|kv|kw|ccs|ccb|ccc|ccd|cce|ccf|ccg|cch|cci|ccj|cck|OA|OB|OC"),]
```

**Birthweight (g)**

* Already extracted from the sample files

**SGA**

* Can been derived from the birthweight variable already extracted from the sample files + gestational age
* BW centiles for GA have been derived in C Flach, M Leese, J Heron, J Evans, G Feder, D Sharp, LM Howard. Antenatal domestic violence,
maternal mental health and subsequent child behaivour: a cohort study. BJOG 2011; 118:1383-1391. and are stored in the birthweight_centile folder in Useful data. SGA can be defined as <=10th %ile.

```{r eval=FALSE, include=TRUE}
child_anthropometry_vars <-c("bestgest","centiles")
```

**LGA**

* Can been derived from the birthweight variable already extracted from the sample files + gestational age and defined as >=90th %ile.

**Birth length (cm)**

* Already extracted from the sample files

**Head circumference (cm)**

* At birth, already extracted from the sample files
* f7ms014 Head circumference (cm): F7
* km6003 F1: Head circumference measurement 1
* km6013 F2: Head circumference measurement 2
* km6023 F3: Head circumference measurement 3
* km6033 F4: Head circumference measurement 4
* kp7003 G1a: Child's head circumference measurement number 1 - cms
* kp7008 G1b: Child's head circumference measurement number 2 - cms

```{r eval=FALSE, include=TRUE}
child_anthropometry_vars <-c(child_anthropometry_vars, "f7ms014","km6003","km6013","km6023","km6033","kp7003","kp7008")
```

**Height (cm)**

* cf054 Height at 25 mth
* cf055 Height at 31 mth
* cf056 Height at 37 mth
* cf057 Height at 43 mth
* cf058 Height at 49 mth
* cf059 Height at 61 mth
* f7ms010 Height (cm): F7
* f8lf020	Child height (cm): lung function: F8
* f9ms010 Height (cm): F9
* fdms010 Height (cm): F10
* kn7000	G1a: Height (cm)
* km6002 F1: Height measurement 1
* km6012 F2: Height measurement 2
* km6022 F3: Height measurement 3
* km6032 F4: Height measurement 4
* kp7002	G1a: Child's height measurement number 1 - cms
* kp7007	G1b: Child's height measurement number 2 - cms
* ku783	H1a: Metric height 1 calculated of child - cms
* ku788	H1b: Metric height 2 calculated of child - cms	
* ku793	H1c: Metric height 3 calculated of child - cms
* ku798	H1d: Metric height 4 calculated of child - cms	

```{r eval=FALSE, include=TRUE}
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf054", "cf055", "cf056", "cf057", "cf058", "cf059", "f7ms010","f8lf020", "f9ms010","fdms010","kn7000","km6002","km6012","km6022","km6032","kp7002","kp7007","ku783","ku788","ku793","ku798"))
```

**Waist circumference (cm)**

* cf075 Waist circ at 31 mth
* cf076 Waist circ at 37 mth
* cf077 Waist circ at 43 mth
* cf078 Waist circ at 49 mth
* cf079 Waist circ at 61 mth
* f7ms018 Waist circumference (cm): F7
* f9ms018	Waist circumference (cm): F9
* fdms018 Waist circumference (cm): F10
* kn7003 G1d: Waist (cm)

```{r eval=FALSE, include=TRUE}
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf075","cf076","cf077","cf078","cf079","f7ms018","f9ms018","fdms018","kn7003"))
```

**Weight (kg)**

* cf040 Weight at 4 mth
* cf041 Weight at 8 mth
* cf042 Weight at 12 mth
* cf043 Weight at 18 mth
* cf044 Weight at 25 mth
* cf045 Weight at 31 mth
* cf046 Weight at 37 mth
* cf047 Weight at 43 mth
* cf048 Weight at 49 mth
* cf049	Weight at 61 mth
* f7ms026 Weight (kg): F7
* f8lf021	Child weight (kg): lung function: F8
* f9ms026 Weight (kg): F9
* fdms026 Weight (kg): F10
* km6001 F1: Weight measurement 1
* km6011 F2: Weight measurement 2
* km6021 F3: Weight measurement 3
* km6031 F4: Weight measurement 4
* kn7001 G1b: Weight (kg)
* kp7001	G1a: Child's weight measurement number 1 - kgs
* kp7006 G1b: Child's weight measurement number 2 - kgs
* ku782	H1a: Metric weight 1 calculated of child - kgs
* ku787	H1b: Metric weight 2 calculated of child - kgs
* ku792	H1c: Metric weight 3 calculated of child - kgs
* ku797	H1d: Metric weight 4 calculated of child - kgs

```{r eval=FALSE, include=TRUE}
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf040","cf041","cf042","cf043","cf044","cf045","cf046","cf047","cf048","cf049","f7ms026","f8lf021","f9ms026","fdms026","km6001","km6011","km6021","km6031","kn7001","kp7001","kp7006","ku782","ku787","ku792","ku797"))
```

**BMI (kg/m2)**

* cf060 BMI at 4 mth
* cf061 BMI at 8 mth
* cf062 BMI at 12 mth
* cf063 BMI at 18 mth
* cf064 BMI at 25 mth
* cf065 BMI at 31 mth
* cf066 BMI at 37 mth
* cf067 BMI at 43 mth
* cf068 BMI at 49 mth
* cf069 BMI at 61 mth
* f7ms026a BMI: F7
* f9ms026a BMI: F9
* fdms026a BMI: F10

```{r eval=FALSE, include=TRUE}
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf060","cf061","cf062","cf063","cf064","cf065","cf066","cf067","cf068","cf069","f7ms026a","f9ms026a","fdms026a"))
```

**WHO categories of obese/underweight vs normal weight**

* Can be derived from BMI variables

**Fat mass percentage**

* f9dx135	Total Body - fat mass (g): F9

```{r eval=FALSE, include=TRUE}
child_anthropometry_vars <-c(child_anthropometry_vars,c("f9dx135"))
```

### Set up final meta data object
```{r eval=FALSE, include=TRUE}
anthropometry_meta_data <- findVars(unique(child_anthropometry_vars)) #create a table with all the necessary meta data for each variable

anthropometry_meta_data <- subset(anthropometry_meta_data,subset=tolower(name) %in% child_anthropometry_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings

#Some of the variables (bestgest, cf046, cf056, cf066, cf076) have the same name in multiple files, so this line removes the duplicates:
anthropometry_meta_data <- anthropometry_meta_data[which(anthropometry_meta_data$cat2 %in% c("Clinic","Quest","bestgest","birthweight_centile")),]

```

## Extract variables
```{r eval=FALSE, include=TRUE}
child_anthropometry_data <- extractVars(anthropometry_meta_data[anthropometry_meta_data$name %in% unique(child_anthropometry_vars),]) #extracting individual level data and performing withdrawal of consent

child_anthropometry_data<-child_anthropometry_data[,c("aln","qlet",child_anthropometry_vars)]
dat <- left_join(dat,child_anthropometry_data,by=c("aln","qlet")) 
```

## Derive clean variables

**Birthweight**

High BW > 4500g vs >=2500g & <=4500g 
```{r eval=FALSE, include=TRUE}
dat$anthro_birthweight_high_binary <- NA
dat$anthro_birthweight_high_binary[dat$anthro_birthweight>4500]<-1
dat$anthro_birthweight_high_binary[dat$anthro_birthweight>=2500&dat$anthro_birthweight<=4500]<-0
```
Low BW <2500g vs >=2500g & <=4500g
```{r eval=FALSE, include=TRUE}
dat$anthro_birthweight_low_binary <- NA
dat$anthro_birthweight_low_binary[dat$anthro_birthweight<2500]<-1
dat$anthro_birthweight_low_binary[dat$anthro_birthweight>=2500&dat$anthro_birthweight<=4500]<-0
```

SGA and LGA can be derived from the centile data in Useful Data.
```{r eval=FALSE, include=TRUE}
dat$anthro_birthweight_sga_binary <- NA
dat$anthro_birthweight_sga_binary[dat$centiles<=10]<-1
dat$anthro_birthweight_sga_binary[dat$centiles>10]<-0

dat$anthro_birthweight_lga_binary <- NA
dat$anthro_birthweight_lga_binary[dat$centiles>=90]<-1
dat$anthro_birthweight_lga_binary[dat$centiles<90]<-0
```

**Head circumference** at birth already extracted and cleaned.

Head circumference at other ages - there are a few carer-reported measures in the km and kp files, but it's unclear why there are multiple measurements and anyway there is likely to be a lot of measurement error in these, so we will stick to HC measured at the F7 clinic (stage 3):
```{r eval=FALSE, include=TRUE}
dat$anthro_head_circ_stage3 <-dat$f7ms014
dat$anthro_head_circ_stage3[dat$anthro_head_circ_stage3<0]<-NA
dat$anthro_head_circ_stage3_zscore <- scale(dat$anthro_head_circ_stage3)
```

**Height** - there are lots of measures, but we will prioritise clinic measures of parental report of child's height

Height at stage 1 (12 months to 35 months) - take the latest timepoint first, then supplement with earlier timepoints if NA
```{r eval=FALSE, include=TRUE}
temp31<-dat$cf055
temp31[temp31<0]<-NA
temp25<-dat$cf054
temp25[temp25<0]<-NA
timepoint <-ifelse(is.na(temp31),NA,"31")
temp3125 <-temp31
temp3125[is.na(timepoint)]<-temp25[is.na(timepoint)]
timepoint[is.na(timepoint)&(is.na(temp25)==FALSE)]<-"25"
dat$stage1_timepoint <-timepoint

dat$anthro_height_stage1 <- temp3125
dat$anthro_height_stage1_zscore <- scale(dat$anthro_height_stage1)
```

Height at stage 2 (>=3 <5y; 36 to 60 months) - take the latest timepoint first, then supplement with earlier timepoints if NA
```{r eval=FALSE, include=TRUE}
temp37<-dat$cf056
temp37[temp37<0]<-NA
temp43<-dat$cf057
temp43[temp43<0]<-NA
temp49<-dat$cf058
temp49[temp49<0]<-NA
timepoint <-ifelse(is.na(temp49),NA,"49")
temp494337 <-temp49 # take latest timepoint (49 m)
temp494337[is.na(timepoint)]<-temp43[is.na(timepoint)] #if NA, use 43 m
timepoint[is.na(timepoint)&(is.na(temp43)==FALSE)]<-"43"
temp494337[is.na(timepoint)]<-temp37[is.na(timepoint)] #if both 49 and 43 NA, use 37 m
timepoint[is.na(timepoint)&(is.na(temp37)==FALSE)]<-"37"
dat$stage2_timepoint <-timepoint

dat$anthro_height_stage2 <- temp494337
dat$anthro_height_stage2_zscore <- scale(dat$anthro_height_stage2)
```

Height at stage 3 (take age 7 clinic). The earliest possible file we could use is CIF at 61 months (5y 1m), but this is only in about 1000 children. F7 is the earliest complete clinic. There is a clinic at F8, but this is too late for this timepoint
```{r eval=FALSE, include=TRUE}
dat$anthro_height_stage3<-dat$f7ms010
dat$anthro_height_stage3[dat$anthro_height_stage3<0]<-NA
dat$anthro_height_stage3_zscore <- scale(dat$anthro_height_stage3)
```

Height at stage 4 - there are three clinics (8, 9 & 10) and we could do as above for stage 1 and 3 by taking F10 and then filling in missing values with F9 then F8, but then we would be in a potentially tricky situation later when we are calculating fat mass index and need height JUST at F9. So I made the decision to just use F9, which is in the middle of the age category and should suit our purposes fine.
```{r eval=FALSE, include=TRUE}
dat$anthro_height_stage4<-dat$fdms010
dat$anthro_height_stage4[dat$anthro_height_stage4<0]<-NA
dat$anthro_height_stage4_zscore <- scale(dat$anthro_height_stage4)
```

**Waist circumference** 

WC at stage 1 (12 months to 35 months) - only measured at 31
```{r eval=FALSE, include=TRUE}
dat$anthro_waist_stage1<-dat$cf075
dat$anthro_waist_stage1[dat$anthro_waist_stage1<0]<-NA
dat$anthro_waist_stage1_zscore <- scale(dat$anthro_waist_stage1)
```

WC at stage 2 (>=3 <5y; 36 to 60 months) - take measurement at timepoint used for the height measurement above (else different timepoints will be used for different outcomes and this will get confusing and overly clunky)
```{r eval=FALSE, include=TRUE}
dat$anthro_waist_stage2 <- NA
dat$anthro_waist_stage2[which(dat$stage2_timepoint=="37")] <-dat$cf076[which(dat$stage2_timepoint=="37")]
dat$anthro_waist_stage2[which(dat$stage2_timepoint=="43")] <-dat$cf077[which(dat$stage2_timepoint=="43")]
dat$anthro_waist_stage2[which(dat$stage2_timepoint=="49")] <-dat$cf078[which(dat$stage2_timepoint=="49")]
dat$anthro_waist_stage2[dat$anthro_waist_stage2<0]<-NA
dat$anthro_waist_stage2_zscore <- scale(dat$anthro_waist_stage2)
```

WC at stage 3 (take age 7 clinic). The earliest possible file we could use is CIF at 61 months (5y 1m), but this is only in about 1000 children. F7 is the earliest complete clinic. There is a clinic at F8, but this is too late for this timepoint
```{r eval=FALSE, include=TRUE}
dat$anthro_waist_stage3<-dat$f7ms018
dat$anthro_waist_stage3[dat$anthro_waist_stage3<0]<-NA
dat$anthro_waist_stage3_zscore <- scale(dat$anthro_waist_stage3)
```

WC at stage 4 (use age 9 clinic)
```{r eval=FALSE, include=TRUE}
dat$anthro_waist_stage4<-dat$f9ms018
dat$anthro_waist_stage4[dat$anthro_waist_stage4<0]<-NA
dat$anthro_waist_stage4_zscore <- scale(dat$anthro_waist_stage4)
```

**Weight ** 

Weight at stage 1 (12 months to 35 months) - measured at 25 and 31m
```{r eval=FALSE, include=TRUE}
dat$anthro_weight_stage1 <- NA
dat$anthro_weight_stage1[which(dat$stage1_timepoint=="31")] <-dat$cf045[which(dat$stage1_timepoint=="31")]
dat$anthro_weight_stage1[which(dat$stage1_timepoint=="25")] <-dat$cf044[which(dat$stage1_timepoint=="25")]
dat$anthro_weight_stage1[dat$anthro_weight_stage1<0]<-NA
dat$anthro_weight_stage1_zscore <- scale(dat$anthro_weight_stage1)
```

Weight at stage 2 (>=3 <5y; 36 to 60 months) - take measurement at timepoint used for the height measurement above (else different timepoints will be used for different outcomes and this will get confusing and overly clunky)
```{r eval=FALSE, include=TRUE}
dat$anthro_weight_stage2 <- NA
dat$anthro_weight_stage2[which(dat$stage2_timepoint=="37")] <-dat$cf046[which(dat$stage2_timepoint=="37")]
dat$anthro_weight_stage2[which(dat$stage2_timepoint=="43")] <-dat$cf047[which(dat$stage2_timepoint=="43")]
dat$anthro_weight_stage2[which(dat$stage2_timepoint=="49")] <-dat$cf048[which(dat$stage2_timepoint=="49")]
dat$anthro_weight_stage2[dat$anthro_weight_stage2<0]<-NA
dat$anthro_weight_stage2_zscore <- scale(dat$anthro_weight_stage2)
```

Weight at stage 3 (take age 7 clinic). The earliest possible file we could use is CIF at 61 months (5y 1m), but this is only in about 1000 children. F7 is the earliest complete clinic. There is a clinic at F8, but this is too late for this timepoint
```{r eval=FALSE, include=TRUE}
dat$anthro_weight_stage3<-dat$f7ms026
dat$anthro_weight_stage3[dat$anthro_weight_stage3<0]<-NA
dat$anthro_weight_stage3_zscore <- scale(dat$anthro_weight_stage3)
```

Weight at stage 4 (use age 9 clinic)
```{r eval=FALSE, include=TRUE}
dat$anthro_weight_stage4<-dat$f9ms026
dat$anthro_weight_stage4[dat$anthro_weight_stage4<0]<-NA
dat$anthro_weight_stage4_zscore <- scale(dat$anthro_weight_stage4)
```

**Age in months at clinics ** 

Age at months at the 31m, 7y, 1, 9y, 0y clinics
```{r eval=FALSE, include=TRUE}
cif <-read_dta("/Volumes/ALSPAC-Data/Current/Clinic/Child/cif_8b.dta")
f7 <-read_dta("/Volumes/ALSPAC-Data/Current/Clinic/Child/f07_5a.dta")
f9 <-read_dta("/Volumes/ALSPAC-Data/Current/Clinic/Child/f09_4c.dta")

ages <- merge(cif[,c("aln","qlet","cf014","cf015", "cf016", "cf017","cf018")],f7[,c("aln","qlet","f7003c")],by=c("aln","qlet"),all=T)
ages <- merge(ages,f9[,c("aln","qlet","f9003c")],by=c("aln","qlet"),all=T)

#if using the shortcut, the above lines aren't necessary, but you need to do these two lines instead:
ages <- dat[,c("aln","qlet","cf014","cf015", "cf016", "cf017","cf018","f7003c","f9003c")]
dat <- dat[,-which(names(dat)%in% c("cf014","cf015", "cf016", "cf017","cf018","f7003c","f9003c"))]

ages$cf015[ages$cf015<0]<-NA
ages$cf014[ages$cf014<0]<-NA
ages$cf016[ages$cf016<0]<-NA
ages$cf017[ages$cf018<0]<-NA
ages$cf017[ages$cf018<0]<-NA
ages$cf015<-ages$cf015/4 #because age at CIF clinic is in weeks
ages$cf014<-ages$cf014/4 #because age at CIF clinic is in weeks
ages$cf016<-ages$cf016/4 #because age at CIF clinic is in weeks
ages$cf017<-ages$cf017/4 #because age at CIF clinic is in weeks
ages$cf018<-ages$cf018/4 #because age at CIF clinic is in weeks

dat <- left_join(dat,ages,by=c("aln","qlet")) 

dat$covs_age_child_stage1_cif <- NA
dat$covs_age_child_stage1_cif[which(dat$stage1_timepoint=="25")] <-dat$cf014[which(dat$stage1_timepoint=="25")]
dat$covs_age_child_stage1_cif[which(dat$stage1_timepoint=="31")] <-dat$cf015[which(dat$stage1_timepoint=="31")]

dat$covs_age_child_stage2_cif <- NA
dat$covs_age_child_stage2_cif[which(dat$stage2_timepoint=="37")] <-dat$cf016[which(dat$stage2_timepoint=="37")]
dat$covs_age_child_stage2_cif[which(dat$stage2_timepoint=="43")] <-dat$cf017[which(dat$stage2_timepoint=="43")]
dat$covs_age_child_stage2_cif[which(dat$stage2_timepoint=="49")] <-dat$cf018[which(dat$stage2_timepoint=="49")]

dat$covs_age_child_stage3_f7 <- dat$f7003c
dat$covs_age_child_stage4_f9 <- dat$f9003c

dat<-dat[,-which(names(dat)%in%c("f9003c","f7003c"))]

```

**BMI ** 

Calculated from height and weight (kg/m2). No need to calculate Z scores, as we calculate age and sex adjusted sds scores in the next step.

Stage 1
```{r eval=FALSE, include=TRUE}
dat$anthro_bmi_stage1 <- dat$anthro_weight_stage1 / ((dat$anthro_height_stage1/100)^2)
```

Stage 2
```{r eval=FALSE, include=TRUE}
dat$anthro_bmi_stage2 <- dat$anthro_weight_stage2 / ((dat$anthro_height_stage2/100)^2)
```

Stage 3
```{r eval=FALSE, include=TRUE}
dat$anthro_bmi_stage3 <- dat$anthro_weight_stage3 / ((dat$anthro_height_stage3/100)^2)
```

Stage 4
```{r eval=FALSE, include=TRUE}
dat$anthro_bmi_stage4 <- dat$anthro_weight_stage4 / ((dat$anthro_height_stage4/100)^2)
```

**WHO categories of overweight and obesity**

First, we use the childsds package to calculate standard deviation scores (sds) for BMI-for-age and sex according to the WHO UK reference panel (https://www.bmj.com/content/340/bmj.c1140)

Stage 1
```{r eval=FALSE, include=TRUE}
library(childsds)
dat$anthro_bmi_stage1_sds <- sds(value = dat$anthro_bmi_stage1,
		      age = dat$covs_age_child_stage1_cif/12,
		      sex = dat$covs_sex, male = "male", female = "female",
		      ref = ukwho.ref, item = "bmi")
```

Stage 2
```{r eval=FALSE, include=TRUE}
dat$anthro_bmi_stage2_sds <- sds(value = dat$anthro_bmi_stage2,
		      age = dat$covs_age_child_stage2_cif/12,
		      sex = dat$covs_sex, male = "male", female = "female",
		      ref = ukwho.ref, item = "bmi")
```

Stage 3
```{r eval=FALSE, include=TRUE}
dat$anthro_bmi_stage3_sds <- sds(value = dat$anthro_bmi_stage3,
		      age = dat$covs_age_child_stage3_f7/12,
		      sex = dat$covs_sex, male = "male", female = "female",
		      ref = ukwho.ref, item = "bmi")
```

Stage 4
```{r eval=FALSE, include=TRUE}
dat$anthro_bmi_stage4_sds <- sds(value = dat$anthro_bmi_stage4,
		      age = dat$covs_age_child_stage4_f9/12,
		      sex = dat$covs_sex, male = "male", female = "female",
		      ref = ukwho.ref, item = "bmi")
```

Then we can define categories using WHO cut-offs (https://www.who.int/news-room/fact-sheets/detail/obesity-and-overweight)

* underweight <-2SD
* overweight >+1SD
* obese >+2SD

Variables comparing overweight/obese to normal, and obese to normal

Stage 1
```{r eval=FALSE, include=TRUE}
dat$anthro_overweightobese_stage1_binary <- NA
dat$anthro_overweightobese_stage1_binary[dat$anthro_bmi_stage1_sds>1] <-1
dat$anthro_overweightobese_stage1_binary[which(dat$anthro_bmi_stage1_sds>=(-2) &dat$anthro_bmi_stage1_sds<=1)] <-0

dat$anthro_obese_stage1_binary <- NA
dat$anthro_obese_stage1_binary[which(dat$anthro_bmi_stage1_sds>2)] <-1
dat$anthro_obese_stage1_binary[which(dat$anthro_bmi_stage1_sds<=1 &dat$anthro_bmi_stage1_sds>=(-2))] <-0
```

Stage 2
```{r eval=FALSE, include=TRUE}
dat$anthro_overweightobese_stage2_binary <- NA
dat$anthro_overweightobese_stage2_binary[dat$anthro_bmi_stage2_sds>1] <-1
dat$anthro_overweightobese_stage2_binary[which(dat$anthro_bmi_stage2_sds>=(-2) &dat$anthro_bmi_stage2_sds<=1)] <-0

dat$anthro_obese_stage2_binary <- NA
dat$anthro_obese_stage2_binary[which(dat$anthro_bmi_stage2_sds>2)] <-1
dat$anthro_obese_stage2_binary[which(dat$anthro_bmi_stage2_sds<=1 &dat$anthro_bmi_stage2_sds>=(-2))] <-0
```

Stage 3
```{r eval=FALSE, include=TRUE}
dat$anthro_overweightobese_stage3_binary <- NA
dat$anthro_overweightobese_stage3_binary[dat$anthro_bmi_stage3_sds>1] <-1
dat$anthro_overweightobese_stage3_binary[which(dat$anthro_bmi_stage3_sds>=(-2) &dat$anthro_bmi_stage3_sds<=1)] <-0

dat$anthro_obese_stage3_binary <- NA
dat$anthro_obese_stage3_binary[which(dat$anthro_bmi_stage3_sds>2)] <-1
dat$anthro_obese_stage3_binary[which(dat$anthro_bmi_stage3_sds<=1 &dat$anthro_bmi_stage3_sds>=(-2))] <-0
```

Stage 4
```{r eval=FALSE, include=TRUE}
dat$anthro_overweightobese_stage4_binary <- NA
dat$anthro_overweightobese_stage4_binary[dat$anthro_bmi_stage4_sds>1] <-1
dat$anthro_overweightobese_stage4_binary[which(dat$anthro_bmi_stage4_sds>=(-2) &dat$anthro_bmi_stage4_sds<=1)] <-0

dat$anthro_obese_stage4_binary <- NA
dat$anthro_obese_stage4_binary[which(dat$anthro_bmi_stage4_sds>2)] <-1
dat$anthro_obese_stage4_binary[which(dat$anthro_bmi_stage4_sds<=1 &dat$anthro_bmi_stage4_sds>=(-2))] <-0
```

**Fat mass index **

FMI at stage 4 (age 9): fat mass index is calcualted as fat mass in kg / height^2

```{r eval=FALSE, include=TRUE}
dat$anthro_fmi_stage4 <- dat$f9dx135
dat$anthro_fmi_stage4[dat$f9dx135<0]<-NA
dat$anthro_fmi_stage4 <- dat$anthro_fmi_stage4 / dat$anthro_height_stage4
dat$anthro_fmi_stage4_zscore <- scale(dat$anthro_fmi_stage4)
```

## Remove original variables
```{r eval=FALSE, include=TRUE}
dat<- dat[,-which(names(dat) %in% names(ages)[3:nrow(ages)])]
dat<-dat[,-which(colnames(dat)%in%child_anthropometry_vars)]
```

# Child's psychosocial & cognitive outcomes  {.tabset}

## Ideal variables

Ideal harmonised psychosocial and cognitive measures for offspring are:

*	Total behavioural difficulties (SDQ) - maternal report, continuous and dichotomised at >=17 [this is a combination of 4 subscales - excluding prosocial behaviour]
* Hyperactivity (SDQ) - maternal report, continuous and dichotomised at >=17
* Emotional symptoms (SDQ) - maternal report, continuous and dichotomised at >=5
* Conduct problems (SDQ) - maternal report, continuous and dichotomised at >=4
* Peer problems (SDQ) - maternal report, continuous and dichotomised at >=4
* Prosocial behaviour (SDQ)  - maternal report, continuous and dichotomised at <=6
* Internalising problems (SDQ) - maternal report, combination of emotional symptoms and peer problems
* Externalising problems (SDQ) - maternal report, combination of hyperactivity and conduct problems

* SCDC (Skuse Social Communication Score)

* Depressive symptoms (Moods and Feelings questionnaire - MFQ)

*	IQ and/or other or composite measures of intelligence/cognitive performance
*	Educational attainment

Ages are defined as:

* at birth or in first year of life (>0 to <1)
* childhood stage 1 (around age 2; >=1 to <3)
* childhood stage 2 (around age 4; >=3 to <5)
* childhood stage 3 (around age 6; >=5 to <8)
* childhood stage 4 (around age 9; >=8 to <11)

## Set up meta-data
```{r eval=FALSE, include=TRUE}
#get a list of neurodevelopment and cognition measures for offspring
neuro_meta_data <- findVars("internalising", "externalising", "sdq","depression","anxiety", "depressed","anxious", "adhd","attention","conduct","odd","disorder","behavioural","psychosis","eating disorder","ptsd","traumatic","suicide", "self harm","autism", "asd", "intelligence", "IQ", "education","emotional","hyperactivity","peer","prosocial","mfq","education","language","scdc","skuse","mood")
```

**SDQ**

* j557a - emotional symptoms prorated (4y/o)
* j557b - conduct problems prorated (4y/o)
* j557c - hyperactivity prorated (4y/o)
* j557d - peer problems prorated (4y/o)
* j557e - prosocial prorated (4y/o)
* j557f - total difficulties prorated (4y/o)
* kq348a - prosocial prorated (6y/o)
* kq348b - hyperactivity (6y/o)
* kq348c - emotional (6y/o)
* kq348d - conduct (6y/o)
* kq348e - peer problems (6y/o)
* kq348f - total (6y/o)
* ku705b - prosocial
* ku706b - hyperactivity
* ku707b - emotional 
* ku708b - conduct
* ku709b - peer problems
* ku710b - total

**SCDC**

* kr554b

**Depressive symptoms - MFQ**

* fddp130	Depression Score: F10 (mfq)

**IQ and/or other or composite measures of intelligence/cognitive performance**

* cf811	Performance IQ Wppsi 49 mth
* cf812	Verbal IQ Wppsi 49 mth
* cf813	Fullscale IQ Wppsi 49 mth
* f8ws110	WISC - Verbal IQ: F8
* f8ws111	WISC - Performance IQ: F8
* f8ws112	WISC - Total IQ: F8
* f8ws115	WISC - Categorical Total IQ: F8

**Educational attainment**

* Available on request from exec. We need data on Foundation, KS1 and KS2 national assessments/SATS

### Prepare data
```{r eval=FALSE, include=TRUE}
child_neuro_vars <-c("j557a","j557b","j557c","j557d","j557e","j557f","kq348a","kq348b","kq348c","kq348d","kq348e","kq348f","ku705b","ku706b","ku707b","ku708b","ku709b","ku710b","kr554a","kr554b","cf811","cf812","cf813","f8ws110","f8ws111","f8ws112","f8ws115","fddp130")

#Plus the age of the children at each timepoint
child_neuro_vars <-c(child_neuro_vars,"j914","kq998a","ku991a","kr991a","f8003c","fd003c","cf018")
```

### Set up final meta data object
```{r eval=FALSE, include=TRUE}
neuro_meta_data <- findVars(unique(child_neuro_vars)) #create a table with all the necessary meta data for each variable

neuro_meta_data <- subset(neuro_meta_data,subset=tolower(name) %in% child_neuro_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
```{r eval=FALSE, include=TRUE}
neuro_data <- extractVars(neuro_meta_data[neuro_meta_data$name %in% unique(child_neuro_vars),]) #extracting individual level data and performing withdrawal of consent

neuro_data<-neuro_data[,c("aln","qlet",child_neuro_vars)]

dat <- left_join(dat,neuro_data,by=c("aln","qlet")) 
```

## Derive clean variables

**Depression**

F10 clinic - Moods and Feelings Questionnaire (MFQ) - posting exercise

The children were given a series of envelopes with statements written on them about how they might have been feeling or acting in the previous two weeks.
The statements have been taken from the Short Mood and Feelings Questionnaire (Angold et al, 1995), which has been designed to provide a rapidly administered questionnaire for use in epidemiological studies. It correlates highly with more extensive evaluations, like the Children's Depression Inventory (Kovacs, 1983), examining a core set of symptoms. A derived depression score was created (FDDP130), by scoring the variables as follows:
True =2, Sometimes=1 Not at all=0. These variables were then summed, such that a minimum score of 0, represented no signs of depression, while a maximum score of 26 could be achieved.

Angold A, Costelo EJ, Messer SC, Pickles A, Winder F, Silver D. Development of a short questionnaire for use in epidemiological studies of depression in children and adolescents. Int J of Methods in Psychiatric Research 1995; 5: 237-249.

```{r eval=FALSE, include=TRUE}
dat$neuro_depression_stage4<-dat$fddp130
dat$neuro_depression_stage4[dat$fddp130<0]<-NA
dat$covs_age_child_stage4_f10<-dat$fd003c
dat$covs_age_child_stage4_f10[dat$fd003c<0]<-NA
dat$neuro_depression_stage4_zscore<-scale(dat$neuro_depression_stage4)
```

**SDQ**

Parent reported. The cut-offs for the SDQ are described in https://onlinelibrary.wiley.com/doi/full/10.1111/j.1469-7610.1997.tb01545.x?sid=nlm%3Apubmed 

Hyperactivity (SDQ) - maternal report, continuous and dichotomised at >=7
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_hyperactivity_stage2<-dat$j557c
dat$neuro_hyperactivity_stage2[dat$j557c<0]<-NA
dat$neuro_hyperactivity_stage2_zscore <- scale(dat$neuro_hyperactivity_stage2)
dat$neuro_hyperactivity_stage2_binary <- NA
dat$neuro_hyperactivity_stage2_binary[dat$neuro_hyperactivity_stage2>=7]<-1
dat$neuro_hyperactivity_stage2_binary[dat$neuro_hyperactivity_stage2<7]<-0
dat$covs_age_child_stage2_j<-dat$j914
dat$covs_age_child_stage2_j[dat$j914<0]<-NA
dat$covs_age_child_stage2_j <- dat$covs_age_child_stage2_j*12 #years to months
#Stage 3
dat$neuro_hyperactivity_stage3<-dat$kq348b
dat$neuro_hyperactivity_stage3[dat$kq348b<0]<-NA
dat$neuro_hyperactivity_stage3_zscore <- scale(dat$neuro_hyperactivity_stage3)
dat$neuro_hyperactivity_stage3_binary <- NA
dat$neuro_hyperactivity_stage3_binary[dat$neuro_hyperactivity_stage3>=7]<-1
dat$neuro_hyperactivity_stage3_binary[dat$neuro_hyperactivity_stage3<7]<-0
dat$covs_age_child_stage3_kq<-dat$kq998a
dat$covs_age_child_stage3_kq[dat$kq998a<0]<-NA
#Stage 4
dat$neuro_hyperactivity_stage4<-dat$ku706b
dat$neuro_hyperactivity_stage4[dat$ku706b<0]<-NA
dat$neuro_hyperactivity_stage4_zscore <- scale(dat$neuro_hyperactivity_stage4)
dat$neuro_hyperactivity_stage4_binary <- NA
dat$neuro_hyperactivity_stage4_binary[dat$neuro_hyperactivity_stage4>=7]<-1
dat$neuro_hyperactivity_stage4_binary[dat$neuro_hyperactivity_stage4<7]<-0
dat$covs_age_child_stage4_kq<-dat$ku991a
dat$covs_age_child_stage4_kq[dat$ku991a<0]<-NA
```

Emotional symptoms (SDQ) - maternal report, continuous and dichotomised at >=5
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_emotional_stage2<-dat$j557a
dat$neuro_emotional_stage2[dat$j557a<0]<-NA
dat$neuro_emotional_stage2_zscore <- scale(dat$neuro_emotional_stage2)
dat$neuro_emotional_stage2_binary <- NA
dat$neuro_emotional_stage2_binary[dat$neuro_emotional_stage2>=5]<-1
dat$neuro_emotional_stage2_binary[dat$neuro_emotional_stage2<5]<-0
#Stage 3
dat$neuro_emotional_stage3<-dat$kq348c
dat$neuro_emotional_stage3[dat$kq348c<0]<-NA
dat$neuro_emotional_stage3_zscore <- scale(dat$neuro_emotional_stage3)
dat$neuro_emotional_stage3_binary <- NA
dat$neuro_emotional_stage3_binary[dat$neuro_emotional_stage3>=5]<-1
dat$neuro_emotional_stage3_binary[dat$neuro_emotional_stage3<5]<-0
#Stage 4
dat$neuro_emotional_stage4<-dat$ku707b
dat$neuro_emotional_stage4[dat$ku707b<0]<-NA
dat$neuro_emotional_stage4_zscore <- scale(dat$neuro_emotional_stage4)
dat$neuro_emotional_stage4_binary <- NA
dat$neuro_emotional_stage4_binary[dat$neuro_emotional_stage4>=5]<-1
dat$neuro_emotional_stage4_binary[dat$neuro_emotional_stage4<5]<-0
```

Conduct problems (SDQ) - maternal report, continuous and dichotomised at >=4
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_conduct_stage2<-dat$j557b
dat$neuro_conduct_stage2[dat$j557b<0]<-NA
dat$neuro_conduct_stage2_zscore <- scale(dat$neuro_conduct_stage2)
dat$neuro_conduct_stage2_binary <- NA
dat$neuro_conduct_stage2_binary[dat$neuro_conduct_stage2>=4]<-1
dat$neuro_conduct_stage2_binary[dat$neuro_conduct_stage2<4]<-0
#Stage 3
dat$neuro_conduct_stage3<-dat$kq348d
dat$neuro_conduct_stage3[dat$kq348d<0]<-NA
dat$neuro_conduct_stage3_zscore <- scale(dat$neuro_conduct_stage3)
dat$neuro_conduct_stage3_binary <- NA
dat$neuro_conduct_stage3_binary[dat$neuro_conduct_stage2>=4]<-1
dat$neuro_conduct_stage3_binary[dat$neuro_conduct_stage2<4]<-0
#Stage 4
dat$neuro_conduct_stage4<-dat$ku708b
dat$neuro_conduct_stage4[dat$ku708b<0]<-NA
dat$neuro_conduct_stage4_zscore <- scale(dat$neuro_conduct_stage4)
dat$neuro_conduct_stage4_binary <- NA
dat$neuro_conduct_stage4_binary[dat$neuro_conduct_stage4>=4]<-1
dat$neuro_conduct_stage4_binary[dat$neuro_conduct_stage4<4]<-0
```

Peer problems (SDQ) - maternal report, continuous and dichotomised at >=4
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_peer_stage2<-dat$j557d
dat$neuro_peer_stage2[dat$j557d<0]<-NA
dat$neuro_peer_stage2_zscore <- scale(dat$neuro_peer_stage2)
dat$neuro_peer_stage2_binary <- NA
dat$neuro_peer_stage2_binary[dat$neuro_peer_stage2>=4]<-1
dat$neuro_peer_stage2_binary[dat$neuro_peer_stage2<4]<-0
#Stage 3
dat$neuro_peer_stage3<-dat$kq348e
dat$neuro_peer_stage3[dat$kq348e<0]<-NA
dat$neuro_peer_stage3_zscore <- scale(dat$neuro_peer_stage3)
dat$neuro_peer_stage3_binary <- NA
dat$neuro_peer_stage3_binary[dat$neuro_peer_stage3>=4]<-1
dat$neuro_peer_stage3_binary[dat$neuro_peer_stage3<4]<-0
#Stage 4
dat$neuro_peer_stage4<-dat$ku709b
dat$neuro_peer_stage4[dat$ku709b<0]<-NA
dat$neuro_peer_stage4_zscore <- scale(dat$neuro_peer_stage4)
dat$neuro_peer_stage4_binary <- NA
dat$neuro_peer_stage4_binary[dat$neuro_peer_stage4>=4]<-1
dat$neuro_peer_stage4_binary[dat$neuro_peer_stage4<4]<-0
```

Prosocial behaviour (SDQ)  - maternal report, continuous and dichotomised at <=6
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_prosocial_stage2<-dat$j557e
dat$neuro_prosocial_stage2[dat$j557e<0]<-NA
dat$neuro_prosocial_stage2_zscore <- scale(dat$neuro_prosocial_stage2)
dat$neuro_prosocial_stage2_binary <- NA
dat$neuro_prosocial_stage2_binary[dat$neuro_prosocial_stage2<=6]<-1
dat$neuro_prosocial_stage2_binary[dat$neuro_prosocial_stage2>6]<-0
#Stage 3
dat$neuro_prosocial_stage3<-dat$kq348a
dat$neuro_prosocial_stage3[dat$kq348a<0]<-NA
dat$neuro_prosocial_stage3_zscore <- scale(dat$neuro_prosocial_stage3)
dat$neuro_prosocial_stage3_binary <- NA
dat$neuro_prosocial_stage3_binary[dat$neuro_prosocial_stage3<=6]<-1
dat$neuro_prosocial_stage3_binary[dat$neuro_prosocial_stage3>6]<-0
#Stage 4
dat$neuro_prosocial_stage4<-dat$ku705b
dat$neuro_prosocial_stage4[dat$ku705b<0]<-NA
dat$neuro_prosocial_stage4_zscore <- scale(dat$neuro_prosocial_stage4)
dat$neuro_prosocial_stage4_binary <- NA
dat$neuro_prosocial_stage4_binary[dat$neuro_prosocial_stage4<=6]<-1
dat$neuro_prosocial_stage4_binary[dat$neuro_prosocial_stage4>6]<-0
```

Total behavioural difficulties (SDQ) - maternal report, continuous and dichotomised at >=17 [this is a combination of 4 subscales - excluding prosocial behaviour]
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_totalsdq_stage2<-dat$j557f
dat$neuro_totalsdq_stage2[dat$j557f<0]<-NA
dat$neuro_totalsdq_stage2_zscore <- scale(dat$neuro_totalsdq_stage2)
dat$neuro_totalsdq_stage2_binary <- NA
dat$neuro_totalsdq_stage2_binary[dat$neuro_totalsdq_stage2>=17]<-1
dat$neuro_totalsdq_stage2_binary[dat$neuro_totalsdq_stage2<17]<-0
#Stage 3
dat$neuro_totalsdq_stage3<-dat$kq348f
dat$neuro_totalsdq_stage3[dat$kq348f<0]<-NA
dat$neuro_totalsdq_stage3_zscore <- scale(dat$neuro_totalsdq_stage3)
dat$neuro_totalsdq_stage3_binary <- NA
dat$neuro_totalsdq_stage3_binary[dat$neuro_totalsdq_stage3>=17]<-1
dat$neuro_totalsdq_stage3_binary[dat$neuro_totalsdq_stage3<17]<-0
#Stage 4
dat$neuro_totalsdq_stage4<-dat$ku710b
dat$neuro_totalsdq_stage4[dat$ku710b<0]<-NA
dat$neuro_totalsdq_stage4_zscore <- scale(dat$neuro_totalsdq_stage4)
dat$neuro_totalsdq_stage4_binary <- NA
dat$neuro_totalsdq_stage4_binary[dat$neuro_totalsdq_stage4>=17]<-1
dat$neuro_totalsdq_stage4_binary[dat$neuro_totalsdq_stage4<17]<-0
```

Internalising problems (SDQ) - maternal report, combination of emotional symptoms and peer problems
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_internalising_stage2 <- rowSums(dat[,c("neuro_emotional_stage2","neuro_peer_stage2")])
dat$neuro_internalising_stage2_zscore <- scale(dat$neuro_internalising_stage2)
#Stage 3
dat$neuro_internalising_stage3 <- rowSums(dat[,c("neuro_emotional_stage3","neuro_peer_stage3")])
dat$neuro_internalising_stage3_zscore <- scale(dat$neuro_internalising_stage3)
#Stage 4
dat$neuro_internalising_stage4 <- rowSums(dat[,c("neuro_emotional_stage4","neuro_peer_stage4")])
dat$neuro_internalising_stage4_zscore <- scale(dat$neuro_internalising_stage4)
```

Externalising problems (SDQ) - maternal report, combination of hyperactivity and conduct problems
```{r eval=FALSE, include=TRUE}
#Stage 2
dat$neuro_externalising_stage2 <- rowSums(dat[,c("neuro_hyperactivity_stage2","neuro_conduct_stage2")])
dat$neuro_externalising_stage2_zscore <- scale(dat$neuro_externalising_stage2)
#Stage 3
dat$neuro_externalising_stage3 <- rowSums(dat[,c("neuro_hyperactivity_stage3","neuro_conduct_stage3")])
dat$neuro_externalising_stage3_zscore <- scale(dat$neuro_externalising_stage3)
#Stage 4
dat$neuro_externalising_stage4 <- rowSums(dat[,c("neuro_hyperactivity_stage4","neuro_conduct_stage4")])
dat$neuro_externalising_stage4_zscore <- scale(dat$neuro_externalising_stage4)
```

**SCDC (Skuse Social Communication Score) prorated**

```{r eval=FALSE, include=TRUE}
#Stage 3
dat$neuro_scdc_stage3<-dat$kr554b
dat$neuro_scdc_stage3[dat$kr554b<0]<-NA
dat$neuro_scdc_stage3_zscore <- scale(dat$neuro_scdc_stage3)
dat$covs_age_child_stage3_kr<-dat$kr991a
dat$covs_age_child_stage3_kr[dat$kr991a<0]<-NA
```

**IQ and/or other or composite measures of intelligence/cognitive performance**

Verbal, performance and total

Stage 2
```{r eval=FALSE, include=TRUE}
dat$neuro_cognition_performance_stage2<-dat$cf811
dat$neuro_cognition_performance_stage2[dat$cf811<0]<-NA
dat$neuro_cognition_performance_stage2_zscore <- scale(dat$neuro_cognition_performance_stage2)

dat$neuro_cognition_verbal_stage2<-dat$cf812
dat$neuro_cognition_verbal_stage2[dat$cf812<0]<-NA
dat$neuro_cognition_verbal_stage2_zscore <- scale(dat$neuro_cognition_verbal_stage2)

dat$neuro_cognition_total_stage2<-dat$cf813
dat$neuro_cognition_total_stage2[dat$cf813<0]<-NA
dat$neuro_cognition_total_stage2_zscore <- scale(dat$neuro_cognition_total_stage2)

dat$covs_age_child_stage2_cif49<-dat$cf018
dat$covs_age_child_stage2_cif49[dat$cf018<0]<-NA
dat$covs_age_child_stage2_cif49<-dat$covs_age_child_stage2_cif49/12 #weeks to months
```

Stage 4
```{r eval=FALSE, include=TRUE}
dat$neuro_cognition_performance_stage4<-dat$f8ws111
dat$neuro_cognition_performance_stage4[dat$f8ws111<0]<-NA
dat$neuro_cognition_performance_stage4_zscore <- scale(dat$neuro_cognition_performance_stage4)

dat$neuro_cognition_verbal_stage4<-dat$f8ws110
dat$neuro_cognition_verbal_stage4[dat$f8ws110<0]<-NA
dat$neuro_cognition_verbal_stage4_zscore <- scale(dat$neuro_cognition_verbal_stage4)

dat$neuro_cognition_total_stage4<-dat$f8ws112
dat$neuro_cognition_total_stage4[dat$f8ws112<0]<-NA
dat$neuro_cognition_total_stage4_zscore <- scale(dat$neuro_cognition_total_stage4)

dat$covs_age_child_stage4_f8<-dat$f8003c
dat$covs_age_child_stage4_f8[dat$f8003c<0]<-NA
```

**Educational attainment**

Need to add

## Remove original variables
Remove the original (uncleaned, underived) variables from the dataset
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%child_neuro_vars)]
```

# Child's immunological outcomes {.tabset}

## Ideal variables
Ideal harmonised immunological health measures for offspring are:

*	Doctor diagnosed asthma
* Wheeze
*	Doctor diagnosed eczema
* Eczema-like rash
*	Allergies (pet, pollen, mite/dust, bee/wasp, food)

## Set up meta-data
```{r eval=FALSE, include=TRUE}
#get a list of immunological measures asked for offspring
immunological_meta_data <- findVars("asthma", "eczema", "allergies", "allergic", "pollen","df","dp","dust","cat","dog","grass","sesame","almond","nut","peanut","milk","latex", "wheeze", "whistle", "whilstling", "wheezing", "wheezed","reaction","bee","mite","atopy","rash","egg","allergy","wasp")
immunological_meta_data <-immunological_meta_data[immunological_meta_data$cat3 %in% c("Child","Child Based"),]

#select measures up to age 11
immunological_meta_data <-immunological_meta_data[grep(lapply(immunological_meta_data$obj,substring,1,3),pattern="f07|f08|f09|f10|f11|cif|ka|kb|kc|kd|ke|kf|kg|kh|ki|kj|kk|kl|km|kn|ko|kp|kq|kr|ks|kt|ku|kv|kw|ccs|ccb|ccc|ccd|cce|ccf|ccg|cch|cci|ccj|cck"),]

```

**Asthma**

Defined as in Arathimos et al. 2017 https://clinicalepigeneticsjournal.biomedcentral.com/articles/10.1186/s13148-017-0414-7:

"Current asthma case status was defined using a positive answer to the question Has your child ever been diagnosed with asthma by a doctor? and a positive answer to one of the following questions: (1) Has your child had asthma in the past 12 months? (2) Has your child been on asthma medication the last 12 months? or (3) Has your study child had wheezing or whistling on the chest in the past 12 months?. Controls were children with negative answers to all the above questions. Current wheeze was defined using answers to the question Has your child had wheezing or whistling on the chest in the last 12 months?. Wheeze is a respiratory condition that may capture broader respiratory pathology that includes but is not limited to asthma. However, not all asthmatics currently wheeze since many are currently taking medication that supress disease symptoms."

* kb053	Child wheezed
* kd051b	Wheezing since 6mths (adj)
* kd070	Wheezing & Whistling on Chest since 6 MTHS
* kf063	Child wheezed > 18 months, Y/N
* kj043	CH Had Wheezing Y/N
* kj090	CH Had Wheezing Periods Past 12 Months
* kl031	A4l: Child had wheezing since age 3
* kl080	A7a: Child had periods when there was wheezing/whistling as they breathed since age 3
* kn1031	A3l: Child had wheezing in past 15 months
* kn1110	A6a: Child had periods of wheezing with whistling on
* kp3030	C1p: Child had asthma medication in past 12 months
* kq034a	DV: CH Had Asthma In Past Year
* kq024a	DV: CH Had Wheezing In Past Year
* kq070	A6a: CH Had Periods Of Wheezing And Whistling On
* kr031a	DV: Child had wheezing in past 12 months (Y/N)
* kr041a	DV: Child had asthma in past 12 months (Y/N)
* kr050	A4: Doctor has ever said child has asthma
* kr105a	DV: Child had asthma medicine in past 12 months
* ks1031	A3l: Child had wheezing in past year
* ks1041	A3v: Child had asthma in past year
* ks1240	A8p: Child given asthma medication in past year
* ks1260	A10a: Child suffered wheezing with whistling on chest in past year
* kv1050 A4k: Child had wheezing in past 12 months
* kv1059 A4t: Child had asthma in past 12 months
* kv1070	A5: Doctor stated that child has asthma or eczema
* kv1080	A6a: Child had wheezing and whistling in chest in pas

**Eczema**

Defined as in Paternoster et al. 2018 https://www.jacionline.org/article/S0091-6749(17)31739-6/fulltext#sec1.2 as:
"Information regarding the presence/absence of rash consistent with AD was extracted from questionnaires completed by the mothers when the children were aged between 6 months and 16.5 years (at 6, 18, 30, 42, 57, 69, 81, 103, 128, 140, 166, and 198 months). At each time point AD was defined as a positive response to 1 of the following questions: Has your child had an itchy, dry, oozing or crusted rash on the face, forearms or shins? (at age 6 months); Has your child had a skin rash in the joints and creases of their body (e.g. behind the knees, elbows, under the arms) in the past 6-12 months? (18-166 months)"

* ka249	Baby had rash in joints or creases
* ka252	Baby had face forearm or shin rash
* kb086	CH had rash in joints & creases
* kb089	CH had oozing or crusted rash
* kd085	CH Had Rash in Joints Since Aged 6 MTHS
* kd088	Oozing or Crusted Rash Since Aged 6 MTHS
* kf110	Child had rash in joints > 18 months
* kj100	CH Had Dry Itchy Rash In Joints
* kl100	A8a: Child had itchy, dry skin rash in joints since age 3
* kn1120	A7a: Child had dry skin rash on joints and body creas
* kq035a	DV: CH Had Eczema In Past Year
* kq090	A7a: CH Had Itchy/Dry Skin Rash In Past Year
* kr042a	DV: Child had eczema in past 12 months (Y/N)
* ks1042	A3w: Child had eczema in past year
* ks1280	A11a: Child had itchy, dry skin rash in the joints/crea
* kv1060	A4u: Child had eczema in past 12 months
* kv1070	A5: Doctor stated that child has asthma or eczema
* kv1110	A8a: Child had itchy/dry skin rash in joints
* kv1111	A8b: Child had itchy/dry rash in joints in past year

**Allergies**

We have opted to use maternal-report of allergies (rather than skin prick test) because we consider that these data will be available for more cohorts and more participants within cohorts.

* kb419	CH allergic to milk
* kk262a	DV: Known Food or Drinks CH Allergic to
* kk286	B13b1: Child Allergic to Pollen
* kk287	B13b2: Child Allergic to Cat
* kk288	B13b3: Child Allergic to Dog
* kk289	B13b4: Child Allergic to Bee Sting
* kk290	B13b5: Child Allergic to House Dust
* km2200	B12: Child allergic to foods or drinks
* km2231	B13b1: Child allergic to pollen
* km2232	B13b2: Child allergic to cat
* km2233	B13b3: Child allergic to dog
* km2234	B13b4: Child allergic to bee sting
* km2235	B13b5: Child allergic to house dust
* kq195a	A19: Is CH Allergic To Any Food or Drink
* kq217	A20b1: CH Allergic To Pollen
* kq218	A20b2: CH Allergic To Cats
* kq219	A20b3: CH Allergic To Dogs
* kq220	A20b4: CH Allergic To Bee or Wasp Sting
* kq221	A20b5: CH Allergic To House Dust
* ks3000	C1: Child had an allergic reaction to food/drink since 7th birthday
* ks3031	C2b1: Child allergic to pollen
* ks3032	C2b2: Child allergic to cats
* ks3033	C2b3: Child allergic to dogs
* ks3034	C2b4: Child allergic to bee/wasp stings
* ks3035	C2b5: Child allergic to house dust


### Make list of variables
```{r eval=FALSE, include=TRUE}
immuno_vars <- unique(c("ka249","ka252","kb086","kb089","kd085","kd088","kf110","kj100","kl100","kn1120","kq035a","kq090","kr042a","ks1042","ks1280","kv1070","kv1110","kv1111","kb053","kd051b","kd070","kf063","kj043","kj090","kl031","kl080","kn1031","kn1110","kp3030","kq034a","kq024a","kq070","kr031a","kr041a","kr050","kr105a","ks1031","ks1041","ks1240","ks1260","kv1060","kv1070","kv1080","kb419","kk262a","kk286","kk287","kk288","kk289","kk290","km2200","km2231","km2232","km2233","km2234","km2235","kq195a","kq217","kq218","kq219","kq220","kq221","ks3000","ks3031","ks3032","ks3033","ks3034","ks3035","kv1050","kv1059","ks3030"))

# add ages:
immuno_vars <-c(immuno_vars, "ka497", "kb879a","kd990","kf999", "kj999a","kk998a","kl991a","km9991a", "kn9991a","kp9991a","ks9991a","kv9991a")
```

### Set up final meta data object
```{r eval=FALSE, include=TRUE}
immuno_meta_data <- findVars(unique(immuno_vars)) #create a table with all the necessary meta data for each variable

immuno_meta_data <- subset(immuno_meta_data,subset=tolower(name) %in% immuno_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
```{r eval=FALSE, include=TRUE}
immuno_data <- extractVars(immuno_meta_data[immuno_meta_data$name %in% unique(immuno_vars),]) #extracting individual level data and performing withdrawal of consent

immuno_data<-immuno_data[,c("aln","qlet",immuno_vars)]
dat <- left_join(dat,immuno_data,by=c("aln","qlet")) 
```

## Derive clean variables

Where there are multiple variables within a stage, the final variable is coded so that any answer of no = no, but then this is replaced with yes if they gave any answer of yes. If all NA, the value == NA.

**Wheeze**

Stage 0 (>0,<1) - kb child wheezed (selected just for the participants answering before 12 months)
```{r eval=FALSE, include=TRUE}
dat$immuno_wheeze_stage0_binary <-NA
dat$immuno_wheeze_stage0_binary[which(dat$kb053==1&dat$kb879a>0&dat$kb879a<12)]<-1
dat$immuno_wheeze_stage0_binary[which(dat$kb053==2&dat$kb879a>0&dat$kb879a<12)]<-0
```

Stage 1 (around age 2; >=1 to <3) - kd, kf: kd051b Wheezing since 6mths and 
kd070 Wheezing & Whistling on Chest since 6 MTHS and kf063 Child wheezed > 18 months, Y/N. 
```{r eval=FALSE, include=TRUE}
dat$immuno_wheeze_stage1_binary <-NA
dat$immuno_wheeze_stage1_binary[which(dat$kd051b==2 | dat$kd070==2 | dat$kf063==2)] <-0
dat$immuno_wheeze_stage1_binary[which(dat$kd051b==1 | dat$kd070==1 | dat$kf063==1)] <-1
```

Stage 2 (around age 4; >=3 to <5) - kj and kl have questions.
```{r eval=FALSE, include=TRUE}
dat$immuno_wheeze_stage2_binary <-NA
dat$immuno_wheeze_stage2_binary[which(dat$kj043==2 | dat$kj090==2 | dat$kl031==3| dat$kl080==2)] <-0
dat$immuno_wheeze_stage2_binary[which(dat$kj043==1 | dat$kj090==1 | dat$kl031%in%c(1,2)| dat$kl080==1)] <-1
```

Stage 3 (around age 6; >=5 to <8) - kn, kq, kr, kp have questions.
```{r eval=FALSE, include=TRUE}
dat$immuno_wheeze_stage3_binary <-NA
dat$immuno_wheeze_stage3_binary[which(dat$kn1031==3 | dat$kn1110==2 | dat$kq024a==2| dat$kq070==2| dat$kr031a==2)] <-0
dat$immuno_wheeze_stage3_binary[which(dat$kn1031%in%c(1,2) | dat$kn1110==1 | dat$kq024a==1| dat$kq070==1| dat$kr031a==1)] <-1
```

Stage 4 (around age 9; >=8 to <11) - ks, kv have questions
```{r eval=FALSE, include=TRUE}
dat$immuno_wheeze_stage4_binary <-NA
dat$immuno_wheeze_stage4_binary[which(dat$ks1031==3 | dat$ks1260==2 | dat$kv1080==2|dat$kv1050==3)] <-0
dat$immuno_wheeze_stage4_binary[which(dat$ks1031%in%c(1,2) | dat$ks1260==1 | dat$kv1080==1|dat$kv1050%in%c(1,2))] <-1
```

Any timepoint
```{r eval=FALSE, include=TRUE}
dat$immuno_wheeze_allstages_binary <- rowSums(dat[,c("immuno_wheeze_stage0_binary","immuno_wheeze_stage1_binary","immuno_wheeze_stage2_binary","immuno_wheeze_stage3_binary","immuno_wheeze_stage4_binary")],na.rm=T)
dat$immuno_wheeze_allstages_binary[dat$immuno_wheeze_allstages_binary>0]<-1
dat$immuno_wheeze_allstages_binary[which(is.na(dat$immuno_wheeze_stage0_binary) & is.na(dat$immuno_wheeze_stage1_binary) & is.na(dat$immuno_wheeze_stage2_binary)&is.na(dat$immuno_wheeze_stage3_binary)&is.na(dat$immuno_wheeze_stage4_binary))] <- NA
```

**Asthma**

Stage 3 (around age 6; >=5 to <8) - kn, kq, kr, kp have questions. kp3030 C1p: Child had asthma medication in past 12 months, kq034a DV: CH Had Asthma In Past Year, kr041a DV: Child had asthma in past 12 months (Y/N), kr050 A4: Doctor has ever said child has asthma, kr105a DV: Child had asthma medicine in past 12 months
```{r eval=FALSE, include=TRUE}
dat$immuno_asthma_stage3_binary <-NA
dat$immuno_asthma_stage3_binary[which(dat$kp3030==1 | dat$kq034a==2 | dat$kr041a==2 | dat$kr105a==2)] <-0
dat$immuno_asthma_stage3_binary[which(dat$kp3030%in%c(2,3) | dat$kq034a==1 | dat$kr041a==1 | dat$kr105a==1)]  <-1

# replace if doctor diagnosed
dat$immuno_asthma_stage3_binary[which(dat$kr050==1)] <-1
```
Stage 4 (around age 9; >=8 to <11) - ks1041 A3v: Child had asthma in past year, ks1240 A8p: Child given asthma medication in past year, kv1059 A4t: Child had asthma in past 12 months
```{r eval=FALSE, include=TRUE}
dat$immuno_asthma_stage4_binary <-NA
dat$immuno_asthma_stage4_binary[which(dat$ks1041==3 | dat$ks1240==1 | dat$kv1059==3)] <-0
dat$immuno_asthma_stage4_binary[which(dat$ks1041%in%c(1,2) | dat$ks1240%in%c(2,3)|dat$kv1059%in%c(1,2))] <-1

# replace if doctor diagnosed
dat$immuno_asthma_stage4_binary[which(dat$kv1070%in%c(1,3))] <-1
```

Any timepoint
```{r eval=FALSE, include=TRUE}
dat$immuno_asthma_allstages_binary <- rowSums(dat[,c("immuno_asthma_stage3_binary","immuno_asthma_stage4_binary")],na.rm=T)
dat$immuno_asthma_allstages_binary[dat$immuno_asthma_allstages_binary>0]<-1
dat$immuno_asthma_allstages_binary[which(is.na(dat$immuno_asthma_stage3_binary)&is.na(dat$immuno_asthma_stage4_binary))] <- NA
```

**Eczema-like rash or Eczema**

Stage 0 (>0,<1) ka,kb
```{r eval=FALSE, include=TRUE}
dat$immuno_eczema_stage0_binary <-NA
dat$immuno_eczema_stage0_binary[which(dat$ka249 ==2| dat$ka252==2 | dat$kb086==2 & (dat$kb879a>0&dat$kb879a<12))] <-0
dat$immuno_eczema_stage0_binary[which(dat$ka249 ==1| dat$ka252==1 | dat$kb086==1 & (dat$kb879a>0&dat$kb879a<12))] <-1
```
Stage 1 (around age 2; >=1 to <3) - kd, kf (kd questions refer to since 6 months, not clear if this is stage 0 or 1, so omitted)
```{r eval=FALSE, include=TRUE}
dat$immuno_eczema_stage1_binary <-NA
dat$immuno_eczema_stage1_binary[which(dat$kf110 ==2)]<-0
dat$immuno_eczema_stage1_binary[which(dat$kf110 ==1)]<-1
```
Stage 2 (around age 4; >=3 to <5) - kj and kl have questions.
```{r eval=FALSE, include=TRUE}
dat$immuno_eczema_stage2_binary <-NA
dat$immuno_eczema_stage2_binary[which(dat$kj100 ==2| dat$kl100==2)] <-0
dat$immuno_eczema_stage2_binary[which(dat$kj100 ==1| dat$kl100==1)] <-1
```
Stage 3 (around age 6; >=5 to <8) - kn, kq, kr, kp 
```{r eval=FALSE, include=TRUE}
dat$immuno_eczema_stage3_binary <-NA
dat$immuno_eczema_stage3_binary[which(dat$kn1120==2| dat$kq035a==2 | dat$kq090==2 | dat$kr042a==2)] <-0
dat$immuno_eczema_stage3_binary[which(dat$kn1120==1| dat$kq035a==1 | dat$kq090==1 | dat$kr042a==1)] <-1
```
Stage 4 (around age 9; >=8 to <11) - ks, kv 
```{r eval=FALSE, include=TRUE}
dat$immuno_eczema_stage4_binary <-NA
dat$immuno_eczema_stage4_binary[which(dat$ks1042==3 | dat$ks1280==2 | dat$kv1110==2 | dat$kv1111==2)] <-0
dat$immuno_eczema_stage4_binary[which(dat$ks1042%in%c(1,2) | dat$ks1280==1 | dat$kv1110==1 | dat$kv1111==1)] <-1

dat$immuno_eczema_stage4_binary[which(dat$kv1070%in%c(2,3))] <-1
```

Any timepoint
```{r eval=FALSE, include=TRUE}
dat$immuno_eczema_allstages_binary <- rowSums(dat[,c("immuno_eczema_stage0_binary","immuno_eczema_stage1_binary","immuno_eczema_stage2_binary","immuno_eczema_stage3_binary","immuno_eczema_stage4_binary")],na.rm=T)
dat$immuno_eczema_allstages_binary[dat$immuno_eczema_allstages_binary>0]<-1
dat$immuno_eczema_allstages_binary[which(is.na(dat$immuno_eczema_stage0_binary) & is.na(dat$immuno_eczema_stage1_binary) & is.na(dat$immuno_eczema_stage2_binary)&is.na(dat$immuno_eczema_stage3_binary)&is.na(dat$immuno_eczema_stage4_binary))] <- NA
```


**Allergies**

*Food*

Stage 2 (around age 4; >=3 to <5) - kk
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_food_stage2_binary <- NA
dat$immuno_allergy_food_stage2_binary[which(dat$kk262a==1)]<-1
dat$immuno_allergy_food_stage2_binary[which(dat$kk262a==2)]<-0
```
Stage 3 (around age 6; >=5 to <8) - kq km have questions, but joining both together (as in commented out code below) gives too many "cases" (compared with variables at other stages and each single variable). Therefore, have decided to choose just one variable - kq is more towards the middle of this stage
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_food_stage3_binary <- NA
dat$immuno_allergy_food_stage3_binary[which(dat$kq195a==2)]<-0
dat$immuno_allergy_food_stage3_binary[which(dat$kq195a==1)]<-1
```
Stage 4 (around age 9; >=8 to <11) - ks
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_food_stage4_binary <- NA
dat$immuno_allergy_food_stage4_binary[which(dat$ks3000%in%c(1,2))]<-1
dat$immuno_allergy_food_stage4_binary[which(dat$ks3000==3)]<-0
```

*Pollen*

Stage 2 (around age 4; >=3 to <5) - kk (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_pollen_stage2_binary <- NA
dat$immuno_allergy_pollen_stage2_binary[dat$kk286==2]<-0
dat$immuno_allergy_pollen_stage2_binary[dat$kk286==1]<-1
```
Stage 3 (around age 6; >=5 to <8) - kq (6.5) km (5.5)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_pollen_stage3_binary <- NA
dat$immuno_allergy_pollen_stage3_binary[dat$kq217==2]<-0
dat$immuno_allergy_pollen_stage3_binary[dat$kq217==1]<-1
```
Stage 4 (around age 9; >=8 to <11) - ks
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_pollen_stage4_binary <-NA
dat$immuno_allergy_pollen_stage4_binary[which(dat$ks3030%in%c(1,2))] <-0 #set to no if give valid answer to "apart from food and drink, is your child allergic to anything else?
dat$immuno_allergy_pollen_stage4_binary[which(dat$ks3031==1)] <-1 # replace with yes if they ticked this box
```

*Cats*

Stage 2 (around age 4; >=3 to <5) - kk (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_cat_stage2_binary <- NA
dat$immuno_allergy_cat_stage2_binary[dat$kk287==2]<-0
dat$immuno_allergy_cat_stage2_binary[dat$kk287==1]<-1
```
Stage 3 (around age 6; >=5 to <8) - kq (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_cat_stage3_binary <- NA
dat$immuno_allergy_cat_stage3_binary[dat$kq218==2]<-0
dat$immuno_allergy_cat_stage3_binary[dat$kq218==1]<-1
```
Stage 4 (around age 9; >=8 to <11) - ks
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_cat_stage4_binary <-NA
dat$immuno_allergy_cat_stage4_binary[which(dat$ks3030%in%c(1,2))] <-0 #set to no if give valid answer to "apart from food and drink, is your child allergic to anything else?
dat$immuno_allergy_cat_stage4_binary[which(dat$ks3032==1)] <-1 # replace with yes if they ticked this box
```

*Dogs*

Stage 2 (around age 4; >=3 to <5) - kk (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_dog_stage2_binary <- NA
dat$immuno_allergy_dog_stage2_binary[dat$kk288==2]<-0
dat$immuno_allergy_dog_stage2_binary[dat$kk288==1]<-1
```
Stage 3 (around age 6; >=5 to <8) - kq (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_dog_stage3_binary <- NA
dat$immuno_allergy_dog_stage3_binary[dat$kq219==2]<-0
dat$immuno_allergy_dog_stage3_binary[dat$kq219==1]<-1
```
Stage 4 (around age 9; >=8 to <11) - ks
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_dog_stage4_binary <-NA
dat$immuno_allergy_dog_stage4_binary[which(dat$ks3030%in%c(1,2))] <-0 #set to no if give valid answer to "apart from food and drink, is your child allergic to anything else?
dat$immuno_allergy_dog_stage4_binary[which(dat$ks3033==1)] <-1 # replace with yes if they ticked this box
```

*Insect sting*

Stage 2 (around age 4; >=3 to <5) - kk (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_insect_stage2_binary <- NA
dat$immuno_allergy_insect_stage2_binary[dat$kk289==2]<-0
dat$immuno_allergy_insect_stage2_binary[dat$kk289==1]<-1
```
Stage 3 (around age 6; >=5 to <8) - kq (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_insect_stage3_binary <- NA
dat$immuno_allergy_insect_stage3_binary[dat$kq220==2]<-0
dat$immuno_allergy_insect_stage3_binary[dat$kq220==1]<-1
```
Stage 4 (around age 9; >=8 to <11) - ks
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_insect_stage4_binary <-NA
dat$immuno_allergy_insect_stage4_binary[which(dat$ks3030%in%c(1,2))] <-0 #set to no if give valid answer to "apart from food and drink, is your child allergic to anything else?
dat$immuno_allergy_insect_stage4_binary[which(dat$ks3034==1)] <-1 # replace with yes if they ticked this box
```

*House dust*

Stage 2 (around age 4; >=3 to <5) - kk (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_dust_stage2_binary <- NA
dat$immuno_allergy_dust_stage2_binary[dat$kk290==2]<-0
dat$immuno_allergy_dust_stage2_binary[dat$kk290==1]<-1
```
Stage 3 (around age 6; >=5 to <8) - kq (km?)
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_dust_stage3_binary <- NA
dat$immuno_allergy_dust_stage3_binary[dat$kq221==2]<-0
dat$immuno_allergy_dust_stage3_binary[dat$kq221==1]<-1
```
Stage 4 (around age 9; >=8 to <11) - ks
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_dust_stage4_binary <-NA
dat$immuno_allergy_dust_stage4_binary[which(dat$ks3030%in%c(1,2))] <-0 #set to no if give valid answer to "apart from food and drink, is your child allergic to anything else?
dat$immuno_allergy_dust_stage4_binary[which(dat$ks3035==1)] <-1 # replace with yes if they ticked this box
```

*Any allergy*

Stage 2
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_any_stage2_binary <- NA
dat$immuno_allergy_any_stage2_binary[which(dat$immuno_allergy_food_stage2_binary==0&dat$immuno_allergy_pollen_stage2_binary==0&dat$immuno_allergy_cat_stage2_binary==0&dat$immuno_allergy_dog_stage2_binary==0&dat$immuno_allergy_insect_stage2_binary==0&dat$immuno_allergy_dust_stage2_binary==0)] <-0
dat$immuno_allergy_any_stage2_binary[which(dat$immuno_allergy_food_stage2_binary==1|dat$immuno_allergy_pollen_stage2_binary==1|dat$immuno_allergy_cat_stage2_binary==1|dat$immuno_allergy_dog_stage2_binary==1|dat$immuno_allergy_insect_stage2_binary==1|dat$immuno_allergy_dust_stage2_binary==1)] <-1
```

Stage 3
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_any_stage3_binary <- NA
dat$immuno_allergy_any_stage3_binary[which(dat$immuno_allergy_food_stage3_binary==0&dat$immuno_allergy_pollen_stage3_binary==0&dat$immuno_allergy_cat_stage3_binary==0&dat$immuno_allergy_dog_stage3_binary==0&dat$immuno_allergy_insect_stage3_binary==0&dat$immuno_allergy_dust_stage3_binary==0)] <-0
dat$immuno_allergy_any_stage3_binary[which(dat$immuno_allergy_food_stage3_binary==1|dat$immuno_allergy_pollen_stage3_binary==1|dat$immuno_allergy_cat_stage3_binary==1|dat$immuno_allergy_dog_stage3_binary==1|dat$immuno_allergy_insect_stage3_binary==1|dat$immuno_allergy_dust_stage3_binary==1)] <-1
```

Stage 4
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_any_stage4_binary <- NA
dat$immuno_allergy_any_stage4_binary[which(dat$immuno_allergy_food_stage2_binary==0&dat$immuno_allergy_pollen_stage4_binary==0&dat$immuno_allergy_cat_stage4_binary==0&dat$immuno_allergy_dog_stage4_binary==0&dat$immuno_allergy_insect_stage4_binary==0&dat$immuno_allergy_dust_stage4_binary==0)] <-0
dat$immuno_allergy_any_stage4_binary[which(dat$immuno_allergy_food_stage4_binary==1|dat$immuno_allergy_pollen_stage4_binary==1|dat$immuno_allergy_cat_stage4_binary==1|dat$immuno_allergy_dog_stage4_binary==1|dat$immuno_allergy_insect_stage4_binary==1|dat$immuno_allergy_dust_stage4_binary==1)] <-1
```

Any time
```{r eval=FALSE, include=TRUE}
dat$immuno_allergy_any_allstages_binary <- rowSums(dat[,c("immuno_allergy_any_stage2_binary","immuno_allergy_any_stage3_binary","immuno_allergy_any_stage4_binary")],na.rm=T)
dat$immuno_allergy_any_allstages_binary[dat$immuno_allergy_any_allstages_binary>0]<-1
dat$immuno_allergy_any_allstages_binary[which(is.na(dat$immuno_allergy_any_stage2_binary) & is.na(dat$immuno_allergy_any_stage3_binary) & is.na(dat$immuno_allergy_any_stage4_binary))] <- NA
```

## Remove original variables
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%immuno_vars)]
```

# Child's cardiometabolic health outcomes {.tabset}

## Set up meta-data

* Systolic BP
* Diastolic BP

The following were measured on non-fasting blood samples:

* triglycerides
* total cholesterol
* low-density lipoprotein (LDL) cholesterol
* high-density lipoprotein (HDL) cholesterol
* glucose
* insulin
* C-reactive protein
* apolipoprotein a
* apolipoprotein b
* interleukin 6


```{r eval=FALSE, include=TRUE}
cardiomet_vars <- c("cf124", "cf134", "cf144", "f7sa022", "f9sa022", "fdar118","cf143", "cf133", "cf123", "f7sa021", "f9sa021", "fdar117","trig_cord", "Trig_CIF31", "Trig_CIF43", "TRIG_F7", "trig_f9","HDL_cord", "HDL_CIF31", "HDL_CIF43", "HDL_F7", "HDL_f9","LDL_cord", "LDL_CIF31", "LDL_CIF43", "LDL_F7", "LDL_f9","chol_cord", "Chol_CIF31", "Chol_CIF43", "CHOL_F7", "CHOL_F9","Glc_F7","Insulin_cord", "insulin_F9","CRP_cord", "CRP_f9","ApoAI_f9","ApoB_f9","ApoA1_F7","ApoB_F7","IL6_f9")

cardiomet_meta_data <- findVars(cardiomet_vars)

cardiomet_meta_data <- subset(cardiomet_meta_data,subset=name %in% cardiomet_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
cardiomet_meta_data <- cardiomet_meta_data[cardiomet_meta_data$cat1=="Current",]
```

## Extract variables
```{r eval=FALSE, include=TRUE}
cardiomet_data <- extractVars(cardiomet_meta_data[cardiomet_meta_data$name %in% unique(cardiomet_vars),]) #extracting individual level data and performing withdrawal of consent

cardiomet_data<-cardiomet_data[,c("aln","qlet",cardiomet_vars)]
dat <- left_join(dat,cardiomet_data,by=c("aln","qlet")) 
```

## Derive clean variables
SBP
```{r eval=FALSE, include=TRUE}

remove.iqr3.outliers <- function(x){x[(x<(quantile(x,0.25,na.rm=T)-(IQR(x,na.rm=T)*3)))|(x>(quantile(x,0.75,na.rm=T)+(IQR(x,na.rm=T)*3)))]<-NA;x}

dat$cardio_sbp_stage2<- NA
dat$cardio_sbp_stage2[which(dat$stage2_timepoint=="37")] <-dat$cf123[which(dat$stage2_timepoint=="37")]
dat$cardio_sbp_stage2[which(dat$stage2_timepoint=="49")] <-dat$cf133[which(dat$stage2_timepoint=="49")]
dat$cardio_sbp_stage2[dat$cardio_sbp_stage2<0]<-NA
dat$cardio_sbp_stage2<- remove.iqr3.outliers(dat$cardio_sbp_stage2)
dat$cardio_sbp_stage2_zscore <- scale(dat$cardio_sbp_stage2)

dat$cardio_sbp_stage3 <- dat$f7sa021
dat$cardio_sbp_stage3[dat$f7sa021<0]<-NA
dat$cardio_sbp_stage3<- remove.iqr3.outliers(dat$cardio_sbp_stage3)
dat$cardio_sbp_stage3_zscore <- scale(dat$cardio_sbp_stage3)

dat$cardio_sbp_stage4 <- dat$f9sa021
dat$cardio_sbp_stage4[dat$f9sa021<0]<-NA
dat$cardio_sbp_stage4<- remove.iqr3.outliers(dat$cardio_sbp_stage4)
dat$cardio_sbp_stage4_zscore <- scale(dat$cardio_sbp_stage4)
```
DBP
```{r eval=FALSE, include=TRUE}
dat$cardio_dbp_stage2<- NA
dat$cardio_dbp_stage2[which(dat$stage2_timepoint=="37")] <-dat$cf124[which(dat$stage2_timepoint=="37")]
dat$cardio_dbp_stage2[which(dat$stage2_timepoint=="49")] <-dat$cf134[which(dat$stage2_timepoint=="49")]
dat$cardio_dbp_stage2[dat$cardio_dbp_stage2<0]<-NA
dat$cardio_dbp_stage2<- remove.iqr3.outliers(dat$cardio_dbp_stage2)
dat$cardio_dbp_stage2_zscore <- scale(dat$cardio_dbp_stage2)

dat$cardio_dbp_stage3 <- dat$f7sa022
dat$cardio_dbp_stage3[dat$f7sa022<0]<-NA
dat$cardio_dbp_stage3<- remove.iqr3.outliers(dat$cardio_dbp_stage3)
dat$cardio_dbp_stage3_zscore <- scale(dat$cardio_dbp_stage3)

dat$cardio_dbp_stage4 <- dat$f9sa022
dat$cardio_dbp_stage4[dat$f9sa022<0]<-NA
dat$cardio_dbp_stage4<- remove.iqr3.outliers(dat$cardio_dbp_stage4)
dat$cardio_dbp_stage4_zscore <- scale(dat$cardio_dbp_stage4)
```
Triglycerides
```{r eval=FALSE, include=TRUE}
dat$cardio_trig_stage0 <- dat$trig_cord
dat$cardio_trig_stage0[dat$trig_cord<0]<-NA
dat$cardio_trig_stage0<- remove.iqr3.outliers(dat$cardio_trig_stage0)
dat$cardio_trig_stage0_zscore <- scale(dat$cardio_trig_stage0)

dat$cardio_trig_stage1 <- dat$Trig_CIF31
dat$cardio_trig_stage1[dat$Trig_CIF31<0]<-NA
dat$cardio_trig_stage1<- remove.iqr3.outliers(dat$cardio_trig_stage1)
dat$cardio_trig_stage1_zscore <- scale(dat$cardio_trig_stage1)

dat$cardio_trig_stage2 <- dat$Trig_CIF43
dat$cardio_trig_stage2[dat$Trig_CIF43<0]<-NA
dat$cardio_trig_stage2<- remove.iqr3.outliers(dat$cardio_trig_stage2)
dat$cardio_trig_stage2_zscore <- scale(dat$cardio_trig_stage2)

dat$cardio_trig_stage3 <- dat$TRIG_F7
dat$cardio_trig_stage3[dat$TRIG_F7<0]<-NA
dat$cardio_trig_stage3<- remove.iqr3.outliers(dat$cardio_trig_stage3)
dat$cardio_trig_stage3_zscore <- scale(dat$cardio_trig_stage3)

dat$cardio_trig_stage4 <- dat$trig_f9
dat$cardio_trig_stage4[dat$trig_f9<0]<-NA
dat$cardio_trig_stage4<- remove.iqr3.outliers(dat$cardio_trig_stage4)
dat$cardio_trig_stage4_zscore <- scale(dat$cardio_trig_stage4)

```
HDL
```{r eval=FALSE, include=TRUE}
dat$cardio_HDL_stage0 <- dat$HDL_cord
dat$cardio_HDL_stage0[dat$HDL_cord<0]<-NA
dat$cardio_HDL_stage0<- remove.iqr3.outliers(dat$cardio_HDL_stage0)
dat$cardio_HDL_stage0_zscore <- scale(dat$cardio_HDL_stage0)

dat$cardio_HDL_stage1 <- dat$HDL_CIF31
dat$cardio_HDL_stage1[dat$HDL_CIF31<0]<-NA
dat$cardio_HDL_stage1<- remove.iqr3.outliers(dat$cardio_HDL_stage1)
dat$cardio_HDL_stage1_zscore <- scale(dat$cardio_HDL_stage1)

dat$cardio_HDL_stage2 <- dat$HDL_CIF43
dat$cardio_HDL_stage2[dat$HDL_CIF43<0]<-NA
dat$cardio_HDL_stage2<- remove.iqr3.outliers(dat$cardio_HDL_stage2)
dat$cardio_HDL_stage2_zscore <- scale(dat$cardio_HDL_stage2)

dat$cardio_HDL_stage3 <- dat$HDL_F7
dat$cardio_HDL_stage3[dat$HDL_F7<0]<-NA
dat$cardio_HDL_stage3<- remove.iqr3.outliers(dat$cardio_HDL_stage3)
dat$cardio_HDL_stage3_zscore <- scale(dat$cardio_HDL_stage3)

dat$cardio_HDL_stage4 <- dat$HDL_f9
dat$cardio_HDL_stage4[dat$HDL_f9<0]<-NA
dat$cardio_HDL_stage4<- remove.iqr3.outliers(dat$cardio_HDL_stage4)
dat$cardio_HDL_stage4_zscore <- scale(dat$cardio_HDL_stage4)
```
LDL
```{r eval=FALSE, include=TRUE}
dat$cardio_LDL_stage0 <- dat$LDL_cord
dat$cardio_LDL_stage0[dat$LDL_cord<0]<-NA
dat$cardio_LDL_stage0<- remove.iqr3.outliers(dat$cardio_LDL_stage0)
dat$cardio_LDL_stage0_zscore <- scale(dat$cardio_LDL_stage0)

dat$cardio_LDL_stage1 <- dat$LDL_CIF31
dat$cardio_LDL_stage1[dat$LDL_CIF31<0]<-NA
dat$cardio_LDL_stage4<- remove.iqr3.outliers(dat$cardio_LDL_stage1)
dat$cardio_LDL_stage1_zscore <- scale(dat$cardio_LDL_stage1)

dat$cardio_LDL_stage2 <- dat$LDL_CIF43
dat$cardio_LDL_stage2[dat$LDL_CIF43<0]<-NA
dat$cardio_LDL_stage2<- remove.iqr3.outliers(dat$cardio_LDL_stage2)
dat$cardio_LDL_stage2_zscore <- scale(dat$cardio_LDL_stage2)

dat$cardio_LDL_stage3 <- dat$LDL_F7
dat$cardio_LDL_stage3[dat$LDL_F7<0]<-NA
dat$cardio_LDL_stage3<- remove.iqr3.outliers(dat$cardio_LDL_stage3)
dat$cardio_LDL_stage3_zscore <- scale(dat$cardio_LDL_stage3)

dat$cardio_LDL_stage4 <- dat$LDL_f9
dat$cardio_LDL_stage4[dat$LDL_f9<0]<-NA
dat$cardio_LDL_stage4<- remove.iqr3.outliers(dat$cardio_LDL_stage4)
dat$cardio_LDL_stage4_zscore <- scale(dat$cardio_LDL_stage4)
```
Total cholesterol
```{r eval=FALSE, include=TRUE}
dat$cardio_chol_stage0 <- dat$chol_cord
dat$cardio_chol_stage0[dat$chol_cord<0]<-NA
dat$cardio_chol_stage0<- remove.iqr3.outliers(dat$cardio_chol_stage0)
dat$cardio_chol_stage0_zscore <- scale(dat$cardio_chol_stage0)

dat$cardio_chol_stage1 <- dat$Chol_CIF31
dat$cardio_chol_stage1[dat$Chol_CIF31<0]<-NA
dat$cardio_chol_stage1<- remove.iqr3.outliers(dat$cardio_chol_stage1)
dat$cardio_chol_stage1_zscore <- scale(dat$cardio_chol_stage1)

dat$cardio_chol_stage2 <- dat$Chol_CIF43
dat$cardio_chol_stage2[dat$Chol_CIF43<0]<-NA
dat$cardio_chol_stage2<- remove.iqr3.outliers(dat$cardio_chol_stage2)
dat$cardio_chol_stage2_zscore <- scale(dat$cardio_chol_stage2)

dat$cardio_chol_stage3 <- dat$CHOL_F7
dat$cardio_chol_stage3[dat$CHOL_F7<0]<-NA
dat$cardio_chol_stage3<- remove.iqr3.outliers(dat$cardio_chol_stage3)
dat$cardio_chol_stage3_zscore <- scale(dat$cardio_chol_stage3)

dat$cardio_chol_stage4 <- dat$CHOL_F9
dat$cardio_chol_stage4[dat$CHOL_F9<0]<-NA
dat$cardio_chol_stage4<- remove.iqr3.outliers(dat$cardio_chol_stage4)
dat$cardio_chol_stage4_zscore <- scale(dat$cardio_chol_stage4)
```
Glucose
```{r eval=FALSE, include=TRUE}
dat$cardio_glucose_stage3 <- dat$Glc_F7
dat$cardio_glucose_stage3[dat$Glc_F7<0]<-NA
dat$cardio_glucose_stage3<- remove.iqr3.outliers(dat$cardio_glucose_stage3)
dat$cardio_glucose_stage3_zscore <- scale(dat$cardio_glucose_stage3)
```
Insulin
```{r eval=FALSE, include=TRUE}
dat$cardio_insulin_stage0 <- dat$Insulin_cord
dat$cardio_insulin_stage0[dat$Insulin_cord<0]<-NA
dat$cardio_insulin_stage0<- remove.iqr3.outliers(dat$cardio_insulin_stage0)
dat$cardio_insulin_stage0_zscore <- scale(dat$cardio_insulin_stage0)

dat$cardio_insulin_stage4 <- dat$insulin_F9
dat$cardio_insulin_stage4[dat$insulin_F9<0]<-NA
dat$cardio_insulin_stage4<- remove.iqr3.outliers(dat$cardio_insulin_stage4)
dat$cardio_insulin_stage4_zscore <- scale(dat$cardio_insulin_stage4)
```
CRP
```{r eval=FALSE, include=TRUE}
# removed cord because mostly missing or 0; when outliers removed, all 0

#dat$cardio_CRP_stage0 <- dat$CRP_cord
#dat$cardio_CRP_stage0[dat$CRP_cord<0]<-NA
#dat$cardio_CRP_stage0<- remove.iqr3.outliers(dat$cardio_CRP_stage0)
#dat$cardio_CRP_stage0_zscore <- scale(dat$cardio_CRP_stage0)

dat$cardio_CRP_stage4 <- dat$CRP_f9
dat$cardio_CRP_stage4[dat$CRP_f9<0]<-NA
dat$cardio_CRP_stage4<- remove.iqr3.outliers(dat$cardio_CRP_stage4)
dat$cardio_CRP_stage4_zscore <- scale(dat$cardio_CRP_stage4)
```
Apolipoprotein-A
```{r eval=FALSE, include=TRUE}
dat$cardio_apoA_stage3 <- dat$ApoA1_F7
dat$cardio_apoA_stage3[dat$ApoA1_F7<0]<-NA
dat$cardio_apoA_stage3<- remove.iqr3.outliers(dat$cardio_apoA_stage3)
dat$cardio_apoA_stage3_zscore <- scale(dat$cardio_apoA_stage3)

dat$cardio_apoA_stage4 <- dat$ApoAI_f9
dat$cardio_apoA_stage4[dat$ApoAI_f9<0]<-NA
dat$cardio_apoA_stage4<- remove.iqr3.outliers(dat$cardio_apoA_stage4)
dat$cardio_apoA_stage4_zscore <- scale(dat$cardio_apoA_stage4)
```
Apolipoprotein-B
```{r eval=FALSE, include=TRUE}
dat$cardio_apoB_stage3 <- dat$ApoB_F7
dat$cardio_apoB_stage3[dat$ApoB_F7<0]<-NA
dat$cardio_apoB_stage3<- remove.iqr3.outliers(dat$cardio_apoB_stage3)
dat$cardio_apoB_stage3_zscore <- scale(dat$cardio_apoB_stage3)

dat$cardio_apoB_stage4 <- dat$ApoB_f9
dat$cardio_apoB_stage4[dat$ApoB_f9<0]<-NA
dat$cardio_apoB_stage4<- remove.iqr3.outliers(dat$cardio_apoB_stage4)
dat$cardio_apoB_stage4_zscore <- scale(dat$cardio_apoB_stage4)
```
IL-6
```{r eval=FALSE, include=TRUE}
dat$cardio_il6_stage4 <- dat$IL6_f9
dat$cardio_il6_stage4[dat$IL6_f9<0]<-NA
dat$cardio_il6_stage4<- remove.iqr3.outliers(dat$cardio_il6_stage4)
dat$cardio_il6_stage4_zscore <- scale(dat$cardio_il6_stage4)
```


# Covariates {.tabset}

For most covariates, we are better aware of the most appropriate variables to use, so we can just make a list and extract in one go.

## Ideal variables

### Demographics

*	Ethnicity / country of birth (TC) - c800 (mother), c801 (partner), c804 (child)
*	Maternal and paternal age at conception and/or delivery (TC) - maternal age already derived from the mz file, paternal age at completion of first questionnaire = pa910, at birth = pc996
*	Maternal and paternal education (TC) - c666a for father; c645a for mother
*	Maternal and paternal occupation (TC) From answers to questions about employment, the occupation was coded using the OPCS job codes, and the social class categorisation was derived on a scale of 1-6. c755 for mother, c765 for partner.
*	Parity (number of previous children) (TC) b032
*	Social support (e.g. government child benefits or income support in the UK) (CH/TC) - can't find
*	Sex of child (CH) - already derived from kz sample file

### Family relationships

*	Relationship of partner to child (e.g. are they the biological father) (CH) - already derived
*	Marital status of mother and partner (IN) a525, pa065 (present marital status - not necessarily to each other)
*	Co-habitation of mother and partner during preg(IN) - already derived

### Parents' health

* Maternal and paternal current illnesses and health problems (diabetes, asthma, eczema, etc) (before and during pregnancy) (TC) self-rated health: a524, b040 (up to preg), b041 b042 during, pa064; asthma pa173a d153a, eczema pa174a d154a, any allergy pa210 d190, diabetes d040
* Maternal and paternal stressful life events in pregnancy (TC) b613 pb202
* Maternal and paternal exposure to harmful substances or radiation (e.g. pesticides; mother/father) (TC) C768 PTNR in contact with fumes or chemicals, b560 chemicals/fumes during first trimester
* Maternal and paternal mental health (e.g. diagnosis of anxiety or depression; mother, father) (TC) schizophrenia  pa189a d169a, eating disorder bulimia d152a pa172a anorexia pa190a d170a, severe depression pa191a d171a, other psychiatric problem pa192a d172a
* Maternal and paternal dietary supplements during pregnancy (TC) b140-b144, c110-c115
* Maternal and paternal height and weight (TC) dw021 dw002 paw002 paw009
* Maternal and paternal illegal drug use during pregnancy (TC)
b700-b714, pb088-pb098

Stressful life events: "These involved a life-events inventory of 42 items which were derived for the present study using previous inventories as a basis for selection of items. Three main sources were used in this way - those of Brown & Harris, Barnett et al and Stanley. Each life event had 5 response categories indicating not only whether or not the event occurred but also how greatly the respondent was affected by it. An additional open item in which the respondent was able to name any event which was not on the inventory was also included. On the basis of these items, two scores may be calculated, the number of life events occurring, and the perceived impact of life events. Analysis of pilot study results showed these two scores to be highly correlated, but the impact scored being a more sensitive predicator of emotional well-being.
Listed below are a number of events which may have brought changes in your life. Have any of these occurred since you became pregnant?"

### Pregnancy complications

* Hypertensive disorders of pregnancy (CH) v1dab5_hypertension
* Gestational diabetes (CH) pregnancy_diabetes

### Child's health

* Gestational age at delivery (CH) -bestgest
* Duration of breastfeeding (CH) - kb277 and kc403 (age stopped), kb280 and kc404 (duration)
* Child's physical activity (TC) - difficult to derive (would need to develop a weighted score?) and can't find anything that has already been derived
* Child's own smoking (TC) - not available before 14
* Child's own drinking of alcohol (TC) - from FFQs completed by mother at 3 (kg475), 4 (kk800), 7 (kq885) and 9 (kt6400)
* Child's own consumption of caffeine (TC) - from child-based questionnaires to mother at multiple timepoints: 
  + kb388 CH had coffee
  + kb385 had tea
  + kb361 had cola
  + kc542 CH had coffee
  + kc539 had tea
  + kb361 had cola
  + ke340 CH had coffee
  + ke338 had tea
  + ke320 had cola
  + kg423 times per week coffee
  + kg420 times per week tea
  + kg369a times per week cola
  + kk645 coffee
  + kk640 tea
  + kk598 cola
  + kq830 coffee
  + kq823 tea
  + kq798a cola
  + kt6290 coffee
  + kt6280 tea
  + kt6193 cola
  
* Child's exposure to passive smoke (CH): 
  + kc363	Degree of weekly exposure to smoke
  + ke195a	Wkday exposure to smoking
  + ke196a	Wkend exposure to smoking
  + kb549	Ch known to be in smokers rm on wkdays
  + kb551	Ch known to be in smokers rm at wkends
  + kg174	DV: Child's Exposure to Tobacco Smoke
  + kk312	DV: Child's Exposure to Tobacco Smoke
  + km3030	C4i: Time child in room with smokers on weekdays
  + km3031	C4ii: Time child in room with smokers at weekends
  + kp5090	E5i: Frequency child in room with smokers - weekdays
  + kp5091	E5ii: Frequency child in room with smokers - weekend days
  + kt1210	A8i: During weekdays number of hours child in a smoky room
  + kt1211	A8ii: During weekend days number of hours child in a smoky room
  
## Set up meta-data

```{r eval=FALSE, include=TRUE}
cov_vars <- c("c800","c801","c804","pa910","pc996","c666a","c645a","c755","c765","b032","a525","pa065","a522","a524","b040","b041","b042","pa064","pa173a","d153a","pa174a","d154a","pa210","d190","d040","b613","pb202","c768","b560","pa189a","d169a","d152a","pa172a","pa190a","d170a","pa191a","d171a","pa192a","d172a","b140","b141","b142","b143","b144","b149","c110","c111","c112","c113","c114","c115","dw021","dw002","paw002","paw010","b700","b701","b702","b706","b707","b708","b709","b710","b711","b712","b713","b714","pb088","pb089","pb090","pb091","pb092","pb093","pb094","pb095","pb096","pb097","pb098","bestgest","kb277","kc403","kb280","kc404","kg475","kk800","kq885","kt6400","HDP","pregnancy_diabetes","kb390","kb390a","kb387","kb387a","kc514", "kc544","kc541","kc572","kc574","kc570","ke341","ke350","ke352","ke354","ke339","ke321","kc363","ke195","ke195a","ke196","ke196a","kb549","kb551","kb388","kb385","kb361","kc542","kc539","kc361","ke340","ke338","ke320","kg423","kg420","kg369a","kk645","kk640","kk598","kq830","kq823","kq798a","kt6290","kt6280","kt6193","kg174","kk312","km3030","km3031","kp5090","kp5091","kt1210","kt1211","kc512")

covariates_meta_data <- findVars(unique(cov_vars)) #create a table with all the necessary meta data for each variable

covariates_meta_data<- subset(covariates_meta_data,subset=name %in% cov_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings

#Some variable names are duplicated, so we have to select the ones we're interested in:
covariates_meta_data<-covariates_meta_data[which(covariates_meta_data$cat2 %in% c("Quest","bestgest","Other")),]

```

## Extract variables
```{r eval=FALSE, include=TRUE}
covs_data <- extractVars(covariates_meta_data[covariates_meta_data$name %in% unique(cov_vars),]) #extracting individual level data and performing withdrawal of consent

covs_data<-covs_data[,c("aln","qlet",cov_vars)]
dat <- left_join(dat,covs_data,by=c("aln","qlet")) 
```

## Derive clean variables

### Demographics
```{r eval=FALSE, include=TRUE}
dat$covs_ethnicity_mother_binary <- NA
dat$covs_ethnicity_mother_binary[dat$c800==1]<-"white"
dat$covs_ethnicity_mother_binary[dat$c800>1]<-"not white"

dat$covs_ethnicity_father_binary <- NA
dat$covs_ethnicity_father_binary[dat$c801==1]<-"white"
dat$covs_ethnicity_father_binary[dat$c801>1]<-"not white"

dat$covs_ethnicity_child_binary <- NA
dat$covs_ethnicity_child_binary[dat$c804==1]<-"white"
dat$covs_ethnicity_child_binary[dat$c804==2]<-"not white"

dat$covs_age_father_pregnancy <-dat$pa910
dat$covs_age_father_pregnancy[dat$covs_age_father_pregnancy<0]<-NA

dat$covs_edu_father <- NA
dat$covs_edu_father[dat$c666a==1]<-0 #CSE/none
dat$covs_edu_father[dat$c666a%in%c(2,3)]<-1 #O-level or vocational
dat$covs_edu_father[dat$c666a==4]<-2 #A-level
dat$covs_edu_father[dat$c666a==5]<-3 #Degree

dat$covs_edu_mother <- NA
dat$covs_edu_mother[dat$c645a==1]<-0 #CSE/none
dat$covs_edu_mother[dat$c645a%in%c(2,3)]<-1 #O-level or vocational
dat$covs_edu_mother[dat$c645a==4]<-2 #A-level
dat$covs_edu_mother[dat$c645a==5]<-3 #Degree

dat$covs_occup_mother <- NA
dat$covs_occup_mother[dat$c755%in%c(1,2)]<-3 #professional or managerial/technica
dat$covs_occup_mother[dat$c755==3]<-2 #skilled non-manual
dat$covs_occup_mother[dat$c755==4]<-1 #skilled manual
dat$covs_occup_mother[dat$c755%in%c(5,6)]<-0 #semi-skilled or unskilled

dat$covs_occup_father <- NA
dat$covs_occup_father[dat$c765%in%c(1,2)]<-3 #professional or managerial/technica
dat$covs_occup_father[dat$c765==3]<-2 #skilled non-manual
dat$covs_occup_father[dat$c765==4]<-1 #skilled manual
dat$covs_occup_father[dat$c765%in%c(5,6)]<-0 #semi-skilled or unskilled

dat$covs_parity_mother_binary <-NA
dat$covs_parity_mother_binary[dat$b032==0]<-0
dat$covs_parity_mother_binary[dat$b032>0]<-1
```

### Family relationships
```{r eval=FALSE, include=TRUE}
dat$covs_married_mother_binary<-NA
dat$covs_married_mother_binary[dat$a525%in%c(1,2,3,4)]<-0
dat$covs_married_mother_binary[dat$a525%in%c(5,6)]<-1

dat$covs_married_father_binary<-NA
dat$covs_married_father_binary[dat$pa065%in%c(1,2,3,4)]<-0
dat$covs_married_father_binary[dat$pa065%in%c(5,6)]<-1
```

### Parents' health
```{r eval=FALSE, include=TRUE}
#self-rated health had very low numbers reporting anything other than always or usually well, so not enough variation to be useful

dat$covs_asthma_father_binary <- NA
dat$covs_asthma_father_binary[dat$pa173a==2]<-0
dat$covs_asthma_father_binary[dat$pa173a==1]<-1

dat$covs_asthma_mother_binary <- NA
dat$covs_asthma_mother_binary[dat$d153a==2]<-0
dat$covs_asthma_mother_binary[dat$d153a==1]<-1

dat$covs_eczema_father_binary <- NA
dat$covs_eczema_father_binary[dat$pa174a==2]<-0
dat$covs_eczema_father_binary[dat$pa174a==1]<-1

dat$covs_eczema_mother_binary <- NA
dat$covs_eczema_mother_binary[dat$d154a==2]<-0
dat$covs_eczema_mother_binary[dat$d154a==1]<-1

dat$covs_allergy_father_binary <- NA
dat$covs_allergy_father_binary[dat$pa210==2]<-0
dat$covs_allergy_father_binary[dat$pa210==1]<-1

dat$covs_allergy_mother_binary <- NA
dat$covs_allergy_mother_binary[dat$d190==2]<-0
dat$covs_allergy_mother_binary[dat$d190==1]<-1

#diabetes can be derived from the obstetric records

dat$covs_stressfulevents_mother <-dat$b613
dat$covs_stressfulevents_mother[dat$b613<0]<-NA
dat$covs_stressfulevents_mother_zscore <- scale(dat$covs_stressfulevents_mother)

dat$covs_stressfulevents_father <-dat$pb202
dat$covs_stressfulevents_father[dat$pb202<0]<-NA
dat$covs_stressfulevents_father_zscore <- scale(dat$covs_stressfulevents_father)

#confusion over paternal variable name for exposure to fumes/chemicals (c768 extracted but doesn't appear in the documentation) so not included here


dat$covs_mother_depression_binary <- NA #severe depression
dat$covs_mother_depression_binary[dat$d171a==0]<-0
dat$covs_mother_depression_binary[dat$d171a>0]<-1

dat$covs_father_depression_binary <- NA
dat$covs_father_depression_binary[dat$pa191a==0]<-0
dat$covs_father_depression_binary[dat$pa191a>0]<-1

dat$covs_supplements_mother_binary <-NA
temp <- dat[,c("b140","b141","b142","b143","b144","b149")]
temp[temp==2]<-0
temp[temp<0]<-NA
temp1 <- rowSums(temp,na.rm = T)
temp1[apply(temp,1,function(x) all(is.na(x)))]<-NA
dat$covs_supplements_mother_binary[temp1==0]<-0
dat$covs_supplements_mother_binary[temp1>0]<-1
rm(temp,temp1)
# took the earlier time point when supplements are more necessary

dat$covs_height_mother <-dat$dw021
dat$covs_weight_mother <-dat$dw002
dat$covs_height_mother[dat$dw021<1] <- NA
dat$covs_weight_mother[dat$dw002<1] <- NA
dat$covs_bmi_mother <- dat$covs_weight_mother/(dat$covs_height_mother/100)^2

dat$covs_bmi_mother_zscore <-scale(dat$covs_bmi_mother)

dat$covs_height_father <-dat$paw010
dat$covs_weight_father <-dat$paw002
dat$covs_height_father[dat$paw010<1] <- NA
dat$covs_weight_father[dat$paw002<1] <- NA
dat$covs_bmi_father <- dat$covs_weight_father/(dat$covs_height_father/100)^2

dat$covs_bmi_father_zscore <-scale(dat$covs_bmi_father)

#drugs not including cannabis, only 57 women
dat$covs_drugs_mother_binary <-NA
dat$covs_drugs_mother_binary[which(dat$b706==4|dat$b707==4|dat$b708==4|dat$b709==4|dat$b710==4|dat$b711==4|dat$b712==4|dat$b713==4)]<-0
dat$covs_drugs_mother_binary[which(dat$b706%in%c(1,2,3)|dat$b707%in%c(1,2,3)|dat$b708%in%c(1,2,3)|dat$b709%in%c(1,2,3)|dat$b710%in%c(1,2,3)|dat$b711%in%c(1,2,3)|dat$b712%in%c(1,2,3)|dat$b713%in%c(1,2,3))]<-1

dat$covs_drugs_father_binary <-NA
dat$covs_drugs_father_binary[which(dat$pb090==4|dat$pb091==4|dat$pb092==4|dat$pb093==4|dat$pb094==4|dat$pb095==4|dat$pb096==4|dat$pb097==4)]<-0
dat$covs_drugs_father_binary[which(dat$pb090%in%c(1,2,3)|dat$pb091%in%c(1,2,3)|dat$pb092%in%c(1,2,3)|dat$pb093%in%c(1,2,3)|dat$pb094%in%c(1,2,3)|dat$pb095%in%c(1,2,3)|dat$pb096%in%c(1,2,3)|dat$pb097%in%c(1,2,3))]<-1
```

### Pregnancy complications
```{r eval=FALSE, include=TRUE}

# didn't appear in meta_data last time I ran this, so may have to read in file and add manually, like this:
obstet <- read_dta("/Volumes/sscm alspac/Data/Current/Other/Obstetric/OA_r1b.dta")
obstet <- obstet[,c("aln","qlet","HDP","pregnancy_diabetes")]
dat <- left_join(dat,obstet,by=c("aln","qlet"))

dat$covs_hdp_binary<-dat$HDP
dat$covs_hdp_binary[dat$covs_hdp_binary<0]<-NA

#glycosuria, existing diabetes or gestational diabetes vs no glycosuria or diabetes
dat$covs_glycosuria_binary<-NA
dat$covs_glycosuria_binary[dat$pregnancy_diabetes==0]<-0
dat$covs_glycosuria_binary[dat$pregnancy_diabetes%in%c(1,2,3)]<-1
```

### Child's health
```{r eval=FALSE, include=TRUE}
dat$covs_gestage <- dat$bestgest

dat$covs_preterm_binary <- NA
dat$covs_preterm_binary[dat$covs_gestage<37]<-1
dat$covs_preterm_binary[dat$covs_gestage>=37]<-0

dat$covs_breastfeeding_ordinal <- NA
dat$covs_breastfeeding_ordinal[dat$kb280==1] <-0 #never
dat$covs_breastfeeding_ordinal[dat$kb280==2] <-1 #less than 1 month (combined to make less than 3m - matches categories for kc404)
dat$covs_breastfeeding_ordinal[dat$kb280==3] <- 1 #1-<3 months (combined to make less than 3m - matches categories for kc404)
dat$covs_breastfeeding_ordinal[dat$kb280==4] <- 2 #3-<6 months
dat$covs_breastfeeding_ordinal[dat$kb280==5]<- 3 #6 or more months
dat$covs_breastfeeding_ordinal[dat$kc404==0] <- 0
dat$covs_breastfeeding_ordinal[dat$kc404==1] <- 1
dat$covs_breastfeeding_ordinal[dat$kc404==2] <- 2
dat$covs_breastfeeding_ordinal[dat$kc404==3] <- 3

dat$covs_alcohol_child_stage2_binary<-NA
dat$covs_alcohol_child_stage2_binary[which(dat$kg475==0|dat$kk800==0)]<-0
dat$covs_alcohol_child_stage2_binary[which(dat$kg475>0|dat$kk800>0)]<-1

dat$covs_alcohol_child_stage3_binary<-NA
dat$covs_alcohol_child_stage3_binary[dat$kq885==0]<-0
dat$covs_alcohol_child_stage3_binary[dat$kq885>0]<-1

dat$covs_alcohol_child_stage4_binary<-NA
dat$covs_alcohol_child_stage4_binary[dat$kt6400==0]<-0
dat$covs_alcohol_child_stage4_binary[dat$kt6400>0]<-1

dat$covs_alcohol_child_any_binary <- NA
dat$covs_alcohol_child_any_binary[dat$covs_alcohol_child_stage2_binary==0|dat$covs_alcohol_child_stage3_binary==0|dat$covs_alcohol_child_stage4_binary==0]<-0
dat$covs_alcohol_child_any_binary[dat$covs_alcohol_child_stage2_binary==1|dat$covs_alcohol_child_stage3_binary==1|dat$covs_alcohol_child_stage4_binary==1]<-1

# difficult to derive a continuous or ordinal measure for caffeine intake, so have used a binary variable for each type of drink
dat$covs_coffee_child_stage0_binary<-NA
dat$covs_coffee_child_stage0_binary[dat$kb388==1]<-0
dat$covs_coffee_child_stage0_binary[dat$kb388==2]<-1

dat$covs_tea_child_stage0_binary<-NA
dat$covs_tea_child_stage0_binary[dat$kb385==1]<-0
dat$covs_tea_child_stage0_binary[dat$kb385==2]<-1

dat$covs_cola_child_stage0_binary<-NA
dat$covs_cola_child_stage0_binary[dat$kb361==1]<-0
dat$covs_cola_child_stage0_binary[dat$kb361==2]<-1

dat$covs_caffeinedrinks_child_stage0_binary <-NA
dat$covs_caffeinedrinks_child_stage0_binary[dat$covs_coffee_child_stage0_binary==0&dat$covs_tea_child_stage0_binary==0&dat$covs_cola_child_stage0_binary==0] <-0
dat$covs_caffeinedrinks_child_stage0_binary[dat$covs_coffee_child_stage0_binary==1|dat$covs_tea_child_stage0_binary==1|dat$covs_cola_child_stage0_binary==1] <-1

dat$covs_coffee_child_stage1_binary<-NA
dat$covs_coffee_child_stage1_binary[dat$kc542==1|dat$ke340==1]<-0
dat$covs_coffee_child_stage1_binary[dat$kc542==2|dat$ke340==2]<-1

dat$covs_tea_child_stage1_binary<-NA
dat$covs_tea_child_stage1_binary[dat$kc539==1|dat$ke338==1]<-0
dat$covs_tea_child_stage1_binary[dat$kc539==2|dat$ke338==2]<-1

dat$covs_cola_child_stage1_binary<-NA
dat$covs_cola_child_stage1_binary[dat$kc512==1|dat$ke320==1]<-0
dat$covs_cola_child_stage1_binary[dat$kc512==2|dat$ke320==2]<-1

dat$covs_caffeinedrinks_child_stage1_binary <-NA
dat$covs_caffeinedrinks_child_stage1_binary[dat$covs_coffee_child_stage1_binary==0&dat$covs_tea_child_stage1_binary==0&dat$covs_cola_child_stage1_binary==0] <-0
dat$covs_caffeinedrinks_child_stage1_binary[dat$covs_coffee_child_stage1_binary==1|dat$covs_tea_child_stage1_binary==1|dat$covs_cola_child_stage1_binary==1] <-1

dat$covs_coffee_child_stage2_binary<-NA
dat$covs_coffee_child_stage2_binary[which(dat$kg423==0|dat$kk645==2)]<-0
dat$covs_coffee_child_stage2_binary[which(dat$kg423>0|dat$kk645==1)]<-1

dat$covs_tea_child_stage2_binary<-NA
dat$covs_tea_child_stage2_binary[which(dat$kg420==0|dat$kk640==2)]<-0
dat$covs_tea_child_stage2_binary[which(dat$kg420>0|dat$kk640==1)]<-1

dat$covs_cola_child_stage2_binary<-NA
dat$covs_cola_child_stage2_binary[which(dat$kg369a==1|dat$kk598==1)]<-0
dat$covs_cola_child_stage2_binary[which(dat$kg369a>1|dat$kk598>1)]<-1

dat$covs_caffeinedrinks_child_stage2_binary <-NA
dat$covs_caffeinedrinks_child_stage2_binary[dat$covs_coffee_child_stage2_binary==0&dat$covs_tea_child_stage2_binary==0&dat$covs_cola_child_stage2_binary==0] <-0
dat$covs_caffeinedrinks_child_stage2_binary[dat$covs_coffee_child_stage2_binary==1|dat$covs_tea_child_stage2_binary==1|dat$covs_cola_child_stage2_binary==1] <-1

dat$covs_coffee_child_stage3_binary<-NA
dat$covs_coffee_child_stage3_binary[dat$kq830==1]<-1
dat$covs_coffee_child_stage3_binary[dat$kq830==2]<-0

dat$covs_tea_child_stage3_binary<-NA
dat$covs_tea_child_stage3_binary[dat$kq823==1]<-1
dat$covs_tea_child_stage3_binary[dat$kq823==2]<-0

dat$covs_cola_child_stage3_binary<-NA
dat$covs_cola_child_stage3_binary[dat$kq798a==1]<-1
dat$covs_cola_child_stage3_binary[dat$kq798a==2]<-0

dat$covs_caffeinedrinks_child_stage3_binary <-NA
dat$covs_caffeinedrinks_child_stage3_binary[dat$covs_coffee_child_stage3_binary==0&dat$covs_tea_child_stage3_binary==0&dat$covs_cola_child_stage3_binary==0] <-0
dat$covs_caffeinedrinks_child_stage3_binary[dat$covs_coffee_child_stage3_binary==1|dat$covs_tea_child_stage3_binary==1|dat$covs_cola_child_stage3_binary==1] <-1

dat$covs_coffee_child_stage4_binary<-NA
dat$covs_coffee_child_stage4_binary[dat$kt6290==1]<-1
dat$covs_coffee_child_stage4_binary[dat$kt6290==2]<-0

dat$covs_tea_child_stage4_binary<-NA
dat$covs_tea_child_stage4_binary[dat$kt6280==1]<-1
dat$covs_tea_child_stage4_binary[dat$kt6280==2]<-0

dat$covs_cola_child_stage4_binary<-NA
dat$covs_cola_child_stage4_binary[dat$kt6193==1]<-0
dat$covs_cola_child_stage4_binary[dat$kt6193>1]<-1

dat$covs_caffeinedrinks_child_stage4_binary <-NA
dat$covs_caffeinedrinks_child_stage4_binary[dat$covs_coffee_child_stage4_binary==0&dat$covs_tea_child_stage4_binary==0&dat$covs_cola_child_stage4_binary==0] <-0
dat$covs_caffeinedrinks_child_stage4_binary[dat$covs_coffee_child_stage4_binary==1|dat$covs_tea_child_stage4_binary==1|dat$covs_cola_child_stage4_binary==1] <-1

dat$covs_caffeinedrinks_child_before2_binary <-NA
dat$covs_caffeinedrinks_child_before2_binary[dat$covs_caffeinedrinks_child_stage0_binary==0|dat$covs_caffeinedrinks_child_stage1_binary==0] <-0
dat$covs_caffeinedrinks_child_before2_binary[dat$covs_caffeinedrinks_child_stage0_binary==1|dat$covs_caffeinedrinks_child_stage1_binary==1] <-1

dat$covs_passivesmk_child_stage0_binary <- NA
dat$covs_passivesmk_child_stage0_binary[which(dat$kb549==2&dat$kb551==2)]<-0
dat$covs_passivesmk_child_stage0_binary[which(dat$kb549==1&dat$kb551==1)]<-1

dat$covs_passivesmk_child_stage1_binary <- NA
dat$covs_passivesmk_child_stage1_binary[which(dat$kc363==0|(dat$ke196a==2&dat$ke195a==2))]<-0
dat$covs_passivesmk_child_stage1_binary[which(dat$kc363>0|(dat$ke196a==1|dat$ke195a==1))]<-1

dat$covs_passivesmk_child_stage2_binary <- NA
dat$covs_passivesmk_child_stage2_binary[which(dat$kg174==1|dat$kk312==1)]<-0
dat$covs_passivesmk_child_stage2_binary[which(dat$kg174>1|dat$kk312>1)]<-1

dat$covs_passivesmk_child_stage3_binary <- NA
dat$covs_passivesmk_child_stage3_binary[which((dat$kp5090==6&dat$kp5091==6)|(dat$km3030==6&dat$km3031==6))]<-0
dat$covs_passivesmk_child_stage3_binary[which((dat$kp5090%in%c(1:5)|dat$kp5091%in%c(1:5))|(dat$km3030%in%c(1:5)|dat$km3031%in%c(1:5)))]<-1

dat$covs_passivesmk_child_stage4_binary <- NA
dat$covs_passivesmk_child_stage4_binary[which(dat$kt1210==6&dat$kt1211==6)]<-0
dat$covs_passivesmk_child_stage4_binary[which(dat$kt1210%in%c(1:5)&dat$kt1211%in%c(1:5))]<-1

dat$covs_passivesmk_child_before2_binary <-NA
dat$covs_passivesmk_child_before2_binary[dat$covs_passivesmk_child_stage0_binary==0|dat$covs_passivesmk_child_stage1_binary==0] <-0
dat$covs_passivesmk_child_before2_binary[dat$covs_passivesmk_child_stage0_binary==1|dat$covs_passivesmk_child_stage1_binary==1] <-1


```

## Remove original variables
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%cov_vars)]
```

# Negative control outcomes  {.tabset}

## Ideal variables

### Pigeon infestation

* g572	Home invaded by pigeons
* g572a	Home invaded by pigeons
* h462	G10c: Home invaded by pigeons
* h462a	DV: Home invaded by pigeons, Y/N

### Mice infestation

* g571	Home invaded by mice
* g571a	Home invaded by mice
* h461	G10b: Home invaded by mice
* h461a	DV: Home invaded by mice, Y/N

### Left handedness

* kf527	Child uses right hand to draw
* kf528	Child uses left hand to draw
* kj517	CH Uses Right Hand to Draw
* kj518	CH Uses Left Hand to Draw

## Set up meta-data

```{r eval=FALSE, include=TRUE}
negcon_vars <- c("g572","g572a","h462","h462a","g571","g571a","h461","h461a","kf527","kf528","kj517","kj518")

negcon_meta_data <- findVars(unique(negcon_vars)) #create a table with all the necessary meta data for each variable

negcon_meta_data<- subset(negcon_meta_data,subset=tolower(name) %in% negcon_vars) #necessary to make sure we only have the variables we're interested in, not variables with longer names that include the same strings
```

## Extract variables
```{r eval=FALSE, include=TRUE}
negcon_data <- extractVars(negcon_meta_data[negcon_meta_data$name %in% unique(negcon_vars),]) #extracting individual level data and performing withdrawal of consent

negcon_data<-negcon_data[,c("aln","qlet",negcon_vars)]
dat <- left_join(dat,negcon_data,by=c("aln","qlet")) 
```
## Derive clean variables

### Pigeon infestation
```{r eval=FALSE, include=TRUE}
dat$negcon_pigeons_binary <-NA
dat$negcon_pigeons_binary[which(dat$g572a==2|dat$h462a==2)]<-0
dat$negcon_pigeons_binary[which(dat$g572a==1|dat$h462a==1)]<-1
```

### Mice infestation
```{r eval=FALSE, include=TRUE}
dat$negcon_mice_binary <-NA
dat$negcon_mice_binary[which(dat$g571a==2|dat$h461a==2)]<-0
dat$negcon_mice_binary[which(dat$g571a==1|dat$h461a==1)]<-1
```

### Left handedness (to draw)
```{r eval=FALSE, include=TRUE}
dat$negcon_lefthand_binary <-NA
dat$negcon_lefthand_binary[which(dat$kf528%in%c(2,3)|dat$kj518%in%c(2,3))]<-0 #not very well or hasn't done yet
dat$negcon_lefthand_binary[which(dat$kf528==1|dat$kj518==1)]<-1#does well
```

## Remove original variables
```{r eval=FALSE, include=TRUE}
dat<-dat[,-which(colnames(dat)%in%negcon_vars)]
```

# Polygenic risk scores (PRS) {.tabset}

## GWAS info

**Alcohol and tobacco**

* Alcohol and tobacco SNPs were taken from Liu et al 2019 'Association Studies of Up to 1.2 Million Individuals Yield New Insights Into the Genetic Etiology of Tobacco and Alcohol Use'

There are 4 different exposures for smoking:
1)	Cigarettes per day. Categorical (5)
2)	Smoking initiation. Binary (Regular current V non smoker)
3)	Cessation. Binary (Regular current V former smoker)
4)	Age of initiation (Age first regularly smoked)

NB. The Smoking initiation one is problematic, and seems to be related to impulsive behaviour instead

**Caffeine**

* Caffeine SNPs were taken from Cornelis et al 2015 (same as Treur et al) 'Genome-wide meta-analysis identifies six novel loci associated with habitual coffee consumption'

**Physical activity** 

* Physical activity SNPs were taken from Doherty et al 2018 'GWAS identifies 14 loci for device-measured physical activity and sleep duration'

* There are 2 different exposures for physical acitivity:
    1) Sedentary behaviour
    2) Overall activity

* PRS for physical activity were only derived for 'sedentary', there are SNPs relared to 'overall activity' but the n(=3) is so small it is unlikely to be worthwhilse making PRS. The SNP n for sedentary is also very small (n=4) to note .


## Read in data
```{r eval=FALSE, include=TRUE}
prs_alcohol_child<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200429_scores_alcohol_children.profile", header=TRUE,stringsAsFactors = F)
prs_alcohol_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200429_scores_alcohol_mothers.profile", header=TRUE,stringsAsFactors = F)

prs_caffeine_child<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200504_scores_caffeine_children.profile", header=TRUE,stringsAsFactors = F)
prs_caffeine_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200504_scores_caffeine_mothers.profile", header=TRUE,stringsAsFactors = F)

prs_smoking_age_init_child<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200502_scores_smoking_age_init_children.profile", header=TRUE,stringsAsFactors = F)
prs_smoking_age_init_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200502_scores_smoking_age_init_mothers.profile", header=TRUE,stringsAsFactors = F)

prs_smoking_cessation_child<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200501_scores_smoking_cessation_children.profile", header=TRUE,stringsAsFactors = F)
prs_smoking_cessation_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200501_scores_smoking_cessation_mothers.profile", header=TRUE,stringsAsFactors = F)

prs_smoking_cigs_pd_child<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200430_scores_smoking_cigs_pd_children.profile", header=TRUE,stringsAsFactors = F)
prs_smoking_cigs_pd_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200430_scores_smoking_cigs_pd_mothers.profile", header=TRUE,stringsAsFactors = F)

prs_smoking_initiation_child<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200501_scores_smoking_initiation_children.profile", header=TRUE,stringsAsFactors = F)
prs_smoking_initiation_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200501_scores_smoking_initiation_mothers.profile", header=TRUE,stringsAsFactors = F)

prs_pa_sedentary_child<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200507_scores_pa_sedentary_children.profile", header=TRUE,stringsAsFactors = F)
prs_pa_sedentary_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/20200506_scores_pa_sedentary_mothers.profile", header=TRUE,stringsAsFactors = F)

```

## Clean variables

Create aln, qlet and rename score column
```{r eval=FALSE, include=TRUE}
prs_alcohol_child$aln <-as.numeric(substr(prs_alcohol_child$FID, 1, 5))
prs_alcohol_child$qlet<-substr(prs_alcohol_child$FID, 6, 6)
prs_alcohol_child$prs_score_child_alcohol<-prs_alcohol_child$SCORE
## keep only the columns we want
prs_alcohol_child<-prs_alcohol_child[, which(names(prs_alcohol_child) %in% c("aln", "qlet","prs_score_child_alcohol"))]

prs_alcohol_mother$aln <-as.numeric(substr(prs_alcohol_mother$FID, 1, 5))
prs_alcohol_mother$qlet<-substr(prs_alcohol_mother$FID, 6, 6)
prs_alcohol_mother$prs_score_mother_alcohol<-prs_alcohol_mother$SCORE
prs_alcohol_mother<-prs_alcohol_mother[, which(names(prs_alcohol_mother) %in% c("aln", "qlet","prs_score_mother_alcohol"))]

prs_caffeine_child$aln <-as.numeric(substr(prs_caffeine_child$FID, 1, 5))
prs_caffeine_child$qlet<-substr(prs_caffeine_child$FID, 6, 6)
prs_caffeine_child$prs_score_child_caffeine<-prs_caffeine_child$SCORE
prs_caffeine_child<-prs_caffeine_child[, which(names(prs_caffeine_child) %in% c("aln", "qlet","prs_score_child_caffeine"))]

prs_caffeine_mother$aln <-as.numeric(substr(prs_caffeine_mother$FID, 1, 5))
prs_caffeine_mother$qlet<-substr(prs_caffeine_mother$FID, 6, 6)
prs_caffeine_mother$prs_score_mother_caffeine<-prs_caffeine_mother$SCORE
prs_caffeine_mother<-prs_caffeine_mother[, which(names(prs_caffeine_mother) %in% c("aln", "qlet","prs_score_mother_caffeine"))]

prs_smoking_age_init_child$aln <-as.numeric(substr(prs_smoking_age_init_child$FID, 1, 5))
prs_smoking_age_init_child$qlet<-substr(prs_smoking_age_init_child$FID, 6, 6)
prs_smoking_age_init_child$prs_score_child_smoking_age_init<-prs_smoking_age_init_child$SCORE
prs_smoking_age_init_child<-prs_smoking_age_init_child[, which(names(prs_smoking_age_init_child) %in% c("aln", "qlet","prs_score_child_smoking_age_init"))]

prs_smoking_age_init_mother$aln <-as.numeric(substr(prs_smoking_age_init_mother$FID, 1, 5))
prs_smoking_age_init_mother$qlet<-substr(prs_smoking_age_init_mother$FID, 6, 6)
prs_smoking_age_init_mother$prs_score_mother_smoking_age_init<-prs_smoking_age_init_mother$SCORE
prs_smoking_age_init_mother<-prs_smoking_age_init_mother[, which(names(prs_smoking_age_init_mother) %in% c("aln", "qlet","prs_score_mother_smoking_age_init"))]

prs_smoking_cessation_child$aln <-as.numeric(substr(prs_smoking_cessation_child$FID, 1, 5))
prs_smoking_cessation_child$qlet<-substr(prs_smoking_cessation_child$FID, 6, 6)
prs_smoking_cessation_child$prs_score_child_smoking_cessation<-prs_smoking_cessation_child$SCORE
prs_smoking_cessation_child<-prs_smoking_cessation_child[, which(names(prs_smoking_cessation_child) %in% c("aln", "qlet","prs_score_child_smoking_cessation"))]

prs_smoking_cessation_mother$aln <-as.numeric(substr(prs_smoking_cessation_mother$FID, 1, 5))
prs_smoking_cessation_mother$qlet<-substr(prs_smoking_cessation_mother$FID, 6, 6)
prs_smoking_cessation_mother$prs_score_mother_smoking_cessation<-prs_smoking_cessation_mother$SCORE
prs_smoking_cessation_mother<-prs_smoking_cessation_mother[, which(names(prs_smoking_cessation_mother) %in% c("aln", "qlet","prs_score_mother_smoking_cessation"))]

prs_smoking_cigs_pd_child$aln <-as.numeric(substr(prs_smoking_cigs_pd_child$FID, 1, 5))
prs_smoking_cigs_pd_child$qlet<-substr(prs_smoking_cigs_pd_child$FID, 6, 6)
prs_smoking_cigs_pd_child$prs_score_child_smoking_cigs_pd<-prs_smoking_cigs_pd_child$SCORE
prs_smoking_cigs_pd_child<-prs_smoking_cigs_pd_child[, which(names(prs_smoking_cigs_pd_child) %in% c("aln", "qlet","prs_score_child_smoking_cigs_pd"))]

prs_smoking_cigs_pd_mother$aln <-as.numeric(substr(prs_smoking_cigs_pd_mother$FID, 1, 5))
prs_smoking_cigs_pd_mother$qlet<-substr(prs_smoking_cigs_pd_mother$FID, 6, 6)
prs_smoking_cigs_pd_mother$prs_score_mother_smoking_cigs_pd<-prs_smoking_cigs_pd_mother$SCORE
prs_smoking_cigs_pd_mother<-prs_smoking_cigs_pd_mother[, which(names(prs_smoking_cigs_pd_mother) %in% c("aln", "qlet","prs_score_mother_smoking_cigs_pd"))]

prs_smoking_initiation_child$aln <-as.numeric(substr(prs_smoking_initiation_child$FID, 1, 5))
prs_smoking_initiation_child$qlet<-substr(prs_smoking_initiation_child$FID, 6, 6)
prs_smoking_initiation_child$prs_score_child_smoking_initiation<-prs_smoking_initiation_child$SCORE
prs_smoking_initiation_child<-prs_smoking_initiation_child[, which(names(prs_smoking_initiation_child) %in% c("aln", "qlet","prs_score_child_smoking_initiation"))]

prs_smoking_initiation_mother$aln <-as.numeric(substr(prs_smoking_initiation_mother$FID, 1, 5))
prs_smoking_initiation_mother$qlet<-substr(prs_smoking_initiation_mother$FID, 6, 6)
prs_smoking_initiation_mother$prs_score_mother_smoking_initiation<-prs_smoking_initiation_mother$SCORE
prs_smoking_initiation_mother<-prs_smoking_initiation_mother[, which(names(prs_smoking_initiation_mother) %in% c("aln", "qlet","prs_score_mother_smoking_initiation"))]

prs_pa_sedentary_child$aln <-as.numeric(substr(prs_pa_sedentary_child$FID, 1, 5))
prs_pa_sedentary_child$qlet<-substr(prs_pa_sedentary_child$FID, 6, 6)
prs_pa_sedentary_child$prs_score_child_pa_sedentary<-prs_pa_sedentary_child$SCORE
prs_pa_sedentary_child<-prs_pa_sedentary_child[, which(names(prs_pa_sedentary_child) %in% c("aln", "qlet","prs_score_child_pa_sedentary"))]

prs_pa_sedentary_mother$aln <-as.numeric(substr(prs_pa_sedentary_mother$FID, 1, 5))
prs_pa_sedentary_mother$qlet<-substr(prs_pa_sedentary_mother$FID, 6, 6)
prs_pa_sedentary_mother$prs_score_mother_pa_sedentary<-prs_pa_sedentary_mother$SCORE
prs_pa_sedentary_mother<-prs_pa_sedentary_mother[, which(names(prs_pa_sedentary_mother) %in% c("aln", "qlet","prs_score_mother_pa_sedentary"))]
```

## Merge
```{r eval=FALSE, include=TRUE}
dat<-left_join(dat,prs_alcohol_child, by =c("aln","qlet"))
dat<-left_join(dat,prs_alcohol_mother[,-2], by =c("aln"))

dat<-left_join(dat,prs_caffeine_child, by =c("aln","qlet"))
dat<-left_join(dat,prs_caffeine_mother[,-2], by =c("aln"))

dat<-left_join(dat,prs_smoking_age_init_child, by =c("aln","qlet"))
dat<-left_join(dat,prs_smoking_age_init_mother[,-2], by =c("aln"))

dat<-left_join(dat,prs_smoking_cessation_child, by =c("aln","qlet"))
dat<-left_join(dat,prs_smoking_cessation_mother[,-2], by =c("aln"))

dat<-left_join(dat,prs_smoking_cigs_pd_child, by =c("aln","qlet"))
dat<-left_join(dat,prs_smoking_cigs_pd_mother[,-2], by =c("aln"))

dat<-left_join(dat,prs_smoking_initiation_child, by =c("aln","qlet"))
dat<-left_join(dat,prs_smoking_initiation_mother[,-2], by =c("aln"))

dat<-left_join(dat,prs_pa_sedentary_child, by =c("aln","qlet"))
dat<-left_join(dat,prs_pa_sedentary_mother[,-2], by =c("aln"))

```

# Individual SNP (buestguess) {.tabset}

## Read in data
```{r eval=FALSE, include=TRUE}
snps_alcohol_buestguess_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/alcohol_mothers_excluded_numb.raw", header=T)
snps_alcohol_buestguess_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/alcohol_children_excluded_numb.raw", header=T)

snps_caffeine_buestguess_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/caffeine_mothers_excluded_numb.raw", header=T)
snps_caffeine_buestguess_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/caffeine_children_excluded_numb.raw", header=T)

snps_smoking_age_init_buestguess_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_age_init_mothers_excluded_numb.raw", header=T)
snps_smoking_age_init_buestguess_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_age_init_children_excluded_numb.raw", header=T)

snps_smoking_cessation_buestguess_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_cessation_mothers_excluded_numb.raw", header=T)
snps_smoking_cessation_buestguess_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_cessation_children_excluded_numb.raw", header=T)

snps_smoking_cigs_pd_buestguess_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_cigs_pd_mothers_excluded_numb.raw", header=T)
snps_smoking_cigs_pd_buestguess_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_cigs_pd_children_excluded_numb.raw", header=T)

snps_smoking_initiation_buestguess_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_initiation_mothers_excluded_numb.raw", header=T)
snps_smoking_initiation_buestguess_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/smoking_initiation_children_excluded_numb.raw", header=T)

snps_pa_sedentary_buestguess_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/pa_sedentary_mothers_excluded_numb.raw", header=T)
snps_pa_sedentary_buestguess_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/pa_sedentary_children_excluded_numb.raw", header=T)

snps_pa_overall_activity_mother<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/pa_overall_activity_mothers_excluded_numb.raw", header=T)
snps_pa_overall_activity_children<-read.table("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/individual_snps/pa_overall_activity_children_excluded_numb.raw", header=T)
```

## Clean variables

```{r eval=FALSE, include=TRUE}
# (rename) add mother_snp to start of all snp columns
names(snps_alcohol_buestguess_mother)<-paste("mother_snp_alc_",names(snps_alcohol_buestguess_mother),sep="")
names(snps_alcohol_buestguess_children)<-paste("children_snp_alc_",names(snps_alcohol_buestguess_children),sep="")

names(snps_caffeine_buestguess_mother)<-paste("mother_snp_caff_",names(snps_caffeine_buestguess_mother),sep="")
names(snps_caffeine_buestguess_children)<-paste("children_snp_caff_",names(snps_caffeine_buestguess_children),sep="")

names(snps_smoking_age_init_buestguess_mother)<-paste("mother_snp_age_init_",names(snps_smoking_age_init_buestguess_mother),sep="")
names(snps_smoking_age_init_buestguess_children)<-paste("children_snp_age_init_",names(snps_smoking_age_init_buestguess_children),sep="")

names(snps_smoking_cessation_buestguess_mother)<-paste("mother_snp_cessation_",names(snps_smoking_cessation_buestguess_mother),sep="")
names(snps_smoking_cessation_buestguess_children)<-paste("children_snp_cessation_",names(snps_smoking_cessation_buestguess_children),sep="")

names(snps_smoking_cigs_pd_buestguess_mother)<-paste("mother_snp_cigspd_",names(snps_smoking_cigs_pd_buestguess_mother),sep="")
names(snps_smoking_cigs_pd_buestguess_children)<-paste("children_snp_cigspd_",names(snps_smoking_cigs_pd_buestguess_children),sep="")

names(snps_smoking_initiation_buestguess_mother)<-paste("mother_snp_initiation_",names(snps_smoking_initiation_buestguess_mother),sep="")
names(snps_smoking_initiation_buestguess_children)<-paste("children_snp_initiation_",names(snps_smoking_initiation_buestguess_children),sep="")

names(snps_pa_sedentary_buestguess_mother)<-paste("mother_snp_sedentary_",names(snps_pa_sedentary_buestguess_mother),sep="")
names(snps_pa_sedentary_buestguess_children)<-paste("children_snp_sedentary_",names(snps_pa_sedentary_buestguess_children),sep="")

names(snps_pa_overall_activity_mother)<-paste("mother_snp_overallact_",names(snps_pa_overall_activity_mother),sep="")
names(snps_pa_overall_activity_children)<-paste("children_snp_overallact_",names(snps_pa_overall_activity_children),sep="")
```

###  Create aln and qlet and delete columns dont need

**Maternal**

```{r eval=FALSE, include=TRUE}
snps_alcohol_buestguess_mother$aln <-as.numeric(substr(snps_alcohol_buestguess_mother$mother_snp_alc_FID, 1, 5))
snps_alcohol_buestguess_mother$qlet<-substr(snps_alcohol_buestguess_mother$mother_snp_alc_FID, 6, 6)
snps_alcohol_buestguess_mother<-snps_alcohol_buestguess_mother[-c(1:6)] 

snps_caffeine_buestguess_mother$aln <-as.numeric(substr(snps_caffeine_buestguess_mother$mother_snp_caff_FID, 1, 5))
snps_caffeine_buestguess_mother$qlet<-substr(snps_caffeine_buestguess_mother$mother_snp_caff_FID, 6, 6)
snps_caffeine_buestguess_mother<-snps_caffeine_buestguess_mother[-c(1:6)]

snps_smoking_age_init_buestguess_mother$aln <-as.numeric(substr(snps_smoking_age_init_buestguess_mother$mother_snp_age_init_FID, 1, 5))
snps_smoking_age_init_buestguess_mother$qlet<-substr(snps_smoking_age_init_buestguess_mother$mother_snp_age_init_FID, 6, 6)
snps_smoking_age_init_buestguess_mother<-snps_smoking_age_init_buestguess_mother[-c(1:6)]

snps_smoking_cessation_buestguess_mother$aln <-as.numeric(substr(snps_smoking_cessation_buestguess_mother$mother_snp_cessation_FID, 1, 5))
snps_smoking_cessation_buestguess_mother$qlet<-substr(snps_smoking_cessation_buestguess_mother$mother_snp_cessation_FID, 6, 6)
snps_smoking_cessation_buestguess_mother<-snps_smoking_cessation_buestguess_mother[-c(1:6)]

snps_smoking_cigs_pd_buestguess_mother$aln <-as.numeric(substr(snps_smoking_cigs_pd_buestguess_mother$mother_snp_cigspd_FID, 1, 5))
snps_smoking_cigs_pd_buestguess_mother$qlet<-substr(snps_smoking_cigs_pd_buestguess_mother$mother_snp_cigspd_FID, 6, 6)
snps_smoking_cigs_pd_buestguess_mother<-snps_smoking_cigs_pd_buestguess_mother[-c(1:6)]

snps_smoking_initiation_buestguess_mother$aln <-as.numeric(substr(snps_smoking_initiation_buestguess_mother$mother_snp_initiation_FID, 1, 5))
snps_smoking_initiation_buestguess_mother$qlet<-substr(snps_smoking_initiation_buestguess_mother$mother_snp_initiation_FID, 6, 6)
snps_smoking_initiation_buestguess_mother<-snps_smoking_initiation_buestguess_mother[-c(1:6)]

snps_pa_sedentary_buestguess_mother$aln <-as.numeric(substr(snps_pa_sedentary_buestguess_mother$mother_snp_sedentary_FID, 1, 5))
snps_pa_sedentary_buestguess_mother$qlet<-substr(snps_pa_sedentary_buestguess_mother$mother_snp_sedentary_FID, 6, 6)
snps_pa_sedentary_buestguess_mother<-snps_pa_sedentary_buestguess_mother[-c(1:6)]

snps_pa_overall_activity_mother$aln <-as.numeric(substr(snps_pa_overall_activity_mother$mother_snp_overallact_FID, 1, 5))
snps_pa_overall_activity_mother$qlet<-substr(snps_pa_overall_activity_mother$mother_snp_overallact_FID, 6, 6)
snps_pa_overall_activity_mother<-snps_pa_overall_activity_mother[-c(1:6)]

```

**Child**
```{r eval=FALSE, include=TRUE}
snps_alcohol_buestguess_children$aln <-as.numeric(substr(snps_alcohol_buestguess_children$children_snp_alc_FID, 1, 5))
snps_alcohol_buestguess_children$qlet<-substr(snps_alcohol_buestguess_children$children_snp_alc_FID, 6, 6)
snps_alcohol_buestguess_children<-snps_alcohol_buestguess_children[-c(1:6)]

snps_caffeine_buestguess_children$aln <-as.numeric(substr(snps_caffeine_buestguess_children$children_snp_caff_FID, 1, 5))
snps_caffeine_buestguess_children$qlet<-substr(snps_caffeine_buestguess_children$children_snp_caff_FID, 6, 6)
snps_caffeine_buestguess_children<-snps_caffeine_buestguess_children[-c(1:6)]

snps_smoking_age_init_buestguess_children$aln <-as.numeric(substr(snps_smoking_age_init_buestguess_children$children_snp_age_init_FID, 1, 5))
snps_smoking_age_init_buestguess_children$qlet<-substr(snps_smoking_age_init_buestguess_children$children_snp_age_init_FID, 6, 6)
snps_smoking_age_init_buestguess_children<-snps_smoking_age_init_buestguess_children[-c(1:6)]

snps_smoking_cessation_buestguess_children$aln <-as.numeric(substr(snps_smoking_cessation_buestguess_children$children_snp_cessation_FID, 1, 5))
snps_smoking_cessation_buestguess_children$qlet<-substr(snps_smoking_cessation_buestguess_children$children_snp_cessation_FID, 6, 6)
snps_smoking_cessation_buestguess_children<-snps_smoking_cessation_buestguess_children[-c(1:6)]

snps_smoking_cigs_pd_buestguess_children$aln <-as.numeric(substr(snps_smoking_cigs_pd_buestguess_children$children_snp_cigspd_FID, 1, 5))
snps_smoking_cigs_pd_buestguess_children$qlet<-substr(snps_smoking_cigs_pd_buestguess_children$children_snp_cigspd_FID, 6, 6)
snps_smoking_cigs_pd_buestguess_children<-snps_smoking_cigs_pd_buestguess_children[-c(1:6)]

snps_smoking_initiation_buestguess_children$aln <-as.numeric(substr(snps_smoking_initiation_buestguess_children$children_snp_initiation_FID, 1, 5))
snps_smoking_initiation_buestguess_children$qlet<-substr(snps_smoking_initiation_buestguess_children$children_snp_initiation_FID, 6, 6)
snps_smoking_initiation_buestguess_children<-snps_smoking_initiation_buestguess_children[-c(1:6)]

snps_pa_sedentary_buestguess_children$aln <-as.numeric(substr(snps_pa_sedentary_buestguess_children$children_snp_sedentary_FID, 1, 5))
snps_pa_sedentary_buestguess_children$qlet<-substr(snps_pa_sedentary_buestguess_children$children_snp_sedentary_FID, 6, 6)
snps_pa_sedentary_buestguess_children<-snps_pa_sedentary_buestguess_children[-c(1:6)]

snps_pa_overall_activity_children$aln <-as.numeric(substr(snps_pa_overall_activity_children$children_snp_overallact_FID, 1, 5))
snps_pa_overall_activity_children$qlet<-substr(snps_pa_overall_activity_children$children_snp_overallact_FID, 6, 6)
snps_pa_overall_activity_children<-snps_pa_overall_activity_children[-c(1:6)]

```

## Merge
```{r eval=FALSE, include=TRUE}
dat<-left_join(dat,snps_alcohol_buestguess_mother[,-which(colnames(snps_alcohol_buestguess_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_alcohol_buestguess_children, by =c("aln", "qlet"))

dat<-left_join(dat,snps_caffeine_buestguess_mother[,-which(colnames(snps_caffeine_buestguess_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_caffeine_buestguess_children, by =c("aln", "qlet"))

dat<-left_join(dat,snps_smoking_age_init_buestguess_mother[,-which(colnames(snps_smoking_age_init_buestguess_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_smoking_age_init_buestguess_children, by =c("aln", "qlet"))

dat<-left_join(dat,snps_smoking_cessation_buestguess_mother[,-which(colnames(snps_smoking_cessation_buestguess_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_smoking_cessation_buestguess_children, by =c("aln", "qlet"))

dat<-left_join(dat,snps_smoking_cigs_pd_buestguess_mother[,-which(colnames(snps_smoking_cigs_pd_buestguess_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_smoking_cigs_pd_buestguess_children, by =c("aln", "qlet"))

dat<-left_join(dat,snps_smoking_initiation_buestguess_mother[,-which(colnames(snps_smoking_initiation_buestguess_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_smoking_initiation_buestguess_children, by =c("aln", "qlet"))

dat<-left_join(dat,snps_pa_sedentary_buestguess_mother[,-which(colnames(snps_pa_sedentary_buestguess_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_pa_sedentary_buestguess_children, by =c("aln", "qlet"))

dat<-left_join(dat,snps_pa_overall_activity_mother[,-which(colnames(snps_pa_overall_activity_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,snps_pa_overall_activity_children, by =c("aln", "qlet"))

```

# Principal components {.tabset}
```{r eval=FALSE, include=TRUE}
principal_components_mother<-read_dta("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/principal_components/final_pcs/mums_pcs.dta")
principal_components_child<-read_dta("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/principal_components/final_pcs/children_pcs.dta")


dat<-left_join(dat,principal_components_mother[,-which(colnames(principal_components_mother) =="qlet")], by =c("aln"))
dat<-left_join(dat,principal_components_child, by =c("aln", "qlet"))

```


# Save file {.tabset}
You must be connected to the server: smb://rdsfcifs.acrc.bris.ac.uk/MRC-IEU-research/
```{r eval=FALSE, include=TRUE}
saveRDS(dat,"/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/alspac_pheno.rds")
```

# Short cut to working on this dataset {.tabset}

## Just dat

If you don't want/need to run everything from scratch, you can read 'dat' from the RDSF folder and get ready to add more ALSPAC variables using the following:

```{r eval=FALSE, include=TRUE}

dat <- readRDS("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/alspac_pheno.rds")

library("tidyverse")

library(alspac)

setDataDir("/Volumes/ALSPAC-Data/") #telling the package where the data is stored (make sure you are connected to the ALSPAC server before running this)

data(current)
data(useful)
```

## Generate raw_dat more quickly

If you need to generate the whole file again, it can be a pain and take a long time to keep looking for variables and extracting them for different categories of variable. Instead, you could just extract the variables once using the code below:

```{r eval=FALSE, include=TRUE}
library("tidyverse")
library(haven) #load package to read dta files

cp<-read_dta("/Volumes/sscm alspac/Data/Current/Other/Cohort Profile/cp_2b.dta")

cp$covs_sex <- NA
cp$covs_sex[cp$kz021==2]<-"female" #Female
cp$covs_sex[cp$kz021==1]<-"male" #Male

cp$survive_one_year <-NA
cp$survive_one_year[cp$kz011b==1] <-1
cp$survive_one_year[cp$kz011b==2] <-0

mz<-read_dta("/Volumes/sscm alspac/Data/Current/Other/Sample Definition/mz_5a.dta") #child sample definition
mz <- mz[mz$mz001==1,] #select core sample pregnancies; n=14,541
mz <- mz[-which(mz$mz010<1),] #remove 69 pregnancies with no known outcome; n=14,472
kz<-read_dta("/Volumes/sscm alspac/Data/Current/Other/Sample Definition/kz_5c.dta") #child sample definition
kz <- kz[kz$kz001==1,] #select core sample children; n=14,676

mult_pregs_core<-read_dta("/Volumes/sscm alspac/Data/Useful_data/Multiple pregs/mult_pregs in alspac_CORE.dta") #definition of multiple pregnancies within cohort
mult_pregs_core<-mult_pregs_core[-which(mult_pregs_core$aln %in% c(36891,45649)),] #remove 36891 from multiple pregnancy file as it doesn't appear in mz or kz but is linked as a sib to 45649 (who does appear in mz)
mult_pregs_core<-left_join(mult_pregs_core,mz,by="aln")
mult_pregs_core<-mult_pregs_core[order(mult_pregs_core$mz024b,mult_pregs_core$mz024a),]#order by year and month of delivery
mult_pregs_core$pregnancy_order_no <-1 #first child
mult_pregs_core$pregnancy_order_no[duplicated(mult_pregs_core$mult_no)]<-2 #second child
mult_pregs_core[-which(is.na(mult_pregs_core$aln3)),][3,"pregnancy_order_no"]<-3 #third child
mult_pregs_core<-mult_pregs_core[,c("aln","pregnancy_order_no","mult_no","aln2","aln3")]
mz<-full_join(mz,mult_pregs_core,by="aln")
mz$pregnancy_order_no[is.na(mz$pregnancy_order_no)]<-0 #0 for women with only one pregnancy in core sample
dat<-full_join(kz,mz,by="aln")
dat<-left_join(dat,cp[,-which(names(cp)%in%names(dat)[-which(names(dat) =="aln_qlet")])],by="aln_qlet")
#generate unique identifier for children that incorporates mother ID, pregnancy ID (aln) and twin ID (qlet)
dat$mother_aln_qlet <- paste0(dat$pregnancy_order_no,"_",dat$aln_qlet)
#rename mult_no to siblings_mother_id can be used to link siblings
names(dat)[names(dat)=="mult_no"]<-"siblings_mother_id"
dat$multiple_pregnancy <- dat$mz010a
#live births (could be useful outcome, or way to select sample)
dat$live_birth <-NA
dat$live_birth[dat$kz011==1]<-1
dat$live_birth[dat$kz011==2]<-0
#miscarriage or termination before 20 weeks vs live birth (could be useful outcome, or way to select sample)
dat$miscarriage_termination_b420wks <- NA
dat$miscarriage_termination_b420wks[dat$kz011==0]<-1
dat$miscarriage_termination_b420wks[dat$live_birth==1]<-0
#fetal death/stillbirth after 20 weeks vs live birth (could be useful outcome, or way to select sample)
dat$fetal_death_after_20wks <- NA
dat$fetal_death_after_20wks[dat$kz011==1]<-1
dat$fetal_death_after_20wks[dat$live_birth==1]<-0
#termination for congenital malformation (could be useful outcome, or way to select sample)
dat$congenital_malf_termination <-NA
dat$congenital_malf_termination[dat$mz011b==1]<-1
dat$congenital_malf_termination[dat$mz011b==3]<-0
dat$congenital_malf_termination[dat$mz011b==7]<-0
#death with a congenital defect up to one year (vs deaths before 1 with no congenital malformation)
dat$congenital_malf_death_before_one_year <-NA
dat$congenital_malf_death_before_one_year[dat$kz017==1] <-1
dat$congenital_malf_death_before_one_year[dat$kz017==2] <-0
#birthweight
dat$anthro_birthweight <-dat$kz030
dat$anthro_birthweight[dat$anthro_birthweight<0]<-NA #removing missing data
dat$anthro_birthweight_zscore <-scale(dat$anthro_birthweight)
#neonatal measurements age in days
dat$covs_age_birthmeasures_days <- dat$kz028
dat$covs_age_birthmeasures_days[dat$covs_age_birthmeasures_days<0]<-NA
#head circumference in cm
dat$anthro_head_circ_birth <- dat$kz031
dat$anthro_head_circ_birth[dat$anthro_head_circ_birth<0]<-NA
dat$anthro_head_circ_birth_zscore <-scale(dat$anthro_head_circ_birth)
#crown-heel in cm
dat$anthro_crown_heel_birth <- dat$kz032
dat$anthro_crown_heel_birth[dat$anthro_crown_heel_birth<0]<-NA
dat$anthro_crown_heel_birth_zscore <-scale(dat$anthro_crown_heel_birth)
#mother's age at conception (note that to preserve confidentiality these data have been effectively winsorised)
dat$covs_age_mother_conception <- dat$mz028a
dat$covs_age_mother_conception[dat$covs_age_mother_conception<0]<-NA
#mother's age at delivery (note that to preserve confidentiality these data have been effectively winsorised)
dat$covs_age_mother_delivery <- dat$mz028b
dat$covs_age_mother_delivery[dat$covs_age_mother_delivery<0]<-NA

dat<-dat[,-grep(names(dat),pattern="kz|mz")]

#################################################################################
# Make list of all other variables we want, and extract with the alspac package #
#################################################################################

time_vars <- c("a902","b924","c991","d990","e699","pa900","pa901","pb900","pb901")
parent_status_vars <- c("a521","a522")
parent_smoking_vars <-c("b650")
parent_smoking_vars <-c(parent_smoking_vars,c("b651","b652"))
parent_smoking_vars <-c(parent_smoking_vars,c("b660","b663"))
parent_smoking_vars <-c(parent_smoking_vars,c("b669"))
parent_smoking_vars <-c(parent_smoking_vars,c("b665","b667"))
parent_smoking_vars <-c(parent_smoking_vars,c("c482"))
parent_smoking_vars <-c(parent_smoking_vars,c("e178"))
parent_smoking_vars <-c(parent_smoking_vars,c("b670","b671"))
parent_smoking_vars <-c(parent_smoking_vars,c("b695","c481a"))
parent_smoking_vars <-c(parent_smoking_vars,c("f620", "g820"))
parent_smoking_vars <-c(parent_smoking_vars,c("b683","b685","b690","pb071"))
parent_smoking_vars <-c(parent_smoking_vars,c("pb072","b691","b692"))
parent_smoking_vars <-c(parent_smoking_vars,c("pb077","pb078","pb079","pb075","pb076"))
parent_smoking_vars <-c(parent_smoking_vars,c("e185", "e186","pc250","pc260"))
parent_smoking_vars <-c(parent_smoking_vars,c("pc251", "pd620", "pe450", "ph6180", "pj5050", "pj5051", "pl5010", "pp6020"))
parent_alcohol_vars <-c("b720")
parent_alcohol_vars <-c(parent_alcohol_vars,c("a261", "b721", "c373"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("b722"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("b754","b757","b760","b769", "b775"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("e220"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("b723"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("c360"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("e221", "e222", "f601", "f625", "f626", "g822", "g823", "g825"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("pb099"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("b730"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("pb100"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("pc280"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("pb101"))
parent_alcohol_vars <-c(parent_alcohol_vars,c("pc281","pd625","pe452","pe411","f600","g750","h602","p3190"))
parent_caffeine_vars <-c("a228", "a222", "a225", "a231", "a234", "a237")
parent_caffeine_vars <-c(parent_caffeine_vars,c("b748", "b742", "b745", "b751", "b763", "b766"))
parent_caffeine_vars <-c(parent_caffeine_vars,c("c300", "c303", "c305", "c309", "c310", "c312", "c317"))
parent_caffeine_vars <-c(parent_caffeine_vars,c("e157","e150","e151","e152","e154","e155","e153","e156"))
parent_caffeine_vars <-c(parent_caffeine_vars,c("pb054", "pb057", "pb058", "pb060a", "pb061", "pb062", "pb063", "pb064", "pb066","pb056","pb056a","pb060","pb065","pb065a"))
phys_act_vars <- c("pb013","pb014","b555","b556","b557","b635","b634","b862","c503","c504","g761","g762")
child_anthropometry_vars <-c("bestgest","centiles")
child_anthropometry_vars <-c(child_anthropometry_vars, "f7ms014","km6003","km6013","km6023","km6033","kp7003","kp7008")
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf054", "cf055", "cf056", "cf057", "cf058", "cf059", "f7ms010","f8lf020", "f9ms010","fdms010","kn7000","km6002","km6012","km6022","km6032","kp7002","kp7007","ku783","ku788","ku793","ku798"))
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf075","cf076","cf077","cf078","cf079","f7ms018","f9ms018","fdms018","kn7003"))
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf040","cf041","cf042","cf043","cf044","cf045","cf046","cf047","cf048","cf049","f7ms026","f8lf021","f9ms026","fdms026","km6001","km6011","km6021","km6031","kn7001","kp7001","kp7006","ku782","ku787","ku792","ku797"))
child_anthropometry_vars <-c(child_anthropometry_vars,c("cf060","cf061","cf062","cf063","cf064","cf065","cf066","cf067","cf068","cf069","f7ms026a","f9ms026a","fdms026a"))
child_anthropometry_vars <-c(child_anthropometry_vars,c("f9dx135"))
age_vars <- c("cf014","cf015", "cf016", "cf017","cf018","f7003c","f9003c")
child_neuro_vars <-c("j557a","j557b","j557c","j557d","j557e","j557f","kq348a","kq348b","kq348c","kq348d","kq348e","kq348f","ku705b","ku706b","ku707b","ku708b","ku709b","ku710b","kr554a","kr554b","cf811","cf812","cf813","f8ws110","f8ws111","f8ws112","f8ws115","fddp130")
child_neuro_vars <-c(child_neuro_vars,"j914","kq998a","ku991a","kr991a","f8003c","fd003c","cf018")
immuno_vars <- unique(c("ka249","ka252","kb086","kb089","kd085","kd088","kf110","kj100","kl100","kn1120","kq035a","kq090","kr042a","ks1042","ks1280","kv1070","kv1110","kv1111","kb053","kd051b","kd070","kf063","kj043","kj090","kl031","kl080","kn1031","kn1110","kp3030","kq034a","kq024a","kq070","kr031a","kr041a","kr050","kr105a","ks1031","ks1041","ks1240","ks1260","kv1060","kv1070","kv1080","kb419","kk262a","kk286","kk287","kk288","kk289","kk290","km2200","km2231","km2232","km2233","km2234","km2235","kq195a","kq217","kq218","kq219","kq220","kq221","ks3000","ks3031","ks3032","ks3033","ks3034","ks3035","kv1050","kv1059","ks3030"))
immuno_vars <-c(immuno_vars, "ka497", "kb879a","kd990","kf999", "kj999a","kk998a","kl991a","km9991a", "kn9991a","kp9991a","ks9991a","kv9991a")
cov_vars <- c("c800","c801","c804","pa910","pc996","c666a","c645a","c755","c765","b032","a525","pa065","a522","a524","b040","b041","b042","pa064","pa173a","d153a","pa174a","d154a","pa210","d190","d040","b613","pb202","c768","b560","pa189a","d169a","d152a","pa172a","pa190a","d170a","pa191a","d171a","pa192a","d172a","b140","b141","b142","b143","b144","b149","c110","c111","c112","c113","c114","c115","dw021","dw002","paw002","paw010","b700","b701","b702","b706","b707","b708","b709","b710","b711","b712","b713","b714","pb088","pb089","pb090","pb091","pb092","pb093","pb094","pb095","pb096","pb097","pb098","bestgest","kb277","kc403","kb280","kc404","kg475","kk800","kq885","kt6400","HDP","pregnancy_diabetes","kb390","kb390a","kb387","kb387a","kc514", "kc544","kc541","kc572","kc574","kc570","ke341","ke350","ke352","ke354","ke339","ke321","kc363","ke195","ke195a","ke196","ke196a","kb549","kb551","kb388","kb385","kb361","kc542","kc539","kc361","ke340","ke338","ke320","kg423","kg420","kg369a","kk645","kk640","kk598","kq830","kq823","kq798a","kt6290","kt6280","kt6193","kg174","kk312","km3030","km3031","kp5090","kp5091","kt1210","kt1211","kc512")
negcon_vars <- c("g572","g572a","h462","h462a","g571","g571a","h461","h461a","kf527","kf528","kj517","kj518")
cardiomet_vars <- c("cf124", "cf134", "cf144", "f7sa022", "f9sa022", "fdar118","cf143", "cf133", "cf123", "f7sa021", "f9sa021", "fdar117","trig_cord", "Trig_CIF31", "Trig_CIF43", "TRIG_F7", "trig_f9","HDL_cord", "HDL_CIF31", "HDL_CIF43", "HDL_F7", "HDL_f9","LDL_cord", "LDL_CIF31", "LDL_CIF43", "LDL_F7", "LDL_f9","chol_cord", "Chol_CIF31", "Chol_CIF43", "CHOL_F7", "CHOL_f9","Glc_F7","Insulin_cord", "insulin_F9","CRP_cord", "CRP_f9","ApoAI_f9","ApoB_f9","ApoA1_F7","ApoB_F7","IL6_f9")
## NEED TO ADD FFQ VARIABLES TOO

all_vars<-unique(c(unlist(lapply(ls()[grep(ls(),pattern="vars")],get))))


library(alspac)
setDataDir("/Volumes/sscm alspac/Data/") #telling the package where the data is stored (make sure you are connected to the ALSPAC server before running this)
options(mc.cores=4)
#current <- createDictionary("Current", name="current")
#useful <- createDictionary("Useful_data", name="useful")

data(current)
data(useful)

meta_data <- findVars(all_vars)
meta_data <- subset(meta_data,subset=tolower(name) %in% all_vars)
meta_data <- meta_data[which(meta_data$cat2 %in% c("Clinic","Quest","bestgest","birthweight_centile")),]
meta_data <- meta_data[!duplicated(meta_data$name),]
raw_dat <- extractVars(meta_data)

raw_dat <- left_join(dat,raw_dat,by=c("aln","qlet")) 

require(readstata13)
A<-read.dta13("/Volumes/sscm alspac/Data/Current/Quest/Partner/pa_4a.dta")
B<-read.dta13("/Volumes/sscm alspac/Data/Current/Quest/Partner/pb_4b.dta")
C<-read.dta13("/Volumes/sscm alspac/Data/Current/Quest/Partner/pc_3a.dta")

A[A=="-1"]<-NA #changing -1 to missing
B[B=="-1"]<-NA #changing -1 to missing
C[C=="-1"]<-NA #changing -1 to missing

raw_dat$paternal_participation <- ifelse(raw_dat$aln %in% unique(c(A$aln,B$aln,C$aln)),1,0)
rm(A,B,C)

saveRDS(raw_dat,"/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/alspac/alspac_pheno_raw.rds")

dat <- raw_dat


#### NOW YOU CAN COPY AND PASTE THE SECTIONS OF THE SCRIPT THAT DERIVE THE CLEANED VARIABLES ####

```

# TO DO

* Add education variables


